<!DOCTYPE html>
<!-- saved from url=(0087)https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader -->
<html lang="en" class="js webfonts-loaded"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="applicable-device" content="pc,mobile">
        <meta name="viewport" content="width=device-width, initial-scale=1">

        <title>Ultrafast and memory-efficient alignment of short DNA sequences to the human genome | Genome Biology | Full Text</title>

        
    
    <meta name="citation_abstract" content="Bowtie is an ultrafast, memory-efficient alignment program for aligning short DNA sequence reads to large genomes. For the human genome, Burrows-Wheeler indexing allows Bowtie to align more than 25 million reads per CPU hour with a memory footprint of approximately 1.3 gigabytes. Bowtie extends previous Burrows-Wheeler techniques with a novel quality-aware backtracking algorithm that permits mismatches. Multiple processor cores can be used simultaneously to achieve even greater alignment speeds. Bowtie is open source 
                  http://bowtie.cbcb.umd.edu
                  
                .">
    <meta name="journal_id" content="13059">
    <meta name="dc.title" content="Ultrafast and memory-efficient alignment of short DNA sequences to the human genome">
    <meta name="dc.source" content="Genome Biology 2009 10:3">
    <meta name="dc.format" content="text/html">
    <meta name="dc.publisher" content="BioMed Central">
    <meta name="dc.date" content="2009-03-04">
    <meta name="dc.type" content="OriginalPaper">
    <meta name="dc.language" content="En">
    <meta name="dc.copyright" content="2009 Langmead et al.; licensee BioMed Central Ltd.">
    <meta name="dc.rights" content="http://creativecommons.org/licenses/by/2.0">
    <meta name="dc.rightsAgent" content="reprints@biomedcentral.com">
    <meta name="dc.description" content="Bowtie is an ultrafast, memory-efficient alignment program for aligning short DNA sequence reads to large genomes. For the human genome, Burrows-Wheeler indexing allows Bowtie to align more than 25 million reads per CPU hour with a memory footprint of approximately 1.3 gigabytes. Bowtie extends previous Burrows-Wheeler techniques with a novel quality-aware backtracking algorithm that permits mismatches. Multiple processor cores can be used simultaneously to achieve even greater alignment speeds. Bowtie is open source http://bowtie.cbcb.umd.edu .">
    <meta name="prism.issn" content="1474-760X">
    <meta name="prism.publicationName" content="Genome Biology">
    <meta name="prism.publicationDate" content="2009-03-04">
    <meta name="prism.volume" content="10">
    <meta name="prism.number" content="3">
    <meta name="prism.section" content="OriginalPaper">
    <meta name="prism.startingPage" content="1">
    <meta name="prism.endingPage" content="10">
    <meta name="prism.copyright" content="2009 Langmead et al.; licensee BioMed Central Ltd.">
    <meta name="prism.rightsAgent" content="reprints@biomedcentral.com">
    <meta name="prism.url" content="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25">
    <meta name="prism.doi" content="doi:10.1186/gb-2009-10-3-r25">
    <meta name="citation_pdf_url" content="https://genomebiology.biomedcentral.com/counter/pdf/10.1186/gb-2009-10-3-r25">
    <meta name="citation_fulltext_html_url" content="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25">
    <meta name="citation_journal_title" content="Genome Biology">
    <meta name="citation_journal_abbrev" content="Genome Biol">
    <meta name="citation_publisher" content="BioMed Central">
    <meta name="citation_issn" content="1474-760X">
    <meta name="citation_title" content="Ultrafast and memory-efficient alignment of short DNA sequences to the human genome">
    <meta name="citation_volume" content="10">
    <meta name="citation_issue" content="3">
    <meta name="citation_publication_date" content="2009/03">
    <meta name="citation_online_date" content="2009/03/04">
    <meta name="citation_firstpage" content="1">
    <meta name="citation_lastpage" content="10">
    <meta name="citation_article_type" content="Software">
    <meta name="citation_fulltext_world_readable" content="">
    <meta name="citation_language" content="en">
    <meta name="dc.identifier" content="doi:10.1186/gb-2009-10-3-r25">
    <meta name="DOI" content="10.1186/gb-2009-10-3-r25">
    <meta name="size" content="162539">
    <meta name="citation_doi" content="10.1186/gb-2009-10-3-r25">
    <meta name="citation_springer_api_url" content="http://api.springer.com/xmldata/jats?q=doi:10.1186/gb-2009-10-3-r25&amp;api_key=">
    <meta name="description" content="Bowtie is an ultrafast, memory-efficient alignment program for aligning short DNA sequence reads to large genomes. For the human genome, Burrows-Wheeler indexing allows Bowtie to align more than 25 million reads per CPU hour with a memory footprint of approximately 1.3 gigabytes. Bowtie extends previous Burrows-Wheeler techniques with a novel quality-aware backtracking algorithm that permits mismatches. Multiple processor cores can be used simultaneously to achieve even greater alignment speeds. Bowtie is open source http://bowtie.cbcb.umd.edu .">
    <meta name="dc.creator" content="Langmead, Ben">
    <meta name="dc.creator" content="Trapnell, Cole">
    <meta name="dc.creator" content="Pop, Mihai">
    <meta name="dc.creator" content="Salzberg, Steven L">
    <meta name="dc.subject" content="Animal Genetics and Genomics">
    <meta name="dc.subject" content="Human Genetics">
    <meta name="dc.subject" content="Plant Genetics and Genomics">
    <meta name="dc.subject" content="Microbial Genetics and Genomics">
    <meta name="dc.subject" content="Bioinformatics">
    <meta name="dc.subject" content="Evolutionary Biology">
    <meta name="citation_reference" content="citation_journal_title=Nat Biotechnol; citation_title=A Bayesian deconvolution strategy for immunoprecipitation-based DNA methylome analysis.; citation_author=TA Down, VK Rakyan, DJ Turner, P Flicek, H Li, E Kulesha, S Graf, N Johnson, J Herrero, EM Tomazou, NP Thorne, L Backdahl, M Herberth, KL Howe, DK Jackson, MM Miretti, JC Marioni, E Birney, TJ Hubbard, R Durbin, S Tavare, S Beck; citation_volume=26; citation_publication_date=2008; citation_pages=779-785; citation_doi=10.1038/nbt1414; citation_id=CR1">
    <meta name="citation_reference" content="citation_journal_title=Science; citation_title=Genome-wide mapping of in vivo protein-DNA interactions.; citation_author=DS Johnson, A Mortazavi, RM Myers, B Wold; citation_volume=316; citation_publication_date=2007; citation_pages=1497-1502; citation_doi=10.1126/science.1141319; citation_id=CR2">
    <meta name="citation_reference" content="citation_journal_title=Genome Res; citation_title=RNA-seq: an assessment of technical reproducibility and comparison with gene expression arrays.; citation_author=JC Marioni, CE Mason, SM Mane, M Stephens, Y Gilad; citation_volume=18; citation_publication_date=2008; citation_pages=1509-1517; citation_doi=10.1101/gr.079558.108; citation_id=CR3">
    <meta name="citation_reference" content="citation_journal_title=Nature; citation_title=Accurate whole human genome sequencing using reversible terminator chemistry.; citation_author=DR Bentley, S Balasubramanian, HP Swerdlow, GP Smith, J Milton, CG Brown, KP Hall, DJ Evers, CL Barnes, HR Bignell, JM Boutell, J Bryant, RJ Carter, R Keira Cheetham, AJ Cox, DJ Ellis, MR Flatbush, NA Gormley, SJ Humphray, LJ Irving, MS Karbelashvili, SM Kirk, H Li, X Liu, KS Maisinger, LJ Murray, B Obradovic, T Ost, ML Parkinson, MR Pratt; citation_volume=456; citation_publication_date=2008; citation_pages=53-59; citation_doi=10.1038/nature07517; citation_id=CR4">
    <meta name="citation_reference" content="citation_journal_title=Nature; citation_title=DNA sequencing of a cytogenetically normal acute myeloid leukaemia genome.; citation_author=TJ Ley, ER Mardis, L Ding, B Fulton, MD McLellan, K Chen, D Dooling, BH Dunford-Shore, S McGrath, M Hickenbotham, L Cook, R Abbott, DE Larson, DC Koboldt, C Pohl, S Smith, A Hawkins, S Abbott, D Locke, LW Hillier, T Miner, L Fulton, V Magrini, T Wylie, J Glasscock, J Conyers, N Sander, X Shi, JR Osborne, P Minx; citation_volume=456; citation_publication_date=2008; citation_pages=66-72; citation_doi=10.1038/nature07485; citation_id=CR5">
    <meta name="citation_reference" content="citation_journal_title=Nature; citation_title=The diploid genome sequence of an Asian individual.; citation_author=J Wang, W Wang, R Li, Y Li, G Tian, L Goodman, W Fan, J Zhang, J Li, J Zhang, Y Guo, B Feng, H Li, Y Lu, X Fang, H Liang, Z Du, D Li, Y Zhao, Y Hu, Z Yang, H Zheng, I Hellmann, M Inouye, J Pool, X Yi, J Zhao, J Duan, Y Zhou, J Qin; citation_volume=456; citation_publication_date=2008; citation_pages=60-65; citation_doi=10.1038/nature07484; citation_id=CR6">
    <meta name="citation_reference" content="citation_journal_title=Genome Res; citation_title=Mapping short DNA sequencing reads and calling variants using mapping quality scores.; citation_author=H Li, J Ruan, R Durbin; citation_volume=18; citation_publication_date=2008; citation_pages=1851-1858; citation_doi=10.1101/gr.078212.108; citation_id=CR7">
    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=SOAP: short oligonucleotide alignment program.; citation_author=R Li, Y Li, K Kristiansen, J Wang; citation_volume=24; citation_publication_date=2008; citation_pages=713-714; citation_doi=10.1093/bioinformatics/btn025; citation_id=CR8">
    <meta name="citation_reference" content="citation_journal_title=Science; citation_title=DNA sequencing. A plan to capture human diversity in 1000 genomes.; citation_author=J Kaiser; citation_volume=319; citation_publication_date=2008; citation_pages=395; citation_doi=10.1126/science.319.5862.395; citation_id=CR9">
    <meta name="citation_reference" content="citation_journal_title=BMC Bioinformatics; citation_title=Using quality scores and longer reads improves accuracy of Solexa read mapping.; citation_author=AD Smith, Z Xuan, MQ Zhang; citation_volume=9; citation_publication_date=2008; citation_pages=128; citation_doi=10.1186/1471-2105-9-128; citation_id=CR10">
    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=ZOOM! Zillions Of Oligos Mapped.; citation_author=H Lin, Z Zhang, MQ Zhang, B Ma, M Li; citation_volume=24; citation_publication_date=2008; citation_pages=2431-2437; citation_doi=10.1093/bioinformatics/btn416; citation_id=CR11">
    <meta name="citation_reference" content="SHRiMP - SHort Read Mapping Package. [
                    http://compbio.cs.toronto.edu/shrimp/
                    
                  ]">
    <meta name="citation_reference" content="citation_journal_title=Inf Process Lett; citation_title=Fast and practical approximate string matching.; citation_author=RA Baeza-Yates, CH Perleberg; citation_volume=59; citation_publication_date=1996; citation_pages=21-27; citation_doi=10.1016/0020-0190(96)00083-X; citation_id=CR13">
    <meta name="citation_reference" content="citation_journal_title=Fundam Inf; citation_title=Better Filtering with Gapped q-Grams.; citation_author=S Burkhardt, J Kärkkäinen; citation_volume=56; citation_publication_date=2003; citation_pages=51-70; citation_id=CR14">
    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=PatternHunter: faster and more sensitive homology search.; citation_author=B Ma, J Tromp, M Li; citation_volume=18; citation_publication_date=2002; citation_pages=440-445; citation_doi=10.1093/bioinformatics/18.3.440; citation_id=CR15">
    <meta name="citation_reference" content="citation_journal_title=J Mol Biol; citation_title=Identification of common molecular subsequences.; citation_author=TF Smith, MS Waterman; citation_volume=147; citation_publication_date=1981; citation_pages=195-197; citation_doi=10.1016/0022-2836(81)90087-5; citation_id=CR16">
    <meta name="citation_reference" content="citation_title=A Block Sorting Lossless Data Compression Algorithm. Technical Report 124; citation_publication_date=1994; citation_id=CR17; citation_author=M Burrows; citation_author=DJ Wheeler; citation_publisher=Digital Equipment Corporation">
    <meta name="citation_reference" content="Ferragina P, Manzini G: Opportunistic data structures with applications. [
                    http://web.unipmn.it/~manzini/papers/focs00draft.pdf
                    
                  ]">
    <meta name="citation_reference" content="citation_title=An experimental study of an opportunistic index.; citation_inbook_title=Proceedings of the Twelfth Annual ACM-SIAM Symposium on Discrete algorithms; citation_publication_date=2001; citation_pages=269-278; citation_id=CR19; citation_author=P Ferragina; citation_author=G Manzini; citation_publisher=Society for Industrial and Applied Mathematics">
    <meta name="citation_reference" content="citation_journal_title=Genome Res; citation_title=Annotating large genomes with exact word matches.; citation_author=J Healy, EE Thomas, JT Schwartz, M Wigler; citation_volume=13; citation_publication_date=2003; citation_pages=2306-2315; citation_doi=10.1101/gr.1350803; citation_id=CR20">
    <meta name="citation_reference" content="citation_journal_title=J Comput Biol; citation_title=Space-efficient whole genome comparisons with Burrows-Wheeler transforms.; citation_author=RA Lippert; citation_volume=12; citation_publication_date=2005; citation_pages=407-415; citation_doi=10.1089/cmb.2005.12.407; citation_id=CR21">
    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=Optimized design and assessment of whole genome tiling arrays.; citation_author=S Graf, FG Nielsen, S Kurtz, MA Huynen, E Birney, H Stunnenberg, P Flicek; citation_volume=23; citation_publication_date=2007; citation_pages=i195-i204; citation_doi=10.1093/bioinformatics/btm200; citation_id=CR22">
    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=Compressed indexing and local alignment of DNA.; citation_author=TW Lam, WK Sung, SL Tam, CK Wong, SM Yiu; citation_volume=24; citation_publication_date=2008; citation_pages=791-797; citation_doi=10.1093/bioinformatics/btn032; citation_id=CR23">
    <meta name="citation_reference" content="citation_journal_title=Genome Res; citation_title=Base-calling of automated sequencer traces using phred. II. Error probabilities.; citation_author=B Ewing, P Green; citation_volume=8; citation_publication_date=1998; citation_pages=186-194; citation_doi=10.1101/gr.8.3.186; citation_id=CR24">
    <meta name="citation_reference" content="Bowtie: An ultrafast memory-efficient short read aligner. [
                    http://bowtie.cbcb.umd.edu/
                    
                  ]">
    <meta name="citation_reference" content="citation_journal_title=Nat Genet; citation_title=Identification of somatically acquired rearrangements in cancer using genome-wide massively parallel paired-end sequencing.; citation_author=PJ Campbell, PJ Stephens, ED Pleasance, S O&#39;Meara, H Li, T Santarius, LA Stebbings, C Leroy, S Edkins, C Hardy, JW Teague, A Menzies, I Goodhead, DJ Turner, CM Clee, MA Quail, A Cox, C Brown, R Durbin, ME Hurles, PA Edwards, GR Bignell, MR Stratton, PA Futreal; citation_volume=40; citation_publication_date=2008; citation_pages=722-729; citation_doi=10.1038/ng.128; citation_id=CR26">
    <meta name="citation_reference" content="citation_journal_title=Nat Genet; citation_title=High-throughput sequencing provides insights into genome variation and evolution in Salmonella typhi.; citation_author=KE Holt, J Parkhill, CJ Mazzoni, P Roumagnac, FX Weill, I Goodhead, R Rance, S Baker, DJ Maskell, J Wain, C Dolecek, M Achtman, G Dougan; citation_volume=40; citation_publication_date=2008; citation_pages=987-993; citation_doi=10.1038/ng.195; citation_id=CR27">
    <meta name="citation_reference" content="citation_journal_title=Science; citation_title=The transcriptional landscape of the yeast genome defined by RNA sequencing.; citation_author=U Nagalakshmi, Z Wang, K Waern, C Shou, D Raha, M Gerstein, M Snyder; citation_volume=320; citation_publication_date=2008; citation_pages=1344-1349; citation_doi=10.1126/science.1158441; citation_id=CR28">
    <meta name="citation_reference" content="citation_journal_title=Theor Comput Sci; citation_title=Fast BWT in small space by blockwise suffix sorting.; citation_author=J Kärkkäinen; citation_volume=387; citation_publication_date=2007; citation_pages=249-257; citation_doi=10.1016/j.tcs.2007.07.018; citation_id=CR29">
    <meta name="citation_reference" content="citation_journal_title=BMC Bioinformatics; citation_title=SeqAn an efficient, generic C++ library for sequence analysis.; citation_author=A Doring, D Weese, T Rausch, K Reinert; citation_volume=9; citation_publication_date=2008; citation_pages=11; citation_doi=10.1186/1471-2105-9-11; citation_id=CR30">
    <meta name="citation_author" content="Langmead, Ben">
    <meta name="citation_author_institution" content="Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, USA">
    <meta name="citation_author" content="Trapnell, Cole">
    <meta name="citation_author_institution" content="Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, USA">
    <meta name="citation_author" content="Pop, Mihai">
    <meta name="citation_author_institution" content="Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, USA">
    <meta name="citation_author" content="Salzberg, Steven L">
    <meta name="citation_author_institution" content="Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, USA">
    


        <meta name="format-detection" content="telephone=no">

        
<link rel="apple-touch-icon" sizes="180x180" href="https://genomebiology.biomedcentral.com/static/img/favicons/bmc/apple-touch-icon-582ef1d0f5.png">
<link rel="icon" type="image/png" sizes="32x32" href="https://genomebiology.biomedcentral.com/static/img/favicons/bmc/favicon-32x32-5d7879efe1.png">
<link rel="icon" type="image/png" sizes="16x16" href="https://genomebiology.biomedcentral.com/static/img/favicons/bmc/favicon-16x16-c241ac1a2f.png">
<link rel="manifest" href="https://genomebiology.biomedcentral.com/static/app-bmc/manifest-3bb9ad383b.json">
<link rel="mask-icon" href="https://genomebiology.biomedcentral.com/static/img/favicons/bmc/safari-pinned-tab-427338756f.svg" color="#e6e6e6">
<link rel="shortcut icon" data-test="shortcut-icon" href="https://genomebiology.biomedcentral.com/static/img/favicons/bmc/favicon-1886637b78.ico">
<meta name="msapplication-config" content="/static/app-bmc/browserconfig-ecd9aa8205.xml">
<meta name="msapplication-TileColor" content="#1b3051">
<meta name="msapplication-TileImage" content="/static/img/favicons/bmc/app-icon-144x144.png">
<meta name="theme-color" content="#1b3051">


        
        <script async="" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/gtm.js.下载"></script><script async="" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/MathJax.js.下载"></script><script>(function(H){H.className=H.className.replace(/\bno-js\b/,'js')})(document.documentElement)</script><script src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/consent-bundle-15-12.js.下载" onload="initGTM(window,document,&#39;script&#39;,&#39;dataLayer&#39;,&#39;GTM-MRVXSHQ&#39;)"></script>

        


    
        <link rel="stylesheet" media="screen" href="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/core-article-945374050f.css">
    

    <link rel="stylesheet" media="screen" href="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/core-3eae6c91fc.css">
    <link rel="stylesheet" media="print" href="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/print-b8af42253b.css">
    

    
    
        <!-- This template is only used by BMC for now -->


    <style>@media only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark) { button{line-height:inherit}html,label{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif}html{-webkit-font-smoothing:subpixel-antialiased;box-sizing:border-box;color:#333;font-size:100%;height:100%;line-height:1.61803;overflow-y:scroll}*{box-sizing:inherit}body{background:#fff;margin:0;max-width:100%;min-height:100%}button,div,form,input{margin:0;padding:0}body,p{padding:0}a{color:#004b83;text-decoration:underline;text-decoration-skip-ink:auto}a>img{vertical-align:middle}h1,h2,h3{margin-top:0}h1{font-size:28px;font-weight:700}html.webfonts-loaded h1,html.webfonts-loaded h2,html.webfonts-loaded h3{font-family:Europa,Trebuchet MS}h1,h2,h3{color:#1b3051;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-style:normal;line-height:1.4}h2{font-size:26px;font-weight:700;margin-bottom:.5em}h3{font-size:18px;margin-bottom:.7em}p{margin:0}ul{margin-top:0}p{margin-bottom:1.5em}p:last-child{margin-bottom:0}p:empty{display:none}.c-ad{text-align:center}@media only screen and (min-width:320px){.c-ad{padding:8px}}.c-ad--728x90{background-color:#ccc;display:none}.c-ad--728x90 .c-ad__inner{min-height:calc(1.5em + 94px)}.c-ad--728x90 iframe{height:90px;max-width:970px}@media only screen and (min-width:768px){.js .c-ad--728x90{display:none}.js .u-show-following-ad+.c-ad--728x90{display:block}}.c-ad--300x250{background-color:#f2f2f2;display:none;padding:8px}.c-ad--300x250 .c-ad__inner{min-height:calc(1.5em + 254px)}@media only screen and (min-width:320px){.js .c-ad--300x250{display:block}}.c-ad iframe{border:0;overflow:auto;vertical-align:top}.c-ad__label{color:#333;font-weight:400;line-height:1.5;margin-bottom:4px}.c-ad__label,.c-skip-link{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:.875rem}.c-skip-link{background:#dceaf6;bottom:auto;color:#004b83;padding:8px;position:absolute;text-align:center;transform:translateY(-100%);z-index:9999}@media (prefers-reduced-motion:reduce){.c-skip-link{transition:top .3s ease-in-out 0s}}@media print{.c-skip-link{display:none}}.c-skip-link:link{color:#004b83}.c-dropdown__button:after{border-color:transparent transparent transparent #fff;border-style:solid;border-width:4px 0 4px 14px;content:"";display:block;height:0;margin-left:3px;width:0}.c-dropdown{display:inline-block;position:relative}.c-dropdown__button{background-color:transparent;border:0;display:inline-block;padding:0;white-space:nowrap}.c-dropdown__button:after{border-color:currentcolor transparent transparent;border-width:5px 4px 0 5px;display:inline-block;margin-left:8px;vertical-align:middle}.c-dropdown__menu{background-color:#fff;border:1px solid #d9d9d9;border-radius:3px;box-shadow:0 2px 6px rgba(0,0,0,.1);font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:14px;line-height:1.4;list-style:none;margin:0;padding:8px 0;position:absolute;top:100%;transform:translateY(8px);width:180px;z-index:100}.c-dropdown__menu:after,.c-dropdown__menu:before{border-style:solid;bottom:100%;content:"";display:block;height:0;left:16px;position:absolute;width:0}.c-dropdown__menu:before{border-color:transparent transparent #d9d9d9;border-width:0 9px 9px;transform:translateX(-1px)}.c-dropdown__menu:after{border-color:transparent transparent #fff;border-width:0 8px 8px}.c-dropdown__menu--right{left:auto;right:0}.c-dropdown__menu--right:after,.c-dropdown__menu--right:before{left:auto;right:16px}.c-dropdown__menu--right:before{transform:translateX(1px)}.c-dropdown__link{background-color:transparent;color:#004b83;display:block;padding:4px 16px;text-decoration:none}.c-header{background-color:#1b3051;border-bottom:4px solid #1b3051;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:14px;padding:16px 0}.c-header__container,.c-header__menu{align-items:center;display:flex;flex-wrap:wrap}@supports (gap:2em){.c-header__container,.c-header__menu{gap:2em 2em}}.c-header__menu{list-style:none;margin:0;padding:0}.c-header__item{color:#fff}@supports not (gap:2em){.c-header__item{margin-left:24px}}.c-header__container{justify-content:space-between;margin:0 auto;max-width:1280px;padding:0 16px}@supports not (gap:2em){.c-header__brand{margin-right:48px}}.c-header__brand a{display:block;text-decoration:none}.c-header__link{color:#fff}.c-header__link,.c-journal-title a{text-decoration:none}html.webfonts-loaded .c-journal-title{font-family:Europa,Trebuchet MS}.c-form-field{margin-bottom:1em}.c-form-field__label{color:#666;display:block;font-size:14px;margin-bottom:.4em}.c-form-field__input{border:1px solid #a6a6a6;border-radius:3px;box-shadow:inset 0 1px 3px 0 rgba(0,0,0,.21);font-size:14px;line-height:1.28571;padding:.75em 1em;vertical-align:middle;width:100%}.c-popup-search{background-color:#25426f;box-shadow:0 3px 3px -3px rgba(0,0,0,.21);padding:16px 0;position:relative;z-index:10}@media only screen and (min-width:1024px){.js .c-popup-search{position:absolute;top:100%;width:100%}.c-popup-search__container{margin:auto;max-width:70%}}.c-logo img{display:block}.ctx-search .c-form-field{margin-bottom:0}.ctx-search .c-form-field__label{color:#fff}.ctx-search .c-form-field__input{border-bottom-right-radius:0;border-top-right-radius:0;margin-right:0}.c-journal-header{overflow:hidden}.c-journal-header__inner{padding-bottom:8px;padding-top:8px}@media only screen and (min-width:540px){.c-journal-header__inner{display:flex;justify-content:space-between}}.c-journal-header__identity{background-size:auto 80px;min-height:16px;padding:6px 0}@media only screen and (min-width:540px){.c-journal-header__identity{background-position:0}}.c-journal-header__identity--default{background:url(/static/images/bmc/identities/patterns/default-7e3a6b1388.svg) 0 0 no-repeat #04caa8}.c-journal-title{color:#1b3051;display:flex;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:26px;font-style:normal;font-weight:700;line-height:1.4;margin-bottom:0}.c-journal-title>a{color:#0097ef;color:#1b3051;display:flex;text-decoration:none}.c-journal-title__text{align-self:center;display:block;flex:0 1 auto}.c-logo,.c-logo>a{align-items:baseline;display:flex}.c-logo__strapline{display:none}@media only screen and (min-width:540px){.c-logo__strapline{display:block;margin:0 0 0 16px;transform:translateY(1px)}}.c-navbar{background:#f2f2f2;border-bottom:1px solid #d9d9d9;border-top:1px solid #d9d9d9;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:14px;line-height:1.42857;padding:16px 0}.c-navbar__container{align-items:center;display:flex;flex-wrap:wrap;justify-content:space-between;margin:0 auto;max-width:1280px;padding:0 16px}.c-navbar__content{align-items:center;display:flex;flex:0 1 auto}@media only screen and (min-width:1024px){.c-navbar__content{display:flex!important}}.c-navbar__nav{display:flex;list-style:none;margin:0;padding:0}@media only screen and (min-width:1024px){.c-navbar__nav{transform:translateY(1px)}}.c-navbar__item{flex:1 0 auto}@media only screen and (max-width:539px){.c-navbar__item{font-size:.75rem}}.c-navbar__item:not(:first-child){margin-left:16px}.c-navbar__link{background:0 0;border:0;color:currentcolor;display:block;text-decoration:none;text-transform:capitalize}.c-navbar__link--is-shown{text-decoration:underline}.c-logo{display:block}html.webfonts-loaded .u-h3,html.webfonts-loaded .u-h4{font-family:Europa,Trebuchet MS}.u-button{align-items:center;background-color:#f2f2f2;background-image:linear-gradient(#fff,#f2f2f2);border:1px solid #ccc;border-radius:2px;cursor:pointer;display:inline-flex;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:1rem;justify-content:center;line-height:1.3;margin:0;padding:8px;position:relative;text-decoration:none;transition:all .25s ease 0s,color .25s ease 0s,border-color .25s ease 0s;width:auto}.u-button svg,.u-button--primary svg{fill:currentcolor}.u-button{color:#004b83}.u-button--primary{background-color:#33629d;background-image:linear-gradient(#4d76a9,#33629d);border:1px solid rgba(0,59,132,.5);color:#fff}.u-button--full-width{display:flex;width:100%}.u-clearfix:after,.u-clearfix:before{content:"";display:table}.u-clearfix:after{clear:both}.u-container{margin:0 auto;max-width:1280px;padding:0 16px}.u-display-flex{display:flex;width:100%}.u-align-items-center{align-items:center}.u-justify-content-space-between{justify-content:space-between}.u-flex-static{flex:0 0 auto}.u-display-none{display:none}.js .u-js-hide{display:none;visibility:hidden}.u-show{display:block;visibility:visible}@media print{.u-hide-print{display:none}}.u-icon{fill:currentcolor;display:inline-block;height:1em;transform:translate(0);vertical-align:text-top;width:1em}.u-list-reset{list-style:none;margin:0;padding:0}.u-position-relative{position:relative}.u-mt-16{margin-top:16px}.u-mt-32{margin-top:32px}.u-mr-24{margin-right:24px}.u-mr-48{margin-right:48px}.u-mb-32{margin-bottom:32px}.u-ml-8{margin-left:8px}.u-button-reset{background-color:transparent;border:0;padding:0}.u-text-sm{font-size:14px}.u-h3,.u-h4{color:#1b3051;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:18px;font-style:normal;line-height:1.4;margin-bottom:.7em}.u-h4{font-style:italic;font-weight:700}.u-vh-full{min-height:100vh}.u-hide{display:none;visibility:hidden}@media only screen and (min-width:1024px){.u-hide-at-lg{display:none;visibility:hidden}}@media only screen and (max-width:1023px){.u-hide-at-lt-lg{display:none;visibility:hidden}}.u-visually-hidden{clip:rect(0,0,0,0);border:0;height:1px;margin:-100%;overflow:hidden;padding:0;position:absolute!important;width:1px}body{font-size:1.125rem}.c-header__navigation{display:flex;gap:.5rem .5rem} }</style>
    <style>@media only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark) { .u-h3{font-family:Georgia,Palatino,serif;font-style:normal;line-height:1.4}.u-h3,.u-h4{margin-bottom:1em}.c-reading-companion__figure-title,.u-h4{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-style:normal;line-height:1.4}button{line-height:inherit}html,label{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif}html{-webkit-font-smoothing:subpixel-antialiased;box-sizing:border-box;color:#333;font-size:100%;height:100%;line-height:1.61803;overflow-y:scroll}*{box-sizing:inherit}body{background:#fff;margin:0;max-width:100%;min-height:100%}button,div,form,input{margin:0;padding:0}body,p{padding:0}a{color:#004b83;overflow-wrap:break-word;text-decoration:underline;text-decoration-skip-ink:auto;word-break:break-word}a>img{vertical-align:middle}h1,h2,h3{margin-top:0}h1{font-size:28px;font-weight:700}html.webfonts-loaded h1,html.webfonts-loaded h2,html.webfonts-loaded h3{font-family:Europa,Trebuchet MS}h1,h2,h3{color:#1b3051;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-style:normal;line-height:1.4}h2{font-size:26px;font-weight:700;margin-bottom:.5em}h3{font-size:18px;margin-bottom:.7em}p{margin:0}ul{margin-top:0}p{margin-bottom:1.5em;overflow-wrap:break-word;word-break:break-word}p:last-child{margin-bottom:0}p:empty{display:none}.u-h3{font-size:1.5rem}.u-h3,.u-h4{font-weight:700}.u-h4{font-size:1.25rem}.c-reading-companion__figure-title{font-size:1.25rem;font-weight:700}body{font-size:1.125rem}.c-article-header{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;margin-bottom:40px}.c-article-identifiers{color:#6f6f6f;display:flex;flex-wrap:wrap;font-size:1rem;line-height:1.3;list-style:none;margin:0 0 8px;padding:0}.c-article-identifiers__item{border-right:1px solid #6f6f6f;list-style:none;margin-right:8px;padding-right:8px}.c-article-identifiers__item:last-child{border-right:0;margin-right:0;padding-right:0}.c-article-identifiers__open{color:#cc4b14}.c-article-title{font-size:1.5rem;line-height:1.25;margin-bottom:16px}@media only screen and (min-width:768px){.c-article-title{font-size:1.875rem;line-height:1.2}}.c-article-author-list{display:inline;font-size:1rem;list-style:none;margin:0 8px 0 0;padding:0;width:100%}.c-article-author-list__item{display:inline;padding-right:0}.c-article-author-list svg{margin-left:4px}.c-article-author-list__show-more{display:none;margin-right:4px}.c-article-author-list__button,.js .c-article-author-list__item--hide,.js .c-article-author-list__show-more{display:none}.js .c-article-author-list--long .c-article-author-list__show-more,.js .c-article-author-list--long+.c-article-author-list__button{display:inline}@media only screen and (max-width:539px){.js .c-article-author-list__item--hide-small-screen{display:none}.js .c-article-author-list--short .c-article-author-list__show-more,.js .c-article-author-list--short+.c-article-author-list__button{display:inline}}#uptodate-client,.js .c-article-author-list--expanded .c-article-author-list__show-more{display:none!important}.js .c-article-author-list--expanded .c-article-author-list__item--hide-small-screen{display:inline!important}.c-article-author-list__button{background:#ebf1f5;border:4px solid #ebf1f5;border-radius:20px;color:#666;font-size:.875rem;line-height:1.4;padding:2px 11px 2px 8px;text-decoration:none}.c-article-author-list__button svg{margin:1px 4px 0 0}.c-article-author-list__button:hover{background:#173962;border-color:transparent;color:#fff}.c-article-info-details{font-size:1rem;margin-bottom:8px;margin-top:16px}.c-article-info-details__cite-as{border-left:1px solid #6f6f6f;margin-left:8px;padding-left:8px}.c-article-metrics-bar{display:flex;flex-wrap:wrap;font-size:1rem;line-height:1.3}.c-article-metrics-bar__wrapper{margin:0 0 16px}.c-article-metrics-bar__item{align-items:baseline;border-right:1px solid #6f6f6f;margin-right:8px}.c-article-metrics-bar__item:last-child{border-right:0}.c-article-metrics-bar__count{font-weight:700;margin:0}.c-article-metrics-bar__label{color:#626262;font-style:normal;font-weight:400;margin:0 10px 0 5px}.c-article-metrics-bar__details{margin:0}.c-article-main-column{font-family:Georgia,Palatino,serif;margin-right:8.6%;width:60.2%}@media only screen and (max-width:1023px){.c-article-main-column{margin-right:0;width:100%}}.c-article-extras{float:left;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;width:31.2%}@media only screen and (max-width:1023px){.c-article-extras{display:none}}.c-article-associated-content__container .c-article-associated-content__title,.c-article-section__title{border-bottom:2px solid #d5d5d5;font-size:1.25rem;margin:0;padding-bottom:8px}@media only screen and (min-width:768px){.c-article-associated-content__container .c-article-associated-content__title,.c-article-section__title{font-size:1.5rem;line-height:1.24}}.c-article-section{clear:both}.c-article-section__content{margin-bottom:40px;padding-top:8px}@media only screen and (max-width:1023px){.c-article-section__content{padding-left:0}}.c-article__sub-heading{color:#222;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;font-size:1.125rem;font-style:normal;font-weight:400;line-height:1.3;margin:0 0 8px}@media only screen and (min-width:768px){.c-article__sub-heading{font-size:1.5rem;line-height:1.24}}.c-article-authors-search{margin-bottom:24px;margin-top:0}.c-article-authors-search__item,.c-article-authors-search__title{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif}.c-article-authors-search__title{color:#626262;font-size:1.05rem;font-weight:700;margin:0;padding:0}.c-article-authors-search__item{font-size:1rem}.c-article-authors-search__text{margin:0}.c-article-share-box__no-sharelink-info{font-size:.813rem;font-weight:700;margin-bottom:24px;padding-top:4px}.c-article-share-box__only-read-input{border:1px solid #d5d5d5;display:inline-block;font-size:.875rem;font-weight:700;margin-bottom:8px;padding:8px 10px}.c-article-share-box__button--link-like{background-color:transparent;border:0;color:#0067c5;cursor:pointer;font-size:.875rem;margin-bottom:8px;margin-left:10px}.c-article-associated-content__container .c-article-associated-content__collection-label{font-size:.875rem;line-height:1.4}.c-article-associated-content__container .c-article-associated-content__collection-title{line-height:1.3}.c-context-bar{box-shadow:0 0 10px 0 rgba(51,51,51,.2);position:relative;width:100%}.c-context-bar__title{display:none}.c-reading-companion{clear:both;min-height:389px}.c-reading-companion .u-lazy-ad-wrapper{background-color:transparent!important;min-height:250px}.c-reading-companion__sticky{max-width:389px}.c-reading-companion__scroll-pane{margin:0;min-height:200px;overflow:hidden auto}.c-reading-companion__tabs{display:flex;flex-flow:row nowrap;font-size:1rem;list-style:none;margin:0 0 8px;padding:0}.c-reading-companion__tabs>li{flex-grow:1}.c-reading-companion__tab{background-color:#eee;border:1px solid #d5d5d5;border-image:initial;border-left-width:0;color:#0067c5;font-size:1rem;padding:8px 8px 8px 15px;text-align:left;width:100%}.c-reading-companion__tabs li:first-child .c-reading-companion__tab{border-left-width:1px}.c-reading-companion__tab--active{background-color:#fff;border-bottom:1px solid #fff;color:#222;font-weight:700}.c-reading-companion__sections-list{list-style:none;padding:0}.c-reading-companion__figures-list,.c-reading-companion__references-list{list-style:none;min-height:389px;padding:0}.c-reading-companion__references-list--numeric{list-style:decimal inside}.c-reading-companion__sections-list{margin:0 0 8px;min-height:50px}.c-reading-companion__section-item{font-size:1rem;padding:0}.c-reading-companion__section-item a{display:block;line-height:1.5;overflow:hidden;padding:8px 0 8px 16px;text-overflow:ellipsis;white-space:nowrap}.c-reading-companion__figure-item{border-top:1px solid #d5d5d5;font-size:1rem;padding:16px 8px 16px 0}.c-reading-companion__figure-item:first-child{border-top:none;padding-top:8px}.c-reading-companion__reference-item{border-top:1px solid #d5d5d5;font-size:1rem;padding:8px 8px 8px 16px}.c-reading-companion__reference-item:first-child{border-top:none}.c-reading-companion__reference-item a{word-break:break-word}.c-reading-companion__reference-citation{display:inline}.c-reading-companion__reference-links{font-size:.813rem;font-weight:700;list-style:none;margin:8px 0 0;padding:0;text-align:right}.c-reading-companion__reference-links>a{display:inline-block;padding-left:8px}.c-reading-companion__reference-links>a:first-child{display:inline-block;padding-left:0}.c-reading-companion__figure-title{display:block;margin:0 0 8px}.c-reading-companion__figure-links{display:flex;justify-content:space-between;margin:8px 0 0}.c-reading-companion__figure-links>a{align-items:center;display:flex}.c-reading-companion__figure-full-link svg{height:.8em;margin-left:2px}.c-reading-companion__panel{border-top:none;display:none;margin-top:0;padding-top:0}.c-reading-companion__panel--active{display:block}.c-pdf-download__link .u-icon{padding-top:2px}.c-pdf-download{display:flex;margin-bottom:16px;max-height:48px}@media only screen and (min-width:540px){.c-pdf-download{max-height:none}}@media only screen and (min-width:1024px){.c-pdf-download{max-height:48px}}.c-pdf-download__link{display:flex;flex:1 1 0%;padding:13px 24px!important}.c-pdf-download__text{padding-right:4px}@media only screen and (max-width:539px){.c-pdf-download__text{text-transform:capitalize}}@media only screen and (min-width:540px){.c-pdf-download__text{padding-right:8px}}.c-pdf-container{display:flex;justify-content:flex-end}@media only screen and (max-width:539px){.c-pdf-container .c-pdf-download{display:flex;flex-basis:100%}}.c-article-associated-content__container a,.c-card__summary a{text-decoration:underline}.u-display-none{display:none}.js .u-js-hide,.u-hide{display:none;visibility:hidden}.u-show{display:block;visibility:visible}.u-visually-hidden{clip:rect(0,0,0,0);border:0;height:1px;margin:-100%;overflow:hidden;padding:0;position:absolute!important;width:1px}@media print{.u-hide-print{display:none}}@media only screen and (min-width:1024px){.u-hide-at-lg{display:none;visibility:hidden}}.u-icon{fill:currentcolor;display:inline-block;height:1em;transform:translate(0);vertical-align:text-top;width:1em}.u-list-reset{list-style:none;margin:0;padding:0}.hide{display:none;visibility:hidden}.c-article-associated-content__container .c-article-associated-content__collection.collection~.c-article-associated-content__collection.collection .c-article-associated-content__collection-label,.c-article-associated-content__container .c-article-associated-content__collection.section~.c-article-associated-content__collection.section .c-article-associated-content__collection-label,.c-article-associated-content__container .c-article-associated-content__title{display:none}.c-article-associated-content__container .c-article-associated-content__collection.collection .c-article-associated-content__collection-label,.c-article-associated-content__container .c-article-associated-content__collection.section .c-article-associated-content__collection-label{display:block}.c-article-associated-content__container .c-article-associated-content__collection.collection,.c-article-associated-content__container .c-article-associated-content__collection.section{margin-bottom:5px}.c-article-associated-content__container .c-article-associated-content__collection.section~.c-article-associated-content__collection.collection{margin-top:28px}.c-article-associated-content__container .c-article-associated-content__collection:first-child{margin-top:0}.c-article-associated-content__container .c-article-associated-content__collection:last-child{margin-bottom:2.4rem}.c-article-associated-content__container .c-article-associated-content__collection-label{color:#1b3051}.c-article-associated-content__container .c-article-associated-content__collection-title{font-size:1.063rem;font-weight:400}.webfonts-loaded .c-article__sub-heading{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif} }</style>

    <link rel="stylesheet" data-test="critical-css-handler" data-inline-css-source="critical-css" href="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/enhanced-f6338fbf86.css" media="only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark)" onload="this.media=&#39;only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark)&#39;;this.onload=null">

    <link rel="stylesheet" data-test="critical-css-handler" data-inline-css-source="critical-css" href="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/enhanced-article-1b20241a7b.css" media="only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark)" onload="this.media=&#39;only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark)&#39;;this.onload=null">

    



        

        
    <script type="text/javascript">
        config = {
            env: 'live',
            site: 'genomebiology.biomedcentral.com',
            siteWithPath: 'genomebiology.biomedcentral.com' + window.location.pathname,
            twitterHashtag: '',
            cmsPrefix: 'https://studio-cms.springernature.com/studio/',
            
            doi: '10.1186/gb-2009-10-3-r25',
            figshareScriptUrl: 'https://widgets.figshare.com/static/figshare.js',
            hasFigshareInvoked: false,
            publisherBrand: 'BioMed Central',
            mustardcut: false
        };
    </script>

        
    <script type="text/javascript" data-test="dataLayer">
        window.dataLayer = [{"content":{"article":{"doi":"10.1186/gb-2009-10-3-r25","articleType":"Software","peerReviewType":"Closed","supplement":null,"keywords":"Additional Data File;Central Processing Unit;Human Reference Genome;Memory Footprint;NCBI Short Read Archive"},"contentInfo":{"imprint":"BioMed Central","title":"Ultrafast and memory-efficient alignment of short DNA sequences to the human genome","publishedAt":1236124800000,"publishedAtDate":"2009-03-04","author":["Ben Langmead","Cole Trapnell","Mihai Pop","Steven L Salzberg"],"collection":[]},"attributes":{"deliveryPlatform":"oscar","template":"rebrand","cms":null,"copyright":{"creativeCommonsType":null,"openAccess":false},"environment":"live"},"journal":{"siteKey":"genomebiology.biomedcentral.com","volume":"10","issue":"3","title":"Genome Biology","type":"FLAGSHIP","journalID":13059,"section":[]},"category":{"pmc":{"primarySubject":"Life Sciences"},"contentType":"Software","publishingSegment":"BMC Flagships"}},"session":{"authentication":{"authenticationID":[]}},"version":"1.0.0","page":{"category":{"pageType":"article"},"attributes":{"featureFlags":[],"environment":"live"}},"japan":false,"event":"dataLayerCreated"}];
    </script>

        

        
        
            <script>
                window.dataLayer.push({
                    ga4MeasurementId: 'G-PJCTJWPV25',
                    ga360TrackingId: 'UA-54492316-9',
                    twitterId: 'o47a2',
                    ga4ServerUrl: 'https://collect.biomedcentral.com',
                    imprint: 'biomedcentral'
                });
            </script>
        
        

        
    <script>
        (function(w, d) {
            w.config = w.config || {};
            w.config.mustardcut = false;

            
            if (w.matchMedia && w.matchMedia('only print, only all and (prefers-color-scheme: no-preference), only all and (prefers-color-scheme: light), only all and (prefers-color-scheme: dark)').matches) {
                w.config.mustardcut = true;
                d.classList.add('js');
                d.classList.remove('grade-c');
            }
        })(window, document.documentElement);
    </script>




        
<script>
    (function () {
        if ( typeof window.CustomEvent === "function" ) return false;
        function CustomEvent ( event, params ) {
            params = params || { bubbles: false, cancelable: false, detail: null };
            var evt = document.createEvent( 'CustomEvent' );
            evt.initCustomEvent( event, params.bubbles, params.cancelable, params.detail );
            return evt;
        }

        CustomEvent.prototype = window.Event.prototype;

        window.CustomEvent = CustomEvent;
    })();
</script>

        
    <script class="js-entry">
        if (window.config.mustardcut) {
            (function(w, d) {
                
                
                
                    window.Component = {};
                    window.suppressShareButton = true;
                

                var currentScript = d.currentScript || d.head.querySelector('script.js-entry');

                
                function catchNoModuleSupport() {
                    var scriptEl = d.createElement('script');
                    return (!('noModule' in scriptEl) && 'onbeforeload' in scriptEl)
                }

                var headScripts = [
                    {'src': '/static/js/polyfill-es5-bundle-6881c82d6d.js', 'async': false},
                    {'src': '/static/js/airbrake-es5-bundle-572454ea6e.js', 'async': false},
                ];

                var bodyScripts = [
                    {'src': '/static/js/app-es5-bundle-6a48e032fc.js', 'async': false, 'module': false},
                    {'src': '/static/js/app-es6-bundle-716d0ac638.js', 'async': false, 'module': true}
                    
                    
                        , {'src': '/static/js/global-article-es5-bundle-c4f64c8770.js', 'async': false, 'module': false},
                        {'src': '/static/js/global-article-es6-bundle-93d9e3b3d0.js', 'async': false, 'module': true}
                    
                ];

                function createScript(script) {
                    var scriptEl = d.createElement('script');
                    scriptEl.src = script.src;
                    scriptEl.async = script.async;
                    if (script.module === true) {
                        scriptEl.type = "module";
                        if (catchNoModuleSupport()) {
                            scriptEl.src = '';
                        }
                    } else if (script.module === false) {
                        scriptEl.setAttribute('nomodule', true)
                    }
                    if (script.charset) {
                        scriptEl.setAttribute('charset', script.charset);
                    }

                    return scriptEl;
                }

                for (var i = 0; i < headScripts.length; ++i) {
                    var scriptEl = createScript(headScripts[i]);
                    currentScript.parentNode.insertBefore(scriptEl, currentScript.nextSibling);
                }

                d.addEventListener('DOMContentLoaded', function() {
                    for (var i = 0; i < bodyScripts.length; ++i) {
                        var scriptEl = createScript(bodyScripts[i]);
                        d.body.appendChild(scriptEl);
                    }
                });

                // Webfont repeat view
                var config = w.config;
                if (config && config.publisherBrand && sessionStorage.fontsLoaded === 'true') {
                    d.documentElement.className += ' webfonts-loaded';
                }
            })(window, document);
        }
    </script><script src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/airbrake-es5-bundle-572454ea6e.js.下载"></script><script src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/polyfill-es5-bundle-6881c82d6d.js.下载"></script>




        
            
            
                <script data-test="gtm-head">
    window.initGTM = function() {
        if (window.config.mustardcut) {
            (function (w, d, s, l, i) {
                w[l] = w[l] || [];
                w[l].push({'gtm.start': new Date().getTime(), event: 'gtm.js'});
                var f = d.getElementsByTagName(s)[0],
                        j = d.createElement(s),
                        dl = l != 'dataLayer' ? '&l=' + l : '';
                j.async = true;
                j.src = 'https://collect.biomedcentral.com/gtm.js?id=' + i + dl;
                f.parentNode.insertBefore(j, f);
            })(window, document, 'script', 'dataLayer', 'GTM-MRVXSHQ');
        }
    }
</script>
            
            
        

        <script>
    (function (w, d, t) {
        function cc() {
            var h = w.location.hostname;
            var e = d.createElement(t),
            s = d.getElementsByTagName(t)[0];

            if (h.indexOf('springer.com') > -1) {
                e.src = 'https://push-content.springernature.io/pcf_sb_5_1617714720898560639/production_live/consent-bundle-17-15.js';
                e.setAttribute('onload', "initGTM(window,document,'script','dataLayer','GTM-MRVXSHQ')");
            } else if (h.indexOf('biomedcentral.com') > -1) {
                e.src = 'https://push-content.springernature.io/pcf_sb_5_1617714720898560639/production_live/consent-bundle-15-12.js';
                e.setAttribute('onload', "initGTM(window,document,'script','dataLayer','GTM-MRVXSHQ')");
            } else if (h.indexOf('springeropen.com') > -1) {
                e.src = 'https://push-content.springernature.io/pcf_sb_5_1617714720898560639/production_live/consent-bundle-16-12.js';
                e.setAttribute('onload', "initGTM(window,document,'script','dataLayer','GTM-MRVXSHQ')");
            } else if (h.indexOf('springernature.com') > -1) {
                e.src = 'https://cmp.springernature.com/production_live/consent-bundle-49-16.js';
                e.setAttribute('onload', "initGTM(window,document,'script','dataLayer','GTM-NK22KLS')");
            } else {
                e.src = '/static/js/cookie-consent-es5-bundle-2b0f06c1e4.js';
                e.setAttribute('data-consent', h);
            }
            s.insertAdjacentElement('afterend', e);
        }

        cc();
    })(window, document, 'script');
</script>


        
    
        
    


        
    
    <link rel="canonical" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25">
    

        
        
        
    
    <meta property="og:url" content="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25">
    <meta property="og:type" content="article">
    <meta property="og:site_name" content="BioMed Central">
    <meta property="og:title" content="Ultrafast and memory-efficient alignment of short DNA sequences to the human genome - Genome Biology">
    <meta property="og:description" content="Bowtie is an ultrafast, memory-efficient alignment program for aligning short DNA sequence reads to large genomes. For the human genome, Burrows-Wheeler indexing allows Bowtie to align more than 25 million reads per CPU hour with a memory footprint of approximately 1.3 gigabytes. Bowtie extends previous Burrows-Wheeler techniques with a novel quality-aware backtracking algorithm that permits mismatches. Multiple processor cores can be used simultaneously to achieve even greater alignment speeds. Bowtie is open source http://bowtie.cbcb.umd.edu .">
    <meta property="og:image" content="https://media.springernature.com/w200/springer-static/cover/journal/13059.jpg">
    

        

        
    <script type="application/ld+json">{"mainEntity":{"headline":"Ultrafast and memory-efficient alignment of short DNA sequences to the human genome","description":"Bowtie is an ultrafast, memory-efficient alignment program for aligning short DNA sequence reads to large genomes. For the human genome, Burrows-Wheeler indexing allows Bowtie to align more than 25 million reads per CPU hour with a memory footprint of approximately 1.3 gigabytes. Bowtie extends previous Burrows-Wheeler techniques with a novel quality-aware backtracking algorithm that permits mismatches. Multiple processor cores can be used simultaneously to achieve even greater alignment speeds. Bowtie is open source \n                  http://bowtie.cbcb.umd.edu\n                  \n                .","datePublished":"2009-03-04","dateModified":"2009-03-04","pageStart":"1","pageEnd":"10","sameAs":"https://doi.org/10.1186/gb-2009-10-3-r25","keywords":"Animal Genetics and Genomics,Human Genetics,Plant Genetics and Genomics,Microbial Genetics and Genomics,Bioinformatics,Evolutionary Biology","image":"https://static-content.springer.com/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig1_HTML.jpg","isPartOf":{"name":"Genome Biology","issn":["1474-760X"],"volumeNumber":"10","@type":["Periodical","PublicationVolume"]},"publisher":{"name":"BioMed Central","logo":{"url":"https://www.springernature.com/app-sn/public/images/logo-springernature.png","@type":"ImageObject"},"@type":"Organization"},"author":[{"name":"Langmead, Ben","affiliation":[{"name":"Institute for Advanced Computer Studies, University of Maryland","address":{"name":"Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, USA","@type":"PostalAddress"},"@type":"Organization"}],"email":"langmead@cs.umd.edu","@type":"Person"},{"name":"Trapnell, Cole","affiliation":[{"name":"Institute for Advanced Computer Studies, University of Maryland","address":{"name":"Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, USA","@type":"PostalAddress"},"@type":"Organization"}],"@type":"Person"},{"name":"Pop, Mihai","affiliation":[{"name":"Institute for Advanced Computer Studies, University of Maryland","address":{"name":"Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, USA","@type":"PostalAddress"},"@type":"Organization"}],"@type":"Person"},{"name":"Salzberg, Steven L","affiliation":[{"name":"Institute for Advanced Computer Studies, University of Maryland","address":{"name":"Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, USA","@type":"PostalAddress"},"@type":"Organization"}],"@type":"Person"}],"isAccessibleForFree":true,"@type":"ScholarlyArticle"},"@context":"https://schema.org","@type":"WebPage"}</script>

        

        
    <style type="text/css">.MathJax_Hover_Frame {border-radius: .25em; -webkit-border-radius: .25em; -moz-border-radius: .25em; -khtml-border-radius: .25em; box-shadow: 0px 0px 15px #83A; -webkit-box-shadow: 0px 0px 15px #83A; -moz-box-shadow: 0px 0px 15px #83A; -khtml-box-shadow: 0px 0px 15px #83A; border: 1px solid #A6D ! important; display: inline-block; position: absolute}
.MathJax_Menu_Button .MathJax_Hover_Arrow {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 4px; -webkit-border-radius: 4px; -moz-border-radius: 4px; -khtml-border-radius: 4px; font-family: 'Courier New',Courier; font-size: 9px; color: #F0F0F0}
.MathJax_Menu_Button .MathJax_Hover_Arrow span {display: block; background-color: #AAA; border: 1px solid; border-radius: 3px; line-height: 0; padding: 4px}
.MathJax_Hover_Arrow:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_Hover_Arrow:hover span {background-color: #CCC!important}
</style><style type="text/css">#MathJax_About {position: fixed; left: 50%; width: auto; text-align: center; border: 3px outset; padding: 1em 2em; background-color: #DDDDDD; color: black; cursor: default; font-family: message-box; font-size: 120%; font-style: normal; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 15px; -webkit-border-radius: 15px; -moz-border-radius: 15px; -khtml-border-radius: 15px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_About.MathJax_MousePost {outline: none}
.MathJax_Menu {position: absolute; background-color: white; color: black; width: auto; padding: 2px; border: 1px solid #CCCCCC; margin: 0; cursor: default; font: menu; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
.MathJax_MenuItem {padding: 2px 2em; background: transparent}
.MathJax_MenuArrow {position: absolute; right: .5em; padding-top: .25em; color: #666666; font-size: .75em}
.MathJax_MenuActive .MathJax_MenuArrow {color: white}
.MathJax_MenuArrow.RTL {left: .5em; right: auto}
.MathJax_MenuCheck {position: absolute; left: .7em}
.MathJax_MenuCheck.RTL {right: .7em; left: auto}
.MathJax_MenuRadioCheck {position: absolute; left: 1em}
.MathJax_MenuRadioCheck.RTL {right: 1em; left: auto}
.MathJax_MenuLabel {padding: 2px 2em 4px 1.33em; font-style: italic}
.MathJax_MenuRule {border-top: 1px solid #CCCCCC; margin: 4px 1px 0px}
.MathJax_MenuDisabled {color: GrayText}
.MathJax_MenuActive {background-color: Highlight; color: HighlightText}
.MathJax_MenuDisabled:focus, .MathJax_MenuLabel:focus {background-color: #E8E8E8}
.MathJax_ContextMenu:focus {outline: none}
.MathJax_ContextMenu .MathJax_MenuItem:focus {outline: none}
#MathJax_AboutClose {top: .2em; right: .2em}
.MathJax_Menu .MathJax_MenuClose {top: -10px; left: -10px}
.MathJax_MenuClose {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; font-family: 'Courier New',Courier; font-size: 24px; color: #F0F0F0}
.MathJax_MenuClose span {display: block; background-color: #AAA; border: 1.5px solid; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; line-height: 0; padding: 8px 0 6px}
.MathJax_MenuClose:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_MenuClose:hover span {background-color: #CCC!important}
.MathJax_MenuClose:hover:focus {outline: none}
</style><style type="text/css">.MJX_Assistive_MathML {position: absolute!important; top: 0; left: 0; clip: rect(1px, 1px, 1px, 1px); padding: 1px 0 0 0!important; border: 0!important; height: 1px!important; width: 1px!important; overflow: hidden!important; display: block!important; -webkit-touch-callout: none; -webkit-user-select: none; -khtml-user-select: none; -moz-user-select: none; -ms-user-select: none; user-select: none}
.MJX_Assistive_MathML.MJX_Assistive_MathML_Block {width: 100%!important}
</style><style type="text/css">#MathJax_Zoom {position: absolute; background-color: #F0F0F0; overflow: auto; display: block; z-index: 301; padding: .5em; border: 1px solid black; margin: 0; font-weight: normal; font-style: normal; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; -webkit-box-sizing: content-box; -moz-box-sizing: content-box; box-sizing: content-box; box-shadow: 5px 5px 15px #AAAAAA; -webkit-box-shadow: 5px 5px 15px #AAAAAA; -moz-box-shadow: 5px 5px 15px #AAAAAA; -khtml-box-shadow: 5px 5px 15px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_ZoomOverlay {position: absolute; left: 0; top: 0; z-index: 300; display: inline-block; width: 100%; height: 100%; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
#MathJax_ZoomFrame {position: relative; display: inline-block; height: 0; width: 0}
#MathJax_ZoomEventTrap {position: absolute; left: 0; top: 0; z-index: 302; display: inline-block; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
</style><style type="text/css">.MathJax_Preview {color: #888}
#MathJax_Message {position: fixed; left: 1em; bottom: 1.5em; background-color: #E6E6E6; border: 1px solid #959595; margin: 0px; padding: 2px 8px; z-index: 102; color: black; font-size: 80%; width: auto; white-space: nowrap}
#MathJax_MSIE_Frame {position: absolute; top: 0; left: 0; width: 0px; z-index: 101; border: 0px; margin: 0px; padding: 0px}
.MathJax_Error {color: #CC0000; font-style: italic}
</style><style type="text/css">.MJXp-script {font-size: .8em}
.MJXp-right {-webkit-transform-origin: right; -moz-transform-origin: right; -ms-transform-origin: right; -o-transform-origin: right; transform-origin: right}
.MJXp-bold {font-weight: bold}
.MJXp-italic {font-style: italic}
.MJXp-scr {font-family: MathJax_Script,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-frak {font-family: MathJax_Fraktur,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-sf {font-family: MathJax_SansSerif,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-cal {font-family: MathJax_Caligraphic,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-mono {font-family: MathJax_Typewriter,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-largeop {font-size: 150%}
.MJXp-largeop.MJXp-int {vertical-align: -.2em}
.MJXp-math {display: inline-block; line-height: 1.2; text-indent: 0; font-family: 'Times New Roman',Times,STIXGeneral,serif; white-space: nowrap; border-collapse: collapse}
.MJXp-display {display: block; text-align: center; margin: 1em 0}
.MJXp-math span {display: inline-block}
.MJXp-box {display: block!important; text-align: center}
.MJXp-box:after {content: " "}
.MJXp-rule {display: block!important; margin-top: .1em}
.MJXp-char {display: block!important}
.MJXp-mo {margin: 0 .15em}
.MJXp-mfrac {margin: 0 .125em; vertical-align: .25em}
.MJXp-denom {display: inline-table!important; width: 100%}
.MJXp-denom > * {display: table-row!important}
.MJXp-surd {vertical-align: top}
.MJXp-surd > * {display: block!important}
.MJXp-script-box > *  {display: table!important; height: 50%}
.MJXp-script-box > * > * {display: table-cell!important; vertical-align: top}
.MJXp-script-box > *:last-child > * {vertical-align: bottom}
.MJXp-script-box > * > * > * {display: block!important}
.MJXp-mphantom {visibility: hidden}
.MJXp-munderover, .MJXp-munder {display: inline-table!important}
.MJXp-over {display: inline-block!important; text-align: center}
.MJXp-over > * {display: block!important}
.MJXp-munderover > *, .MJXp-munder > * {display: table-row!important}
.MJXp-mtable {vertical-align: .25em; margin: 0 .125em}
.MJXp-mtable > * {display: inline-table!important; vertical-align: middle}
.MJXp-mtr {display: table-row!important}
.MJXp-mtd {display: table-cell!important; text-align: center; padding: .5em 0 0 .5em}
.MJXp-mtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-mlabeledtr {display: table-row!important}
.MJXp-mlabeledtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mlabeledtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-merror {background-color: #FFFF88; color: #CC0000; border: 1px solid #CC0000; padding: 1px 3px; font-style: normal; font-size: 90%}
.MJXp-scale0 {-webkit-transform: scaleX(.0); -moz-transform: scaleX(.0); -ms-transform: scaleX(.0); -o-transform: scaleX(.0); transform: scaleX(.0)}
.MJXp-scale1 {-webkit-transform: scaleX(.1); -moz-transform: scaleX(.1); -ms-transform: scaleX(.1); -o-transform: scaleX(.1); transform: scaleX(.1)}
.MJXp-scale2 {-webkit-transform: scaleX(.2); -moz-transform: scaleX(.2); -ms-transform: scaleX(.2); -o-transform: scaleX(.2); transform: scaleX(.2)}
.MJXp-scale3 {-webkit-transform: scaleX(.3); -moz-transform: scaleX(.3); -ms-transform: scaleX(.3); -o-transform: scaleX(.3); transform: scaleX(.3)}
.MJXp-scale4 {-webkit-transform: scaleX(.4); -moz-transform: scaleX(.4); -ms-transform: scaleX(.4); -o-transform: scaleX(.4); transform: scaleX(.4)}
.MJXp-scale5 {-webkit-transform: scaleX(.5); -moz-transform: scaleX(.5); -ms-transform: scaleX(.5); -o-transform: scaleX(.5); transform: scaleX(.5)}
.MJXp-scale6 {-webkit-transform: scaleX(.6); -moz-transform: scaleX(.6); -ms-transform: scaleX(.6); -o-transform: scaleX(.6); transform: scaleX(.6)}
.MJXp-scale7 {-webkit-transform: scaleX(.7); -moz-transform: scaleX(.7); -ms-transform: scaleX(.7); -o-transform: scaleX(.7); transform: scaleX(.7)}
.MJXp-scale8 {-webkit-transform: scaleX(.8); -moz-transform: scaleX(.8); -ms-transform: scaleX(.8); -o-transform: scaleX(.8); transform: scaleX(.8)}
.MJXp-scale9 {-webkit-transform: scaleX(.9); -moz-transform: scaleX(.9); -ms-transform: scaleX(.9); -o-transform: scaleX(.9); transform: scaleX(.9)}
.MathJax_PHTML .noError {vertical-align: ; font-size: 90%; text-align: left; color: black; padding: 1px 3px; border: 1px solid}
</style><style type="text/css">.cc-banner{-webkit-box-sizing:border-box;box-sizing:border-box;background-color:#01324b;color:#fff;position:fixed;bottom:0;left:0;right:0;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif!important;line-height:1.5;z-index:99999}.cc-banner *{color:inherit}.cc-banner:focus{outline:none}.cc-banner a{color:#fff;text-decoration:underline}.cc-banner a:focus{outline:none;-webkit-box-shadow:0 0 0 3px #fece3e;box-shadow:0 0 0 3px #fece3e}.cc-banner a:active,.cc-banner a:focus,.cc-banner a:hover{text-decoration:none;color:inherit}.cc-banner__content{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;margin:0 auto;max-width:1320px;max-height:90vh;padding:12px}@media (min-width:680px){.cc-banner__content{padding:16px}}@media (min-width:980px){.cc-banner__content{padding-top:60px;padding-bottom:60px}}.cc-banner__title{font-size:18px;margin:0 0 12px}@media (min-width:980px){.cc-banner__title{margin:0 0 16px;font-size:28px}}.cc-banner__body{-webkit-box-flex:1;-webkit-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;overflow-x:hidden;overflow-y:auto;padding:0 0 16px}@media (min-width:680px){.cc-banner__body{padding:0;margin:0 0 16px}}.cc-banner__copy{font-size:14px;margin:0}@media (min-width:980px){.cc-banner__copy{font-size:16px}}.cc-banner__footer{padding:12px;margin:0 -12px -12px;background-color:#012132;border-top:1px solid rgba(0,0,0,.3);-webkit-box-shadow:0 0 5px 0 rgba(0,0,0,.8);box-shadow:0 0 5px 0 rgba(0,0,0,.8);position:relative}@media (min-width:380px){.cc-banner__footer{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-box-align:center;-webkit-align-items:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between}}@media (min-width:680px){.cc-banner__footer{margin:0;padding:0;background-color:transparent;border-top:0;-webkit-box-shadow:none;box-shadow:none;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start}}.cc-banner__button{width:100%}@media (min-width:380px){.cc-banner__button{width:auto;-webkit-box-flex:1;-webkit-flex:1;-ms-flex:1;flex:1}}@media (min-width:680px){.cc-banner__button{max-width:250px}}.cc-banner__button:not(:last-child){margin-bottom:4px}@media (min-width:680px){.cc-banner__button:not(:last-child){margin-right:8px;margin-bottom:0}}@media (min-width:980px){.cc-banner__button:not(:last-child){margin-right:16px}}@media (min-width:380px){.cc-banner__button-accept,.cc-banner__button-reject{-webkit-box-flex:1;-webkit-flex:1 0 49%;-ms-flex:1 0 49%;flex:1 0 49%;max-width:49%}}@media (min-width:680px){.cc-banner__button-accept,.cc-banner__button-reject{max-width:240px}}.cc-banner__button-preferences{padding-left:0;padding-right:0}@media (min-width:380px){.cc-banner__button-preferences{-webkit-box-flex:0;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;margin:auto}}@media (min-width:680px){.cc-banner__button-preferences{margin:0}}@media (min-width:380px) and (max-width:680px){.cc-banner__button-break{display:block}}.cc-button{font-size:13px;font-weight:700;padding:.5em 1em;color:#fff;border:1px solid #183642;background-color:#000;border-radius:3px;cursor:pointer;line-height:1.2}.cc-button:focus{outline:none;-webkit-box-shadow:0 0 0 3px #fece3e;box-shadow:0 0 0 3px #fece3e}.cc-button:hover{-webkit-box-shadow:0 -2px 4px 0 rgba(60,64,67,.15),0 2px 4px 0 rgba(60,64,67,.15);box-shadow:0 -2px 4px 0 rgba(60,64,67,.15),0 2px 4px 0 rgba(60,64,67,.15);text-decoration:underline}@media (min-width:680px){.cc-button{padding:.75em 1em;font-size:14px}}.cc-button--contrast{color:#000;border-color:#fff;background-color:#fff}.cc-button--outline{background-color:transparent;color:#000}.cc-button--contrast.cc-button--outline{color:#fff}.cc-button--link{color:inherit;text-decoration:underline;background-color:transparent;border-color:transparent}.cc-button--link:hover{text-decoration:none;-webkit-box-shadow:none;box-shadow:none}@media (max-width:380px){.cc-button__truncated{border:0;clip:rect(0,0,0,0);height:1px;margin:-100%;overflow:hidden;padding:0;position:absolute!important;width:1px}}.cc-radio{position:relative}.cc-radio *{cursor:pointer}.cc-radio__input{width:22px;height:22px;position:absolute;top:0;left:0}.cc-radio__input:focus{outline:none}.cc-radio__label{padding-left:30px;font-size:14px;line-height:22px;margin:0;color:inherit}.cc-radio__label:after,.cc-radio__label:before{position:absolute;content:"";display:block;background-color:#fff}.cc-radio__label:before{width:22px;height:22px;top:0;left:0;border:2px solid;border-radius:50%}.cc-radio__label:after{top:6px;left:6px;width:0;height:0;border:5px solid;border-radius:50%;opacity:0}.cc-radio__input:focus+.cc-radio__label:before{outline:none;-webkit-box-shadow:0 0 0 3px #fece3e;box-shadow:0 0 0 3px #fece3e}.cc-radio__label--hidden{display:none}.cc-radio__input:checked+.cc-radio__label:after{opacity:1}.cc-radio__input:disabled{cursor:default}.cc-radio__input:disabled+.cc-radio__label{opacity:.5;cursor:default}.cc-preferences{-webkit-box-sizing:border-box;box-sizing:border-box;color:#111;font-family:sans-serif;overflow:auto;z-index:100000;position:fixed;background-color:rgba(5,10,20,.95);line-height:1.4;top:0;left:0;right:0;bottom:0}.cc-preferences:focus{outline:none}.cc-preferences *,.cc-preferences :after,.cc-preferences :before{-webkit-box-sizing:inherit!important;box-sizing:inherit!important}.cc-preferences h1,.cc-preferences h2,.cc-preferences h3,.cc-preferences h4,.cc-preferences h5,.cc-preferences h6{font-family:sans-serif;font-style:normal}.cc-preferences a{color:inherit;text-decoration:underline}.cc-preferences a:hover{text-decoration:none;color:inherit}.cc-preferences input{margin:0}.cc-preferences__dialog{position:relative;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;padding:16px;margin:auto;max-width:960px;max-height:100vh}.cc-preferences__dialog>:last-child{border-bottom-left-radius:5px;border-bottom-right-radius:5px}.cc-preferences__close{position:absolute;border-radius:3px;top:24px;right:24px;padding:0;width:40px;height:40px;border:0;font-size:40px;line-height:1;cursor:pointer;background:transparent;font-family:Times New Roman,serif}.cc-preferences__close:focus{outline:none;-webkit-box-shadow:0 0 0 3px #fece3e;box-shadow:0 0 0 3px #fece3e}.cc-preferences__close-label{border:0;clip:rect(0,0,0,0);height:1px;margin:-100%;overflow:hidden;padding:0;position:absolute!important;width:1px}.cc-preferences__header{background:#fff;text-align:center;padding:16px;border-bottom:1px solid #d0d0d0;border-top-left-radius:5px;border-top-right-radius:5px}.cc-preferences__title{font-family:sans-serif;font-size:18px;font-weight:700;padding-right:24px;margin:0;color:#111}@media (min-width:480px){.cc-preferences__title{padding-right:0}}@media (min-width:980px){.cc-preferences__title{font-size:22px}}.cc-preferences__body{padding:16px;-webkit-box-flex:1;-webkit-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;min-height:200px;overflow-x:hidden;overflow-y:auto;background:#fff}.cc-preferences__footer{background:#fff;margin-bottom:0;padding:16px;border-top:1px solid #d0d0d0;-webkit-box-shadow:0 0 5px 0 rgba(0,0,0,.2);box-shadow:0 0 5px 0 rgba(0,0,0,.2);position:relative}@media (min-width:480px){.cc-preferences__footer{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-align:center;-webkit-align-items:center;-ms-flex-align:center;align-items:center}}.cc-preferences__footer>.cc-button{display:block;width:100%}@media (min-width:480px){.cc-preferences__footer>.cc-button{-webkit-box-flex:1;-webkit-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;width:auto;max-width:250px}}@media (min-width:980px){.cc-preferences__footer>.cc-button{-webkit-flex-basis:250px;-ms-flex-preferred-size:250px;flex-basis:250px}}@media (min-width:480px){.cc-preferences__footer>.cc-button:not(:first-child){margin-left:16px}}.cc-preferences__footer>.cc-button:not(:last-child){margin-bottom:8px}@media (min-width:480px){.cc-preferences__footer>.cc-button:not(:last-child){margin-bottom:0}}@media (min-width:980px){.cc-preferences__footer>.cc-button:last-child{margin-left:auto}}.cc-preferences__categories{list-style:none;padding:0;margin:0}.cc-preferences__category:not(:last-child){margin-bottom:16px;padding-bottom:16px;border-bottom:1px solid #ececec}.cc-preferences__category input{margin-right:10px}.cc-preferences__category-header{margin:0 0 8px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-align:baseline;-webkit-align-items:baseline;-ms-flex-align:baseline;align-items:baseline}.cc-preferences__category-heading{margin:0;font-size:16px;font-weight:700;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-align:baseline;-webkit-align-items:baseline;-ms-flex-align:baseline;align-items:baseline}.cc-preferences__category-description{margin:0 0 8px;font-size:14px}.cc-preferences__category-footer{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-align:baseline;-webkit-align-items:baseline;-ms-flex-align:baseline;align-items:baseline}.cc-preferences__status{border:0;clip:rect(0,0,0,0);height:1px;margin:-100%;overflow:hidden;padding:0;position:absolute!important;width:1px}.cc-preferences__controls{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin:0}.cc-preferences__controls>:not(:last-child){margin-right:16px}.cc-preferences__always-on{font-size:14px;position:relative;padding-left:26px}.cc-preferences__always-on:before{content:"";display:block;position:absolute;left:0;top:2px;width:18px;height:10px;-webkit-transform:rotate(-45deg);transform:rotate(-45deg);border:solid;border-width:0 0 4px 4px;border-top-color:transparent;background:transparent}.cc-preferences__details-button{position:relative;font-size:14px;padding:2px 4px;font-weight:400;border:1px solid transparent;background-color:#fff;color:#000}.cc-preferences__details-button--show{background-color:#000;border-color:#000;color:#fff}.cc-preferences__details-button--show:before{content:"";display:block;position:absolute;right:50%;top:100%;width:0;height:0;margin-right:-8px;border-left:8px solid transparent;border-right:8px solid transparent;border-top:8px solid #111}.cc-preferences__details{font-size:14px;display:none;padding:12px 0 0;border-top:2px solid #000;margin:12px 0 0}.cc-preferences__details--show{display:block}.cc-preferences__cookie-list,.cc-preferences__provider-list{list-style:none;margin:0;padding:0}.cc-preferences__provider-list{-webkit-columns:170px;-moz-columns:170px;columns:170px}.cc-preferences__cookie-title{font-size:1em;margin:0}.cc-preferences__cookie-description{font-size:1em;margin:0 0 8px}.cc-preferences__cookie-domain{color:#666;padding-left:4px;margin-left:8px;border-left:1px solid #999}body.cc-has-preferences-open{overflow:hidden;position:relative}</style></head>

    <body class="journal journal-fulltext"><div id="MathJax_Message" style="display: none;"></div>
        
    
        
            <!-- Google Tag Manager (noscript) -->
            <noscript>
                <iframe src="https://www.googletagmanager.com/ns.html?id=GTM-MRVXSHQ"
                        height="0" width="0" style="display:none;visibility:hidden"></iframe>
            </noscript>
            <!-- End Google Tag Manager (noscript) -->
        
    

        <div class="u-visually-hidden" aria-hidden="true">
    
    <!--?xml version="1.0" encoding="UTF-8"?--><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><defs><rect id="SVGID_1_" x="0" y="0.74" width="56.72" height="55.24"></rect></defs><symbol id="icon-access" viewBox="0 0 18 18"><path d="m14 8c.5522847 0 1 .44771525 1 1v7h2.5c.2761424 0 .5.2238576.5.5v1.5h-18v-1.5c0-.2761424.22385763-.5.5-.5h2.5v-7c0-.55228475.44771525-1 1-1s1 .44771525 1 1v6.9996556h8v-6.9996556c0-.55228475.4477153-1 1-1zm-8 0 2 1v5l-2 1zm6 0v7l-2-1v-5zm-2.42653766-7.59857636 7.03554716 4.92488299c.4162533.29137735.5174853.86502537.226108 1.28127873-.1721584.24594054-.4534847.39241464-.7536934.39241464h-14.16284822c-.50810197 0-.92-.41189803-.92-.92 0-.30020869.1464741-.58153499.39241464-.75369337l7.03554714-4.92488299c.34432015-.2410241.80260453-.2410241 1.14692468 0zm-.57346234 2.03988748-3.65526982 2.55868888h7.31053962z" fill-rule="evenodd"></path></symbol><symbol id="icon-account" viewBox="0 0 18 18"><path d="m10.2379028 16.9048051c1.3083556-.2032362 2.5118471-.7235183 3.5294683-1.4798399-.8731327-2.5141501-2.0638925-3.935978-3.7673711-4.3188248v-1.27684611c1.1651924-.41183641 2-1.52307546 2-2.82929429 0-1.65685425-1.3431458-3-3-3-1.65685425 0-3 1.34314575-3 3 0 1.30621883.83480763 2.41745788 2 2.82929429v1.27684611c-1.70347856.3828468-2.89423845 1.8046747-3.76737114 4.3188248 1.01762123.7563216 2.22111275 1.2766037 3.52946833 1.4798399.40563808.0629726.81921174.0951949 1.23790281.0951949s.83226473-.0322223 1.2379028-.0951949zm4.3421782-2.1721994c1.4927655-1.4532925 2.419919-3.484675 2.419919-5.7326057 0-4.418278-3.581722-8-8-8s-8 3.581722-8 8c0 2.2479307.92715352 4.2793132 2.41991895 5.7326057.75688473-2.0164459 1.83949951-3.6071894 3.48926591-4.3218837-1.14534283-.70360829-1.90918486-1.96796271-1.90918486-3.410722 0-2.209139 1.790861-4 4-4s4 1.790861 4 4c0 1.44275929-.763842 2.70711371-1.9091849 3.410722 1.6497664.7146943 2.7323812 2.3054378 3.4892659 4.3218837zm-5.580081 3.2673943c-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9 4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9z" fill-rule="evenodd"></path></symbol><symbol id="icon-alert" viewBox="0 0 18 18"><path d="m4 10h2.5c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-3.08578644l-1.12132034 1.1213203c-.18753638.1875364-.29289322.4418903-.29289322.7071068v.1715729h14v-.1715729c0-.2652165-.1053568-.5195704-.2928932-.7071068l-1.7071068-1.7071067v-3.4142136c0-2.76142375-2.2385763-5-5-5-2.76142375 0-5 2.23857625-5 5zm3 4c0 1.1045695.8954305 2 2 2s2-.8954305 2-2zm-5 0c-.55228475 0-1-.4477153-1-1v-.1715729c0-.530433.21071368-1.0391408.58578644-1.4142135l1.41421356-1.4142136v-3c0-3.3137085 2.6862915-6 6-6s6 2.6862915 6 6v3l1.4142136 1.4142136c.3750727.3750727.5857864.8837805.5857864 1.4142135v.1715729c0 .5522847-.4477153 1-1 1h-4c0 1.6568542-1.3431458 3-3 3-1.65685425 0-3-1.3431458-3-3z" fill-rule="evenodd"></path></symbol><symbol id="icon-arrow-broad" viewBox="0 0 16 16"><path d="m6.10307866 2.97190702v7.69043288l2.44965196-2.44676915c.38776071-.38730439 1.0088052-.39493524 1.38498697-.01919617.38609051.38563612.38643641 1.01053024-.00013864 1.39665039l-4.12239817 4.11754683c-.38616704.3857126-1.01187344.3861062-1.39846576-.0000311l-4.12258206-4.11773056c-.38618426-.38572979-.39254614-1.00476697-.01636437-1.38050605.38609047-.38563611 1.01018509-.38751562 1.4012233.00306241l2.44985644 2.4469734v-8.67638639c0-.54139983.43698413-.98042709.98493125-.98159081l7.89910522-.0043627c.5451687 0 .9871152.44142642.9871152.98595351s-.4419465.98595351-.9871152.98595351z" fill-rule="evenodd" transform="matrix(-1 0 0 -1 14 15)"></path></symbol><symbol id="icon-arrow-down" viewBox="0 0 16 16"><path d="m3.28337502 11.5302405 4.03074001 4.176208c.37758093.3912076.98937525.3916069 1.367372-.0000316l4.03091977-4.1763942c.3775978-.3912252.3838182-1.0190815.0160006-1.4001736-.3775061-.39113013-.9877245-.39303641-1.3700683.003106l-2.39538585 2.4818345v-11.6147896l-.00649339-.11662112c-.055753-.49733869-.46370161-.88337888-.95867408-.88337888-.49497246 0-.90292107.38604019-.95867408.88337888l-.00649338.11662112v11.6147896l-2.39518594-2.4816273c-.37913917-.39282218-.98637524-.40056175-1.35419292-.0194697-.37750607.3911302-.37784433 1.0249269.00013556 1.4165479z" fill-rule="evenodd"></path></symbol><symbol id="icon-arrow-left" viewBox="0 0 16 16"><path d="m4.46975946 3.28337502-4.17620792 4.03074001c-.39120768.37758093-.39160691.98937525.0000316 1.367372l4.1763942 4.03091977c.39122514.3775978 1.01908149.3838182 1.40017357.0160006.39113012-.3775061.3930364-.9877245-.00310603-1.3700683l-2.48183446-2.39538585h11.61478958l.1166211-.00649339c.4973387-.055753.8833789-.46370161.8833789-.95867408 0-.49497246-.3860402-.90292107-.8833789-.95867408l-.1166211-.00649338h-11.61478958l2.4816273-2.39518594c.39282216-.37913917.40056173-.98637524.01946965-1.35419292-.39113012-.37750607-1.02492687-.37784433-1.41654791.00013556z" fill-rule="evenodd"></path></symbol><symbol id="icon-arrow-right" viewBox="0 0 16 16"><path d="m11.5302405 12.716625 4.176208-4.03074003c.3912076-.37758093.3916069-.98937525-.0000316-1.367372l-4.1763942-4.03091981c-.3912252-.37759778-1.0190815-.38381821-1.4001736-.01600053-.39113013.37750607-.39303641.98772445.003106 1.37006824l2.4818345 2.39538588h-11.6147896l-.11662112.00649339c-.49733869.055753-.88337888.46370161-.88337888.95867408 0 .49497246.38604019.90292107.88337888.95867408l.11662112.00649338h11.6147896l-2.4816273 2.39518592c-.39282218.3791392-.40056175.9863753-.0194697 1.3541929.3911302.3775061 1.0249269.3778444 1.4165479-.0001355z" fill-rule="evenodd"></path></symbol><symbol id="icon-arrow-sub" viewBox="0 0 16 16"><path d="m7.89692134 4.97190702v7.69043288l-2.44965196-2.4467692c-.38776071-.38730434-1.0088052-.39493519-1.38498697-.0191961-.38609047.3856361-.38643643 1.0105302.00013864 1.3966504l4.12239817 4.1175468c.38616704.3857126 1.01187344.3861062 1.39846576-.0000311l4.12258202-4.1177306c.3861843-.3857298.3925462-1.0047669.0163644-1.380506-.3860905-.38563612-1.0101851-.38751563-1.4012233.0030624l-2.44985643 2.4469734v-8.67638639c0-.54139983-.43698413-.98042709-.98493125-.98159081l-7.89910525-.0043627c-.54516866 0-.98711517.44142642-.98711517.98595351s.44194651.98595351.98711517.98595351z" fill-rule="evenodd"></path></symbol><symbol id="icon-arrow-up" viewBox="0 0 16 16"><path d="m12.716625 4.46975946-4.03074003-4.17620792c-.37758093-.39120768-.98937525-.39160691-1.367372.0000316l-4.03091981 4.1763942c-.37759778.39122514-.38381821 1.01908149-.01600053 1.40017357.37750607.39113012.98772445.3930364 1.37006824-.00310603l2.39538588-2.48183446v11.61478958l.00649339.1166211c.055753.4973387.46370161.8833789.95867408.8833789.49497246 0 .90292107-.3860402.95867408-.8833789l.00649338-.1166211v-11.61478958l2.39518592 2.4816273c.3791392.39282216.9863753.40056173 1.3541929.01946965.3775061-.39113012.3778444-1.02492687-.0001355-1.41654791z" fill-rule="evenodd"></path></symbol><symbol id="icon-article" viewBox="0 0 18 18"><path d="m13 15v-12.9906311c0-.0073595-.0019884-.0093689.0014977-.0093689l-11.00158888.00087166v13.00506804c0 .5482678.44615281.9940603.99415146.9940603h10.27350412c-.1701701-.2941734-.2675644-.6357129-.2675644-1zm-12 .0059397v-13.00506804c0-.5562408.44704472-1.00087166.99850233-1.00087166h11.00299537c.5510129 0 .9985023.45190985.9985023 1.0093689v2.9906311h3v9.9914698c0 1.1065798-.8927712 2.0085302-1.9940603 2.0085302h-12.01187942c-1.09954652 0-1.99406028-.8927712-1.99406028-1.9940603zm13-9.0059397v9c0 .5522847.4477153 1 1 1s1-.4477153 1-1v-9zm-10-2h7v4h-7zm1 1v2h5v-2zm-1 4h7v1h-7zm0 2h7v1h-7zm0 2h7v1h-7z" fill-rule="evenodd"></path></symbol><symbol id="icon-audio" viewBox="0 0 18 18"><path d="m13.0957477 13.5588459c-.195279.1937043-.5119137.193729-.7072234.0000551-.1953098-.193674-.1953346-.5077061-.0000556-.7014104 1.0251004-1.0168342 1.6108711-2.3905226 1.6108711-3.85745208 0-1.46604976-.5850634-2.83898246-1.6090736-3.85566829-.1951894-.19379323-.1950192-.50782531.0003802-.70141028.1953993-.19358497.512034-.19341614.7072234.00037709 1.2094886 1.20083761 1.901635 2.8250555 1.901635 4.55670148 0 1.73268608-.6929822 3.35779608-1.9037571 4.55880738zm2.1233994 2.1025159c-.195234.193749-.5118687.1938462-.7072235.0002171-.1953548-.1936292-.1954528-.5076613-.0002189-.7014104 1.5832215-1.5711805 2.4881302-3.6939808 2.4881302-5.96012998 0-2.26581266-.9046382-4.3883241-2.487443-5.95944795-.1952117-.19377107-.1950777-.50780316.0002993-.70141031s.5120117-.19347426.7072234.00029682c1.7683321 1.75528196 2.7800854 4.12911258 2.7800854 6.66056144 0 2.53182498-1.0120556 4.90597838-2.7808529 6.66132328zm-14.21898205-3.6854911c-.5523759 0-1.00016505-.4441085-1.00016505-.991944v-3.96777631c0-.54783558.44778915-.99194407 1.00016505-.99194407h2.0003301l5.41965617-3.8393633c.44948677-.31842296 1.07413994-.21516983 1.39520191.23062232.12116339.16823446.18629727.36981184.18629727.57655577v12.01603479c0 .5478356-.44778914.9919441-1.00016505.9919441-.20845738 0-.41170538-.0645985-.58133413-.184766l-5.41965617-3.8393633zm0-.991944h2.32084805l5.68047235 4.0241292v-12.01603479l-5.68047235 4.02412928h-2.32084805z" fill-rule="evenodd"></path></symbol><symbol id="icon-block" viewBox="0 0 24 24"><path d="m0 0h24v24h-24z" fill-rule="evenodd"></path></symbol><symbol id="icon-book" viewBox="0 0 18 18"><path d="m4 13v-11h1v11h11v-11h-13c-.55228475 0-1 .44771525-1 1v10.2675644c.29417337-.1701701.63571286-.2675644 1-.2675644zm12 1h-13c-.55228475 0-1 .4477153-1 1s.44771525 1 1 1h13zm0 3h-13c-1.1045695 0-2-.8954305-2-2v-12c0-1.1045695.8954305-2 2-2h13c.5522847 0 1 .44771525 1 1v14c0 .5522847-.4477153 1-1 1zm-8.5-13h6c.2761424 0 .5.22385763.5.5s-.2238576.5-.5.5h-6c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5zm1 2h4c.2761424 0 .5.22385763.5.5s-.2238576.5-.5.5h-4c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5z" fill-rule="evenodd"></path></symbol><symbol id="icon-broad" viewBox="0 0 24 24"><path d="m9.18274226 7.81v7.7999954l2.48162734-2.4816273c.3928221-.3928221 1.0219731-.4005617 1.4030652-.0194696.3911301.3911301.3914806 1.0249268-.0001404 1.4165479l-4.17620796 4.1762079c-.39120769.3912077-1.02508144.3916069-1.41671995-.0000316l-4.1763942-4.1763942c-.39122514-.3912251-.39767006-1.0190815-.01657798-1.4001736.39113012-.3911301 1.02337106-.3930364 1.41951349.0031061l2.48183446 2.4818344v-8.7999954c0-.54911294.4426881-.99439484.99778758-.99557515l8.00221246-.00442485c.5522847 0 1 .44771525 1 1s-.4477153 1-1 1z" fill-rule="evenodd" transform="matrix(-1 0 0 -1 20.182742 24.805206)"></path></symbol><symbol id="icon-calendar" viewBox="0 0 18 18"><path d="m12.5 0c.2761424 0 .5.21505737.5.49047852v.50952148h2c1.1072288 0 2 .89451376 2 2v12c0 1.1072288-.8945138 2-2 2h-12c-1.1072288 0-2-.8945138-2-2v-12c0-1.1072288.89451376-2 2-2h1v1h-1c-.55393837 0-1 .44579254-1 1v3h14v-3c0-.55393837-.4457925-1-1-1h-2v1.50952148c0 .27088381-.2319336.49047852-.5.49047852-.2761424 0-.5-.21505737-.5-.49047852v-3.01904296c0-.27088381.2319336-.49047852.5-.49047852zm3.5 7h-14v8c0 .5539384.44579254 1 1 1h12c.5539384 0 1-.4457925 1-1zm-11 6v1h-1v-1zm3 0v1h-1v-1zm3 0v1h-1v-1zm-6-2v1h-1v-1zm3 0v1h-1v-1zm6 0v1h-1v-1zm-3 0v1h-1v-1zm-3-2v1h-1v-1zm6 0v1h-1v-1zm-3 0v1h-1v-1zm-5.5-9c.27614237 0 .5.21505737.5.49047852v.50952148h5v1h-5v1.50952148c0 .27088381-.23193359.49047852-.5.49047852-.27614237 0-.5-.21505737-.5-.49047852v-3.01904296c0-.27088381.23193359-.49047852.5-.49047852z" fill-rule="evenodd"></path></symbol><symbol id="icon-cart" viewBox="0 0 18 18"><path d="m5 14c1.1045695 0 2 .8954305 2 2s-.8954305 2-2 2-2-.8954305-2-2 .8954305-2 2-2zm10 0c1.1045695 0 2 .8954305 2 2s-.8954305 2-2 2-2-.8954305-2-2 .8954305-2 2-2zm-10 1c-.55228475 0-1 .4477153-1 1s.44771525 1 1 1 1-.4477153 1-1-.44771525-1-1-1zm10 0c-.5522847 0-1 .4477153-1 1s.4477153 1 1 1 1-.4477153 1-1-.4477153-1-1-1zm-12.82032249-15c.47691417 0 .88746157.33678127.98070211.80449199l.23823144 1.19501025 13.36277974.00045554c.5522847.00001882.9999659.44774934.9999659 1.00004222 0 .07084994-.0075361.14150708-.022474.2107727l-1.2908094 5.98534344c-.1007861.46742419-.5432548.80388386-1.0571651.80388386h-10.24805106c-.59173366 0-1.07142857.4477153-1.07142857 1 0 .5128358.41361449.9355072.94647737.9932723l.1249512.0067277h10.35933776c.2749512 0 .4979349.2228539.4979349.4978051 0 .2749417-.2227336.4978951-.4976753.4980063l-10.35959736.0041886c-1.18346732 0-2.14285714-.8954305-2.14285714-2 0-.6625717.34520317-1.24989198.87690425-1.61383592l-1.63768102-8.19004794c-.01312273-.06561364-.01950005-.131011-.0196107-.19547395l-1.71961253-.00064219c-.27614237 0-.5-.22385762-.5-.5 0-.27614237.22385763-.5.5-.5zm14.53193359 2.99950224h-13.11300004l1.20580469 6.02530174c.11024034-.0163252.22327998-.02480398.33844139-.02480398h10.27064786z"></path></symbol><symbol id="icon-chevron-less" viewBox="0 0 10 10"><path d="m5.58578644 4-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4c-.39052429.39052429-1.02368927.39052429-1.41421356 0s-.39052429-1.02368927 0-1.41421356z" fill-rule="evenodd" transform="matrix(0 -1 -1 0 9 9)"></path></symbol><symbol id="icon-chevron-more" viewBox="0 0 10 10"><path d="m5.58578644 6-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4.00000002c-.39052429.3905243-1.02368927.3905243-1.41421356 0s-.39052429-1.02368929 0-1.41421358z" fill-rule="evenodd" transform="matrix(0 1 -1 0 11 1)"></path></symbol><symbol id="icon-chevron-right" viewBox="0 0 10 10"><path d="m5.96738168 4.70639573 2.39518594-2.41447274c.37913917-.38219212.98637524-.38972225 1.35419292-.01894278.37750606.38054586.37784436.99719163-.00013556 1.37821513l-4.03074001 4.06319683c-.37758093.38062133-.98937525.38100976-1.367372-.00003075l-4.03091981-4.06337806c-.37759778-.38063832-.38381821-.99150444-.01600053-1.3622839.37750607-.38054587.98772445-.38240057 1.37006824.00302197l2.39538588 2.4146743.96295325.98624457z" fill-rule="evenodd" transform="matrix(0 -1 1 0 0 10)"></path></symbol><symbol id="icon-circle-fill" viewBox="0 0 16 16"><path d="m8 14c-3.3137085 0-6-2.6862915-6-6s2.6862915-6 6-6 6 2.6862915 6 6-2.6862915 6-6 6z" fill-rule="evenodd"></path></symbol><symbol id="icon-circle" viewBox="0 0 16 16"><path d="m8 12c2.209139 0 4-1.790861 4-4s-1.790861-4-4-4-4 1.790861-4 4 1.790861 4 4 4zm0 2c-3.3137085 0-6-2.6862915-6-6s2.6862915-6 6-6 6 2.6862915 6 6-2.6862915 6-6 6z" fill-rule="evenodd"></path></symbol><symbol id="icon-citation" viewBox="0 0 18 18"><path d="m8.63593473 5.99995183c2.20913897 0 3.99999997 1.79084375 3.99999997 3.99996146 0 1.40730761-.7267788 2.64486871-1.8254829 3.35783281 1.6240224.6764218 2.8754442 2.0093871 3.4610603 3.6412466l-1.0763845.000006c-.5310008-1.2078237-1.5108121-2.1940153-2.7691712-2.7181346l-.79002167-.329052v-1.023992l.63016577-.4089232c.8482885-.5504661 1.3698342-1.4895187 1.3698342-2.51898361 0-1.65683828-1.3431457-2.99996146-2.99999997-2.99996146-1.65685425 0-3 1.34312318-3 2.99996146 0 1.02946491.52154569 1.96851751 1.36983419 2.51898361l.63016581.4089232v1.023992l-.79002171.329052c-1.25835905.5241193-2.23817037 1.5103109-2.76917113 2.7181346l-1.07638453-.000006c.58561612-1.6318595 1.8370379-2.9648248 3.46106024-3.6412466-1.09870405-.7129641-1.82548287-1.9505252-1.82548287-3.35783281 0-2.20911771 1.790861-3.99996146 4-3.99996146zm7.36897597-4.99995183c1.1018574 0 1.9950893.89353404 1.9950893 2.00274083v5.994422c0 1.10608317-.8926228 2.00274087-1.9950893 2.00274087l-3.0049107-.0009037v-1l3.0049107.00091329c.5490631 0 .9950893-.44783123.9950893-1.00275046v-5.994422c0-.55646537-.4450595-1.00275046-.9950893-1.00275046h-14.00982141c-.54906309 0-.99508929.44783123-.99508929 1.00275046v5.9971821c0 .66666024.33333333.99999036 1 .99999036l2-.00091329v1l-2 .0009037c-1 0-2-.99999041-2-1.99998077v-5.9971821c0-1.10608322.8926228-2.00274083 1.99508929-2.00274083zm-8.5049107 2.9999711c.27614237 0 .5.22385547.5.5 0 .2761349-.22385763.5-.5.5h-4c-.27614237 0-.5-.2238651-.5-.5 0-.27614453.22385763-.5.5-.5zm3 0c.2761424 0 .5.22385547.5.5 0 .2761349-.2238576.5-.5.5h-1c-.27614237 0-.5-.2238651-.5-.5 0-.27614453.22385763-.5.5-.5zm4 0c.2761424 0 .5.22385547.5.5 0 .2761349-.2238576.5-.5.5h-2c-.2761424 0-.5-.2238651-.5-.5 0-.27614453.2238576-.5.5-.5z" fill-rule="evenodd"></path></symbol><symbol id="icon-close" viewBox="0 0 16 16"><path d="m2.29679575 12.2772478c-.39658757.3965876-.39438847 1.0328109-.00062148 1.4265779.39651227.3965123 1.03246768.3934888 1.42657791-.0006214l4.27724782-4.27724787 4.2772478 4.27724787c.3965876.3965875 1.0328109.3943884 1.4265779.0006214.3965123-.3965122.3934888-1.0324677-.0006214-1.4265779l-4.27724787-4.2772478 4.27724787-4.27724782c.3965875-.39658757.3943884-1.03281091.0006214-1.42657791-.3965122-.39651226-1.0324677-.39348875-1.4265779.00062148l-4.2772478 4.27724782-4.27724782-4.27724782c-.39658757-.39658757-1.03281091-.39438847-1.42657791-.00062148-.39651226.39651227-.39348875 1.03246768.00062148 1.42657791l4.27724782 4.27724782z" fill-rule="evenodd"></path></symbol><symbol id="icon-collections" viewBox="0 0 18 18"><path d="m15 4c1.1045695 0 2 .8954305 2 2v9c0 1.1045695-.8954305 2-2 2h-8c-1.1045695 0-2-.8954305-2-2h1c0 .5128358.38604019.9355072.88337887.9932723l.11662113.0067277h8c.5128358 0 .9355072-.3860402.9932723-.8833789l.0067277-.1166211v-9c0-.51283584-.3860402-.93550716-.8833789-.99327227l-.1166211-.00672773h-1v-1zm-4-3c1.1045695 0 2 .8954305 2 2v9c0 1.1045695-.8954305 2-2 2h-8c-1.1045695 0-2-.8954305-2-2v-9c0-1.1045695.8954305-2 2-2zm0 1h-8c-.51283584 0-.93550716.38604019-.99327227.88337887l-.00672773.11662113v9c0 .5128358.38604019.9355072.88337887.9932723l.11662113.0067277h8c.5128358 0 .9355072-.3860402.9932723-.8833789l.0067277-.1166211v-9c0-.51283584-.3860402-.93550716-.8833789-.99327227zm-1.5 7c.27614237 0 .5.22385763.5.5s-.22385763.5-.5.5h-5c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5zm0-2c.27614237 0 .5.22385763.5.5s-.22385763.5-.5.5h-5c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5zm0-2c.27614237 0 .5.22385763.5.5s-.22385763.5-.5.5h-5c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5z" fill-rule="evenodd"></path></symbol><symbol id="icon-compare" viewBox="0 0 18 18"><path d="m12 3c3.3137085 0 6 2.6862915 6 6s-2.6862915 6-6 6c-1.0928452 0-2.11744941-.2921742-2.99996061-.8026704-.88181407.5102749-1.90678042.8026704-3.00003939.8026704-3.3137085 0-6-2.6862915-6-6s2.6862915-6 6-6c1.09325897 0 2.11822532.29239547 3.00096303.80325037.88158756-.51107621 1.90619177-.80325037 2.99903697-.80325037zm-6 1c-2.76142375 0-5 2.23857625-5 5 0 2.7614237 2.23857625 5 5 5 .74397391 0 1.44999672-.162488 2.08451611-.4539116-1.27652344-1.1000812-2.08451611-2.7287264-2.08451611-4.5460884s.80799267-3.44600721 2.08434391-4.5463015c-.63434719-.29121054-1.34037-.4536985-2.08434391-.4536985zm6 0c-.7439739 0-1.4499967.16248796-2.08451611.45391156 1.27652341 1.10008123 2.08451611 2.72872644 2.08451611 4.54608844s-.8079927 3.4460072-2.08434391 4.5463015c.63434721.2912105 1.34037001.4536985 2.08434391.4536985 2.7614237 0 5-2.2385763 5-5 0-2.76142375-2.2385763-5-5-5zm-1.4162763 7.0005324h-3.16744736c.15614659.3572676.35283837.6927622.58425872 1.0006671h1.99892988c.23142036-.3079049.42811216-.6433995.58425876-1.0006671zm.4162763-2.0005324h-4c0 .34288501.0345146.67770871.10025909 1.0011864h3.79948181c.0657445-.32347769.1002591-.65830139.1002591-1.0011864zm-.4158423-1.99953894h-3.16831543c-.13859957.31730812-.24521946.651783-.31578599.99935097h3.79988742c-.0705665-.34756797-.1771864-.68204285-.315786-.99935097zm-1.58295822-1.999926-.08316107.06199199c-.34550042.27081213-.65446126.58611297-.91825862.93727862h2.00044041c-.28418626-.37830727-.6207872-.71499149-.99902072-.99927061z" fill-rule="evenodd"></path></symbol><symbol id="icon-download-file" viewBox="0 0 18 18"><path d="m10.0046024 0c.5497429 0 1.3179837.32258606 1.707238.71184039l4.5763192 4.57631922c.3931386.39313859.7118404 1.16760135.7118404 1.71431368v8.98899651c0 1.1092806-.8945138 2.0085302-1.9940603 2.0085302h-12.01187942c-1.10128908 0-1.99406028-.8926228-1.99406028-1.9950893v-14.00982141c0-1.10185739.88743329-1.99508929 1.99961498-1.99508929zm0 1h-7.00498742c-.55709576 0-.99961498.44271433-.99961498.99508929v14.00982141c0 .5500396.44491393.9950893.99406028.9950893h12.01187942c.5463747 0 .9940603-.4506622.9940603-1.0085302v-8.98899651c0-.28393444-.2150684-.80332809-.4189472-1.0072069l-4.5763192-4.57631922c-.2038461-.20384606-.718603-.41894717-1.0001312-.41894717zm-1.5046024 4c.27614237 0 .5.21637201.5.49209595v6.14827645l1.7462789-1.77990922c.1933927-.1971171.5125222-.19455839.7001689-.0069117.1932998.19329992.1910058.50899492-.0027774.70277812l-2.59089271 2.5908927c-.19483374.1948337-.51177825.1937771-.70556873-.0000133l-2.59099079-2.5909908c-.19484111-.1948411-.19043735-.5151448-.00279066-.70279146.19329987-.19329987.50465175-.19237083.70018565.00692852l1.74638684 1.78001764v-6.14827695c0-.27177709.23193359-.49209595.5-.49209595z" fill-rule="evenodd"></path></symbol><symbol id="icon-download" viewBox="0 0 16 16"><path d="m12.9975267 12.999368c.5467123 0 1.0024733.4478567 1.0024733 1.000316 0 .5563109-.4488226 1.000316-1.0024733 1.000316h-9.99505341c-.54671233 0-1.00247329-.4478567-1.00247329-1.000316 0-.5563109.44882258-1.000316 1.00247329-1.000316zm-4.9975267-11.999368c.55228475 0 1 .44497754 1 .99589209v6.80214418l2.4816273-2.48241149c.3928222-.39294628 1.0219732-.4006883 1.4030652-.01947579.3911302.39125371.3914806 1.02525073-.0001404 1.41699553l-4.17620792 4.17752758c-.39120769.3913313-1.02508144.3917306-1.41671995-.0000316l-4.17639421-4.17771394c-.39122513-.39134876-.39767006-1.01940351-.01657797-1.40061601.39113012-.39125372 1.02337105-.3931606 1.41951349.00310701l2.48183446 2.48261871v-6.80214418c0-.55001601.44386482-.99589209 1-.99589209z" fill-rule="evenodd"></path></symbol><symbol id="icon-editors" viewBox="0 0 18 18"><path d="m8.72592184 2.54588137c-.48811714-.34391207-1.08343326-.54588137-1.72592184-.54588137-1.65685425 0-3 1.34314575-3 3 0 1.02947485.5215457 1.96853646 1.3698342 2.51900785l.6301658.40892721v1.02400182l-.79002171.32905522c-1.93395773.8055207-3.20997829 2.7024791-3.20997829 4.8180274v.9009805h-1v-.9009805c0-2.5479714 1.54557359-4.79153984 3.82548288-5.7411543-1.09870406-.71297106-1.82548288-1.95054399-1.82548288-3.3578652 0-2.209139 1.790861-4 4-4 1.09079823 0 2.07961816.43662103 2.80122451 1.1446278-.37707584.09278571-.7373238.22835063-1.07530267.40125357zm-2.72592184 14.45411863h-1v-.9009805c0-2.5479714 1.54557359-4.7915398 3.82548288-5.7411543-1.09870406-.71297106-1.82548288-1.95054399-1.82548288-3.3578652 0-2.209139 1.790861-4 4-4s4 1.790861 4 4c0 1.40732121-.7267788 2.64489414-1.8254829 3.3578652 2.2799093.9496145 3.8254829 3.1931829 3.8254829 5.7411543v.9009805h-1v-.9009805c0-2.1155483-1.2760206-4.0125067-3.2099783-4.8180274l-.7900217-.3290552v-1.02400184l.6301658-.40892721c.8482885-.55047139 1.3698342-1.489533 1.3698342-2.51900785 0-1.65685425-1.3431458-3-3-3-1.65685425 0-3 1.34314575-3 3 0 1.02947485.5215457 1.96853646 1.3698342 2.51900785l.6301658.40892721v1.02400184l-.79002171.3290552c-1.93395773.8055207-3.20997829 2.7024791-3.20997829 4.8180274z" fill-rule="evenodd"></path></symbol><symbol id="icon-email" viewBox="0 0 18 18"><path d="m16.0049107 2c1.1018574 0 1.9950893.89706013 1.9950893 2.00585866v9.98828264c0 1.1078052-.8926228 2.0058587-1.9950893 2.0058587h-14.00982141c-1.10185739 0-1.99508929-.8970601-1.99508929-2.0058587v-9.98828264c0-1.10780515.8926228-2.00585866 1.99508929-2.00585866zm0 1h-14.00982141c-.54871518 0-.99508929.44887827-.99508929 1.00585866v9.98828264c0 .5572961.44630695 1.0058587.99508929 1.0058587h14.00982141c.5487152 0 .9950893-.4488783.9950893-1.0058587v-9.98828264c0-.55729607-.446307-1.00585866-.9950893-1.00585866zm-.0049107 2.55749512v1.44250488l-7 4-7-4v-1.44250488l7 4z" fill-rule="evenodd"></path></symbol><symbol id="icon-error" viewBox="0 0 18 18"><path d="m9 0c4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9zm2.8630343 4.71100931-2.8630343 2.86303426-2.86303426-2.86303426c-.39658757-.39658757-1.03281091-.39438847-1.4265779-.00062147-.39651227.39651226-.39348876 1.03246767.00062147 1.4265779l2.86303426 2.86303426-2.86303426 2.8630343c-.39658757.3965875-.39438847 1.0328109-.00062147 1.4265779.39651226.3965122 1.03246767.3934887 1.4265779-.0006215l2.86303426-2.8630343 2.8630343 2.8630343c.3965875.3965876 1.0328109.3943885 1.4265779.0006215.3965122-.3965123.3934887-1.0324677-.0006215-1.4265779l-2.8630343-2.8630343 2.8630343-2.86303426c.3965876-.39658757.3943885-1.03281091.0006215-1.4265779-.3965123-.39651227-1.0324677-.39348876-1.4265779.00062147z" fill-rule="evenodd"></path></symbol><symbol id="icon-ethics" viewBox="0 0 18 18"><path d="m6.76384967 1.41421356.83301651-.8330165c.77492941-.77492941 2.03133823-.77492941 2.80626762 0l.8330165.8330165c.3750728.37507276.8837806.58578644 1.4142136.58578644h1.3496361c1.1045695 0 2 .8954305 2 2v1.34963611c0 .53043298.2107137 1.03914081.5857864 1.41421356l.8330165.83301651c.7749295.77492941.7749295 2.03133823 0 2.80626762l-.8330165.8330165c-.3750727.3750728-.5857864.8837806-.5857864 1.4142136v1.3496361c0 1.1045695-.8954305 2-2 2h-1.3496361c-.530433 0-1.0391408.2107137-1.4142136.5857864l-.8330165.8330165c-.77492939.7749295-2.03133821.7749295-2.80626762 0l-.83301651-.8330165c-.37507275-.3750727-.88378058-.5857864-1.41421356-.5857864h-1.34963611c-1.1045695 0-2-.8954305-2-2v-1.3496361c0-.530433-.21071368-1.0391408-.58578644-1.4142136l-.8330165-.8330165c-.77492941-.77492939-.77492941-2.03133821 0-2.80626762l.8330165-.83301651c.37507276-.37507275.58578644-.88378058.58578644-1.41421356v-1.34963611c0-1.1045695.8954305-2 2-2h1.34963611c.53043298 0 1.03914081-.21071368 1.41421356-.58578644zm-1.41421356 1.58578644h-1.34963611c-.55228475 0-1 .44771525-1 1v1.34963611c0 .79564947-.31607052 1.55871121-.87867966 2.12132034l-.8330165.83301651c-.38440512.38440512-.38440512 1.00764896 0 1.39205408l.8330165.83301646c.56260914.5626092.87867966 1.3256709.87867966 2.1213204v1.3496361c0 .5522847.44771525 1 1 1h1.34963611c.79564947 0 1.55871121.3160705 2.12132034.8786797l.83301651.8330165c.38440512.3844051 1.00764896.3844051 1.39205408 0l.83301646-.8330165c.5626092-.5626092 1.3256709-.8786797 2.1213204-.8786797h1.3496361c.5522847 0 1-.4477153 1-1v-1.3496361c0-.7956495.3160705-1.5587112.8786797-2.1213204l.8330165-.83301646c.3844051-.38440512.3844051-1.00764896 0-1.39205408l-.8330165-.83301651c-.5626092-.56260913-.8786797-1.32567087-.8786797-2.12132034v-1.34963611c0-.55228475-.4477153-1-1-1h-1.3496361c-.7956495 0-1.5587112-.31607052-2.1213204-.87867966l-.83301646-.8330165c-.38440512-.38440512-1.00764896-.38440512-1.39205408 0l-.83301651.8330165c-.56260913.56260914-1.32567087.87867966-2.12132034.87867966zm3.58698944 11.4960218c-.02081224.002155-.04199226.0030286-.06345763.002542-.98766446-.0223875-1.93408568-.3063547-2.75885125-.8155622-.23496767-.1450683-.30784554-.4531483-.16277726-.688116.14506827-.2349677.45314827-.3078455.68811595-.1627773.67447084.4164161 1.44758575.6483839 2.25617384.6667123.01759529.0003988.03495764.0017019.05204365.0038639.01713363-.0017748.03452416-.0026845.05212715-.0026845 2.4852814 0 4.5-2.0147186 4.5-4.5 0-1.04888973-.3593547-2.04134635-1.0074477-2.83787157-.1742817-.21419731-.1419238-.5291218.0722736-.70340353.2141973-.17428173.5291218-.14192375.7034035.07227357.7919032.97327203 1.2317706 2.18808682 1.2317706 3.46900153 0 3.0375661-2.4624339 5.5-5.5 5.5-.02146768 0-.04261937-.0013529-.06337445-.0039782zm1.57975095-10.78419583c.2654788.07599731.419084.35281842.3430867.61829728-.0759973.26547885-.3528185.419084-.6182973.3430867-.37560116-.10752146-.76586237-.16587951-1.15568824-.17249193-2.5587807-.00064534-4.58547766 2.00216524-4.58547766 4.49928198 0 .62691557.12797645 1.23496.37274865 1.7964426.11035133.2531347-.0053975.5477984-.25853224.6581497-.25313473.1103514-.54779841-.0053975-.65814974-.2585322-.29947131-.6869568-.45606667-1.43097603-.45606667-2.1960601 0-3.05211432 2.47714695-5.50006595 5.59399617-5.49921198.48576182.00815502.96289603.0795037 1.42238033.21103795zm-1.9766658 6.41091303 2.69835-2.94655317c.1788432-.21040373.4943901-.23598862.7047939-.05714545.2104037.17884318.2359886.49439014.0571454.70479387l-3.01637681 3.34277395c-.18039088.1999106-.48669547.2210637-.69285412.0478478l-1.93095347-1.62240047c-.21213845-.17678204-.24080048-.49206439-.06401844-.70420284.17678204-.21213844.49206439-.24080048.70420284-.06401844z" fill-rule="evenodd"></path></symbol><symbol id="icon-expand"><path d="M7.498 11.918a.997.997 0 0 0-.003-1.411.995.995 0 0 0-1.412-.003l-4.102 4.102v-3.51A1 1 0 0 0 .98 10.09.992.992 0 0 0 0 11.092V17c0 .554.448 1.002 1.002 1.002h5.907c.554 0 1.002-.45 1.002-1.003 0-.539-.45-.978-1.006-.978h-3.51zm3.005-5.835a.997.997 0 0 0 .003 1.412.995.995 0 0 0 1.411.003l4.103-4.103v3.51a1 1 0 0 0 1.001 1.006A.992.992 0 0 0 18 6.91V1.002A1 1 0 0 0 17 0h-5.907a1.003 1.003 0 0 0-1.002 1.003c0 .539.45.978 1.006.978h3.51z" fill-rule="evenodd"></path></symbol><symbol id="icon-explore" viewBox="0 0 18 18"><path d="m9 17c4.418278 0 8-3.581722 8-8s-3.581722-8-8-8-8 3.581722-8 8 3.581722 8 8 8zm0 1c-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9 4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9zm0-2.5c-.27614237 0-.5-.2238576-.5-.5s.22385763-.5.5-.5c2.969509 0 5.400504-2.3575119 5.497023-5.31714844.0090007-.27599565.2400359-.49243782.5160315-.48343711.2759957.0090007.4924378.2400359.4834371.51603155-.114093 3.4985237-2.9869632 6.284554-6.4964916 6.284554zm-.29090657-12.99359748c.27587424-.01216621.50937715.20161139.52154336.47748563.01216621.27587423-.20161139.50937715-.47748563.52154336-2.93195733.12930094-5.25315116 2.54886451-5.25315116 5.49456849 0 .27614237-.22385763.5-.5.5s-.5-.22385763-.5-.5c0-3.48142406 2.74307146-6.34074398 6.20909343-6.49359748zm1.13784138 8.04763908-1.2004882-1.20048821c-.19526215-.19526215-.19526215-.51184463 0-.70710678s.51184463-.19526215.70710678 0l1.20048821 1.2004882 1.6006509-4.00162734-4.50670359 1.80268144-1.80268144 4.50670359zm4.10281269-6.50378907-2.6692597 6.67314927c-.1016411.2541026-.3029834.4554449-.557086.557086l-6.67314927 2.6692597 2.66925969-6.67314926c.10164107-.25410266.30298336-.45544495.55708602-.55708602z" fill-rule="evenodd"></path></symbol><symbol id="icon-filter" viewBox="0 0 16 16"><path d="m14.9738641 0c.5667192 0 1.0261359.4477136 1.0261359 1 0 .24221858-.0902161.47620768-.2538899.65849851l-5.6938314 6.34147206v5.49997973c0 .3147562-.1520673.6111434-.4104543.7999971l-2.05227171 1.4999945c-.45337535.3313696-1.09655869.2418269-1.4365902-.1999993-.13321514-.1730955-.20522717-.3836284-.20522717-.5999978v-6.99997423l-5.69383133-6.34147206c-.3731872-.41563511-.32996891-1.0473954.09653074-1.41107611.18705584-.15950448.42716133-.2474224.67571519-.2474224zm-5.9218641 8.5h-2.105v6.491l.01238459.0070843.02053271.0015705.01955278-.0070558 2.0532976-1.4990996zm-8.02585008-7.5-.01564945.00240169 5.83249953 6.49759831h2.313l5.836-6.499z"></path></symbol><symbol id="icon-home" viewBox="0 0 18 18"><path d="m9 5-6 6v5h4v-4h4v4h4v-5zm7 6.5857864v4.4142136c0 .5522847-.4477153 1-1 1h-5v-4h-2v4h-5c-.55228475 0-1-.4477153-1-1v-4.4142136c-.25592232 0-.51184464-.097631-.70710678-.2928932l-.58578644-.5857864c-.39052429-.3905243-.39052429-1.02368929 0-1.41421358l8.29289322-8.29289322 8.2928932 8.29289322c.3905243.39052429.3905243 1.02368928 0 1.41421358l-.5857864.5857864c-.1952622.1952622-.4511845.2928932-.7071068.2928932zm-7-9.17157284-7.58578644 7.58578644.58578644.5857864 7-6.99999996 7 6.99999996.5857864-.5857864z" fill-rule="evenodd"></path></symbol><symbol id="icon-image" viewBox="0 0 18 18"><path d="m10.0046024 0c.5497429 0 1.3179837.32258606 1.707238.71184039l4.5763192 4.57631922c.3931386.39313859.7118404 1.16760135.7118404 1.71431368v8.98899651c0 1.1092806-.8945138 2.0085302-1.9940603 2.0085302h-12.01187942c-1.10128908 0-1.99406028-.8926228-1.99406028-1.9950893v-14.00982141c0-1.10185739.88743329-1.99508929 1.99961498-1.99508929zm-3.49645283 10.1752453-3.89407257 6.7495552c.11705545.048464.24538859.0751995.37998328.0751995h10.60290092l-2.4329715-4.2154691-1.57494129 2.7288098zm8.49779013 6.8247547c.5463747 0 .9940603-.4506622.9940603-1.0085302v-8.98899651c0-.28393444-.2150684-.80332809-.4189472-1.0072069l-4.5763192-4.57631922c-.2038461-.20384606-.718603-.41894717-1.0001312-.41894717h-7.00498742c-.55709576 0-.99961498.44271433-.99961498.99508929v13.98991071l4.50814957-7.81026689 3.08089884 5.33809539 1.57494129-2.7288097 3.5875735 6.2159812zm-3.0059397-11c1.1045695 0 2 .8954305 2 2s-.8954305 2-2 2-2-.8954305-2-2 .8954305-2 2-2zm0 1c-.5522847 0-1 .44771525-1 1s.4477153 1 1 1 1-.44771525 1-1-.4477153-1-1-1z" fill-rule="evenodd"></path></symbol><symbol id="icon-info" viewBox="0 0 18 18"><path d="m9 0c4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9zm0 7h-1.5l-.11662113.00672773c-.49733868.05776511-.88337887.48043643-.88337887.99327227 0 .47338693.32893365.86994729.77070917.97358929l.1126697.01968298.11662113.00672773h.5v3h-.5l-.11662113.0067277c-.42082504.0488782-.76196299.3590206-.85696816.7639815l-.01968298.1126697-.00672773.1166211.00672773.1166211c.04887817.4208251.35902055.761963.76398144.8569682l.1126697.019683.11662113.0067277h3l.1166211-.0067277c.4973387-.0577651.8833789-.4804365.8833789-.9932723 0-.4733869-.3289337-.8699473-.7707092-.9735893l-.1126697-.019683-.1166211-.0067277h-.5v-4l-.00672773-.11662113c-.04887817-.42082504-.35902055-.76196299-.76398144-.85696816l-.1126697-.01968298zm0-3.25c-.69035594 0-1.25.55964406-1.25 1.25s.55964406 1.25 1.25 1.25 1.25-.55964406 1.25-1.25-.55964406-1.25-1.25-1.25z" fill-rule="evenodd"></path></symbol><symbol id="icon-institution" viewBox="0 0 18 18"><path d="m7 16.9998189v-2.0003623h4v2.0003623h2v-3.0005434h-8v3.0005434zm-3-10.00181122h-1.52632364c-.27614237 0-.5-.22389817-.5-.50009056 0-.13995446.05863589-.27350497.16166338-.36820841l1.23156713-1.13206327h-2.36690687v12.00217346h3v-2.0003623h-3v-1.0001811h3v-1.0001811h1v-4.00072448h-1zm10 0v2.00036224h-1v4.00072448h1v1.0001811h3v1.0001811h-3v2.0003623h3v-12.00217346h-2.3695309l1.2315671 1.13206327c.2033191.186892.2166633.50325042.0298051.70660631-.0946863.10304615-.2282126.16169266-.3681417.16169266zm3-3.00054336c.5522847 0 1 .44779634 1 1.00018112v13.00235456h-18v-13.00235456c0-.55238478.44771525-1.00018112 1-1.00018112h3.45499992l4.20535144-3.86558216c.19129876-.17584288.48537447-.17584288.67667324 0l4.2053514 3.86558216zm-4 3.00054336h-8v1.00018112h8zm-2 6.00108672h1v-4.00072448h-1zm-1 0v-4.00072448h-2v4.00072448zm-3 0v-4.00072448h-1v4.00072448zm8-4.00072448c.5522847 0 1 .44779634 1 1.00018112v2.00036226h-2v-2.00036226c0-.55238478.4477153-1.00018112 1-1.00018112zm-12 0c.55228475 0 1 .44779634 1 1.00018112v2.00036226h-2v-2.00036226c0-.55238478.44771525-1.00018112 1-1.00018112zm5.99868798-7.81907007-5.24205601 4.81852671h10.48411203zm.00131202 3.81834559c-.55228475 0-1-.44779634-1-1.00018112s.44771525-1.00018112 1-1.00018112 1 .44779634 1 1.00018112-.44771525 1.00018112-1 1.00018112zm-1 11.00199236v1.0001811h2v-1.0001811z" fill-rule="evenodd"></path></symbol><symbol id="icon-location" viewBox="0 0 18 18"><path d="m9.39521328 16.2688008c.79596342-.7770119 1.59208152-1.6299956 2.33285652-2.5295081 1.4020032-1.7024324 2.4323601-3.3624519 2.9354918-4.871847.2228715-.66861448.3364384-1.29323246.3364384-1.8674457 0-3.3137085-2.6862915-6-6-6-3.36356866 0-6 2.60156856-6 6 0 .57421324.11356691 1.19883122.3364384 1.8674457.50313169 1.5093951 1.53348863 3.1694146 2.93549184 4.871847.74077492.8995125 1.53689309 1.7524962 2.33285648 2.5295081.13694479.1336842.26895677.2602648.39521328.3793207.12625651-.1190559.25826849-.2456365.39521328-.3793207zm-.39521328 1.7311992s-7-6-7-11c0-4 3.13400675-7 7-7 3.8659932 0 7 3.13400675 7 7 0 5-7 11-7 11zm0-8c-1.65685425 0-3-1.34314575-3-3s1.34314575-3 3-3c1.6568542 0 3 1.34314575 3 3s-1.3431458 3-3 3zm0-1c1.1045695 0 2-.8954305 2-2s-.8954305-2-2-2-2 .8954305-2 2 .8954305 2 2 2z" fill-rule="evenodd"></path></symbol><symbol id="icon-minus" viewBox="0 0 16 16"><path d="m2.00087166 7h11.99825664c.5527662 0 1.0008717.44386482 1.0008717 1 0 .55228475-.4446309 1-1.0008717 1h-11.99825664c-.55276616 0-1.00087166-.44386482-1.00087166-1 0-.55228475.44463086-1 1.00087166-1z" fill-rule="evenodd"></path></symbol><symbol id="icon-newsletter" viewBox="0 0 18 18"><path d="m9 11.8482489 2-1.1428571v-1.7053918h-4v1.7053918zm-3-1.7142857v-2.1339632h6v2.1339632l3-1.71428574v-6.41967746h-12v6.41967746zm10-5.3839632 1.5299989.95624934c.2923814.18273835.4700011.50320827.4700011.8479983v8.44575236c0 1.1045695-.8954305 2-2 2h-14c-1.1045695 0-2-.8954305-2-2v-8.44575236c0-.34479003.1776197-.66525995.47000106-.8479983l1.52999894-.95624934v-2.75c0-.55228475.44771525-1 1-1h12c.5522847 0 1 .44771525 1 1zm0 1.17924764v3.07075236l-7 4-7-4v-3.07075236l-1 .625v8.44575236c0 .5522847.44771525 1 1 1h14c.5522847 0 1-.4477153 1-1v-8.44575236zm-10-1.92924764h6v1h-6zm-1 2h8v1h-8z" fill-rule="evenodd"></path></symbol><symbol id="icon-orcid" viewBox="0 0 18 18"><path d="m9 1c4.418278 0 8 3.581722 8 8s-3.581722 8-8 8-8-3.581722-8-8 3.581722-8 8-8zm-2.90107518 5.2732337h-1.41865256v7.1712107h1.41865256zm4.55867178.02508949h-2.99247027v7.14612121h2.91062487c.7673039 0 1.4476365-.1483432 2.0410182-.445034s1.0511995-.7152915 1.3734671-1.2558144c.3222677-.540523.4833991-1.1603247.4833991-1.85942385 0-.68545815-.1602789-1.30270225-.4808414-1.85175082-.3205625-.54904856-.7707074-.97532211-1.3504481-1.27883343-.5797408-.30351132-1.2413173-.45526471-1.9847495-.45526471zm-.1892674 1.07933542c.7877654 0 1.4143875.22336734 1.8798852.67010873.4654977.44674138.698243 1.05546001.698243 1.82617415 0 .74343221-.2310402 1.34447791-.6931277 1.80315511-.4620874.4586773-1.0750688.6880124-1.8389625.6880124h-1.46810075v-4.98745039zm-5.08652545-3.71099194c-.21825533 0-.410525.08444276-.57681478.25333081-.16628977.16888806-.24943341.36245684-.24943341.58071218 0 .22345188.08314364.41961891.24943341.58850696.16628978.16888806.35855945.25333082.57681478.25333082.233845 0 .43390938-.08314364.60019916-.24943342.16628978-.16628977.24943342-.36375592.24943342-.59240436 0-.233845-.08314364-.43131115-.24943342-.59240437s-.36635416-.24163862-.60019916-.24163862z" fill-rule="evenodd"></path></symbol><symbol id="icon-plus" viewBox="0 0 16 16"><path d="m2.00087166 7h4.99912834v-4.99912834c0-.55276616.44386482-1.00087166 1-1.00087166.55228475 0 1 .44463086 1 1.00087166v4.99912834h4.9991283c.5527662 0 1.0008717.44386482 1.0008717 1 0 .55228475-.4446309 1-1.0008717 1h-4.9991283v4.9991283c0 .5527662-.44386482 1.0008717-1 1.0008717-.55228475 0-1-.4446309-1-1.0008717v-4.9991283h-4.99912834c-.55276616 0-1.00087166-.44386482-1.00087166-1 0-.55228475.44463086-1 1.00087166-1z" fill-rule="evenodd"></path></symbol><symbol id="icon-print" viewBox="0 0 18 18"><path d="m16.0049107 5h-14.00982141c-.54941618 0-.99508929.4467783-.99508929.99961498v6.00077002c0 .5570958.44271433.999615.99508929.999615h1.00491071v-3h12v3h1.0049107c.5494162 0 .9950893-.4467783.9950893-.999615v-6.00077002c0-.55709576-.4427143-.99961498-.9950893-.99961498zm-2.0049107-1v-2.00208688c0-.54777062-.4519464-.99791312-1.0085302-.99791312h-7.9829396c-.55661731 0-1.0085302.44910695-1.0085302.99791312v2.00208688zm1 10v2.0018986c0 1.103521-.9019504 1.9981014-2.0085302 1.9981014h-7.9829396c-1.1092806 0-2.0085302-.8867064-2.0085302-1.9981014v-2.0018986h-1.00491071c-1.10185739 0-1.99508929-.8874333-1.99508929-1.999615v-6.00077002c0-1.10435686.8926228-1.99961498 1.99508929-1.99961498h1.00491071v-2.00208688c0-1.10341695.90195036-1.99791312 2.0085302-1.99791312h7.9829396c1.1092806 0 2.0085302.89826062 2.0085302 1.99791312v2.00208688h1.0049107c1.1018574 0 1.9950893.88743329 1.9950893 1.99961498v6.00077002c0 1.1043569-.8926228 1.999615-1.9950893 1.999615zm-1-3h-10v5.0018986c0 .5546075.44702548.9981014 1.0085302.9981014h7.9829396c.5565964 0 1.0085302-.4491701 1.0085302-.9981014zm-9 1h8v1h-8zm0 2h5v1h-5zm9-5c-.5522847 0-1-.44771525-1-1s.4477153-1 1-1 1 .44771525 1 1-.4477153 1-1 1z" fill-rule="evenodd"></path></symbol><symbol id="icon-search" viewBox="0 0 22 22"><path d="M21.697 20.261a1.028 1.028 0 01.01 1.448 1.034 1.034 0 01-1.448-.01l-4.267-4.267A9.812 9.811 0 010 9.812a9.812 9.811 0 1117.43 6.182zM9.812 18.222A8.41 8.41 0 109.81 1.403a8.41 8.41 0 000 16.82z" fill-rule="evenodd"></path></symbol><symbol id="icon-social-facebook" viewBox="0 0 24 24"><path d="m6.00368507 20c-1.10660471 0-2.00368507-.8945138-2.00368507-1.9940603v-12.01187942c0-1.10128908.89451376-1.99406028 1.99406028-1.99406028h12.01187942c1.1012891 0 1.9940603.89451376 1.9940603 1.99406028v12.01187942c0 1.1012891-.88679 1.9940603-2.0032184 1.9940603h-2.9570132v-6.1960818h2.0797387l.3114113-2.414723h-2.39115v-1.54164807c0-.69911803.1941355-1.1755439 1.1966615-1.1755439l1.2786739-.00055875v-2.15974763l-.2339477-.02492088c-.3441234-.03134957-.9500153-.07025255-1.6293054-.07025255-1.8435726 0-3.1057323 1.12531866-3.1057323 3.19187953v1.78079225h-2.0850778v2.414723h2.0850778v6.1960818z" fill-rule="evenodd"></path></symbol><symbol id="icon-social-twitter" viewBox="0 0 24 24"><path d="m18.8767135 6.87445248c.7638174-.46908424 1.351611-1.21167363 1.6250764-2.09636345-.7135248.43394112-1.50406.74870123-2.3464594.91677702-.6695189-.73342162-1.6297913-1.19486605-2.6922204-1.19486605-2.0399895 0-3.6933555 1.69603749-3.6933555 3.78628909 0 .29642457.0314329.58673729.0942985.8617704-3.06469922-.15890802-5.78835241-1.66547825-7.60988389-3.9574208-.3174714.56076194-.49978171 1.21167363-.49978171 1.90536824 0 1.31404706.65223085 2.47224203 1.64236444 3.15218497-.60350999-.0198635-1.17401554-.1925232-1.67222562-.47366811v.04583885c0 1.83355406 1.27302891 3.36609966 2.96411421 3.71294696-.31118484.0886217-.63651445.1329326-.97441718.1329326-.2357461 0-.47149219-.0229194-.69466516-.0672303.47149219 1.5065703 1.83253297 2.6036468 3.44975116 2.632678-1.2651707 1.0160946-2.85724264 1.6196394-4.5891906 1.6196394-.29861172 0-.59093688-.0152796-.88011875-.0504227 1.63450624 1.0726291 3.57548241 1.6990934 5.66104951 1.6990934 6.79263079 0 10.50641749-5.7711113 10.50641749-10.7751859l-.0094298-.48894775c.7229547-.53478659 1.3516109-1.20250585 1.8419628-1.96190282-.6632323.30100846-1.3751855.50422736-2.1217148.59590507z" fill-rule="evenodd"></path></symbol><symbol id="icon-social-youtube" viewBox="0 0 24 24"><path d="m10.1415 14.3973208-.0005625-5.19318431 4.863375 2.60554491zm9.963-7.92753362c-.6845625-.73643756-1.4518125-.73990314-1.803375-.7826454-2.518875-.18714178-6.2971875-.18714178-6.2971875-.18714178-.007875 0-3.7861875 0-6.3050625.18714178-.352125.04274226-1.1188125.04620784-1.8039375.7826454-.5394375.56084773-.7149375 1.8344515-.7149375 1.8344515s-.18 1.49597903-.18 2.99138042v1.4024082c0 1.495979.18 2.9913804.18 2.9913804s.1755 1.2736038.7149375 1.8344515c.685125.7364376 1.5845625.7133337 1.9850625.7901542 1.44.1420891 6.12.1859866 6.12.1859866s3.78225-.005776 6.301125-.1929178c.3515625-.0433198 1.1188125-.0467854 1.803375-.783223.5394375-.5608477.7155-1.8344515.7155-1.8344515s.18-1.4954014.18-2.9913804v-1.4024082c0-1.49540139-.18-2.99138042-.18-2.99138042s-.1760625-1.27360377-.7155-1.8344515z" fill-rule="evenodd"></path></symbol><symbol id="icon-subject-medicine" viewBox="0 0 18 18"><path d="m12.5 8h-6.5c-1.65685425 0-3 1.34314575-3 3v1c0 1.6568542 1.34314575 3 3 3h1v-2h-.5c-.82842712 0-1.5-.6715729-1.5-1.5s.67157288-1.5 1.5-1.5h1.5 2 1 2c1.6568542 0 3-1.34314575 3-3v-1c0-1.65685425-1.3431458-3-3-3h-2v2h1.5c.8284271 0 1.5.67157288 1.5 1.5s-.6715729 1.5-1.5 1.5zm-5.5-1v-1h-3.5c-1.38071187 0-2.5-1.11928813-2.5-2.5s1.11928813-2.5 2.5-2.5h1.02786405c.46573528 0 .92507448.10843528 1.34164078.31671843l1.13382424.56691212c.06026365-1.05041141.93116291-1.88363055 1.99667093-1.88363055 1.1045695 0 2 .8954305 2 2h2c2.209139 0 4 1.790861 4 4v1c0 2.209139-1.790861 4-4 4h-2v1h2c1.1045695 0 2 .8954305 2 2s-.8954305 2-2 2h-2c0 1.1045695-.8954305 2-2 2s-2-.8954305-2-2h-1c-2.209139 0-4-1.790861-4-4v-1c0-2.209139 1.790861-4 4-4zm0-2v-2.05652691c-.14564246-.03538148-.28733393-.08714006-.42229124-.15461871l-1.15541752-.57770876c-.27771087-.13885544-.583937-.21114562-.89442719-.21114562h-1.02786405c-.82842712 0-1.5.67157288-1.5 1.5s.67157288 1.5 1.5 1.5zm4 1v1h1.5c.2761424 0 .5-.22385763.5-.5s-.2238576-.5-.5-.5zm-1 1v-5c0-.55228475-.44771525-1-1-1s-1 .44771525-1 1v5zm-2 4v5c0 .5522847.44771525 1 1 1s1-.4477153 1-1v-5zm3 2v2h2c.5522847 0 1-.4477153 1-1s-.4477153-1-1-1zm-4-1v-1h-.5c-.27614237 0-.5.2238576-.5.5s.22385763.5.5.5zm-3.5-9h1c.27614237 0 .5.22385763.5.5s-.22385763.5-.5.5h-1c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5z" fill-rule="evenodd"></path></symbol><symbol id="icon-success" viewBox="0 0 18 18"><path d="m9 0c4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9zm3.4860198 4.98163161-4.71802968 5.50657859-2.62834168-2.02300024c-.42862421-.36730544-1.06564993-.30775346-1.42283677.13301307-.35718685.44076653-.29927542 1.0958383.12934879 1.46314377l3.40735508 2.7323063c.42215801.3385221 1.03700951.2798252 1.38749189-.1324571l5.38450527-6.33394549c.3613513-.43716226.3096573-1.09278382-.115462-1.46437175-.4251192-.37158792-1.0626796-.31842941-1.4240309.11873285z" fill-rule="evenodd"></path></symbol><symbol id="icon-table" viewBox="0 0 18 18"><path d="m16.0049107 2c1.1018574 0 1.9950893.89706013 1.9950893 2.00585866v9.98828264c0 1.1078052-.8926228 2.0058587-1.9950893 2.0058587l-4.0059107-.001.001.001h-1l-.001-.001h-5l.001.001h-1l-.001-.001-3.00391071.001c-1.10185739 0-1.99508929-.8970601-1.99508929-2.0058587v-9.98828264c0-1.10780515.8926228-2.00585866 1.99508929-2.00585866zm-11.0059107 5h-3.999v6.9941413c0 .5572961.44630695 1.0058587.99508929 1.0058587h3.00391071zm6 0h-5v8h5zm5.0059107-4h-4.0059107v3h5.001v1h-5.001v7.999l4.0059107.001c.5487152 0 .9950893-.4488783.9950893-1.0058587v-9.98828264c0-.55729607-.446307-1.00585866-.9950893-1.00585866zm-12.5049107 9c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-1c-.27614237 0-.5-.2238576-.5-.5s.22385763-.5.5-.5zm12 0c.2761424 0 .5.2238576.5.5s-.2238576.5-.5.5h-2c-.2761424 0-.5-.2238576-.5-.5s.2238576-.5.5-.5zm-6 0c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-2c-.27614237 0-.5-.2238576-.5-.5s.22385763-.5.5-.5zm-6-2c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-1c-.27614237 0-.5-.2238576-.5-.5s.22385763-.5.5-.5zm12 0c.2761424 0 .5.2238576.5.5s-.2238576.5-.5.5h-2c-.2761424 0-.5-.2238576-.5-.5s.2238576-.5.5-.5zm-6 0c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-2c-.27614237 0-.5-.2238576-.5-.5s.22385763-.5.5-.5zm-6-2c.27614237 0 .5.22385763.5.5s-.22385763.5-.5.5h-1c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5zm12 0c.2761424 0 .5.22385763.5.5s-.2238576.5-.5.5h-2c-.2761424 0-.5-.22385763-.5-.5s.2238576-.5.5-.5zm-6 0c.27614237 0 .5.22385763.5.5s-.22385763.5-.5.5h-2c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5zm1.499-5h-5v3h5zm-6 0h-3.00391071c-.54871518 0-.99508929.44887827-.99508929 1.00585866v1.99414134h3.999z" fill-rule="evenodd"></path></symbol><symbol id="icon-tick-circle" viewBox="0 0 24 24"><path d="m12 2c5.5228475 0 10 4.4771525 10 10s-4.4771525 10-10 10-10-4.4771525-10-10 4.4771525-10 10-10zm0 1c-4.97056275 0-9 4.02943725-9 9 0 4.9705627 4.02943725 9 9 9 4.9705627 0 9-4.0294373 9-9 0-4.97056275-4.0294373-9-9-9zm4.2199868 5.36606669c.3613514-.43716226.9989118-.49032077 1.424031-.11873285s.4768133 1.02720949.115462 1.46437175l-6.093335 6.94397871c-.3622945.4128716-.9897871.4562317-1.4054264.0971157l-3.89719065-3.3672071c-.42862421-.3673054-.48653564-1.0223772-.1293488-1.4631437s.99421256-.5003185 1.42283677-.1330131l3.11097438 2.6987741z" fill-rule="evenodd"></path></symbol><symbol id="icon-tick" viewBox="0 0 16 16"><path d="m6.76799012 9.21106946-3.1109744-2.58349728c-.42862421-.35161617-1.06564993-.29460792-1.42283677.12733148s-.29927541 1.04903009.1293488 1.40064626l3.91576307 3.23873978c.41034319.3393961 1.01467563.2976897 1.37450571-.0948578l6.10568327-6.660841c.3613513-.41848908.3096572-1.04610608-.115462-1.4018218-.4251192-.35571573-1.0626796-.30482786-1.424031.11366122z" fill-rule="evenodd"></path></symbol><symbol id="icon-update" viewBox="0 0 18 18"><path d="m1 13v1c0 .5522847.44771525 1 1 1h14c.5522847 0 1-.4477153 1-1v-1h-1v-10h-14v10zm16-1h1v2c0 1.1045695-.8954305 2-2 2h-14c-1.1045695 0-2-.8954305-2-2v-2h1v-9c0-.55228475.44771525-1 1-1h14c.5522847 0 1 .44771525 1 1zm-1 0v1h-4.5857864l-1 1h-2.82842716l-1-1h-4.58578644v-1h5l1 1h2l1-1zm-13-8h12v7h-12zm1 1v5h10v-5zm1 1h4v1h-4zm0 2h4v1h-4z" fill-rule="evenodd"></path></symbol><symbol id="icon-upload" viewBox="0 0 18 18"><path d="m10.0046024 0c.5497429 0 1.3179837.32258606 1.707238.71184039l4.5763192 4.57631922c.3931386.39313859.7118404 1.16760135.7118404 1.71431368v8.98899651c0 1.1092806-.8945138 2.0085302-1.9940603 2.0085302h-12.01187942c-1.10128908 0-1.99406028-.8926228-1.99406028-1.9950893v-14.00982141c0-1.10185739.88743329-1.99508929 1.99961498-1.99508929zm0 1h-7.00498742c-.55709576 0-.99961498.44271433-.99961498.99508929v14.00982141c0 .5500396.44491393.9950893.99406028.9950893h12.01187942c.5463747 0 .9940603-.4506622.9940603-1.0085302v-8.98899651c0-.28393444-.2150684-.80332809-.4189472-1.0072069l-4.5763192-4.57631922c-.2038461-.20384606-.718603-.41894717-1.0001312-.41894717zm-1.85576936 4.14572769c.19483374-.19483375.51177826-.19377714.70556874.00001334l2.59099082 2.59099079c.1948411.19484112.1904373.51514474.0027906.70279143-.1932998.19329987-.5046517.19237083-.7001856-.00692852l-1.74638687-1.7800176v6.14827687c0 .2717771-.23193359.492096-.5.492096-.27614237 0-.5-.216372-.5-.492096v-6.14827641l-1.74627892 1.77990922c-.1933927.1971171-.51252214.19455839-.70016883.0069117-.19329987-.19329988-.19100584-.50899493.00277731-.70277808z" fill-rule="evenodd"></path></symbol><symbol id="icon-video" viewBox="0 0 18 18"><path d="m16.0049107 2c1.1018574 0 1.9950893.89706013 1.9950893 2.00585866v9.98828264c0 1.1078052-.8926228 2.0058587-1.9950893 2.0058587h-14.00982141c-1.10185739 0-1.99508929-.8970601-1.99508929-2.0058587v-9.98828264c0-1.10780515.8926228-2.00585866 1.99508929-2.00585866zm0 1h-14.00982141c-.54871518 0-.99508929.44887827-.99508929 1.00585866v9.98828264c0 .5572961.44630695 1.0058587.99508929 1.0058587h14.00982141c.5487152 0 .9950893-.4488783.9950893-1.0058587v-9.98828264c0-.55729607-.446307-1.00585866-.9950893-1.00585866zm-8.30912922 2.24944486 4.60460462 2.73982242c.9365543.55726659.9290753 1.46522435 0 2.01804082l-4.60460462 2.7398224c-.93655425.5572666-1.69578148.1645632-1.69578148-.8937585v-5.71016863c0-1.05087579.76670616-1.446575 1.69578148-.89375851zm-.67492769.96085624v5.5750128c0 .2995102-.10753745.2442517.16578928.0847713l4.58452283-2.67497259c.3050619-.17799716.3051624-.21655446 0-.39461026l-4.58452283-2.67497264c-.26630747-.15538481-.16578928-.20699944-.16578928.08477139z" fill-rule="evenodd"></path></symbol><symbol id="icon-warning" viewBox="0 0 18 18"><path d="m9 11.75c.69035594 0 1.25.5596441 1.25 1.25s-.55964406 1.25-1.25 1.25-1.25-.5596441-1.25-1.25.55964406-1.25 1.25-1.25zm.41320045-7.75c.55228475 0 1.00000005.44771525 1.00000005 1l-.0034543.08304548-.3333333 4c-.043191.51829212-.47645714.91695452-.99654578.91695452h-.15973424c-.52008864 0-.95335475-.3986624-.99654576-.91695452l-.33333333-4c-.04586475-.55037702.36312325-1.03372649.91350028-1.07959124l.04148683-.00259031zm-.41320045 14c-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9 4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9z" fill-rule="evenodd"></path></symbol><symbol id="icon-facebook-bordered" viewBox="463.812 263.868 32 32"><path d="M479.812,263.868c-8.837,0-16,7.163-16,16s7.163,16,16,16s16-7.163,16-16S488.649,263.868,479.812,263.868z
	 M479.812,293.868c-7.732,0-14-6.269-14-14s6.268-14,14-14s14,6.269,14,14S487.545,293.868,479.812,293.868z"></path><path d="M483.025,280.48l0.32-2.477h-2.453v-1.582c0-0.715,0.199-1.207,1.227-1.207h1.311v-2.213
	c-0.227-0.029-1.003-0.098-1.907-0.098c-1.894,0-3.186,1.154-3.186,3.271v1.826h-2.142v2.477h2.142v6.354h2.557v-6.354
	L483.025,280.48L483.025,280.48z"></path></symbol><symbol id="icon-twitter-bordered" viewBox="463.812 263.868 32 32"><g><path d="M486.416,276.191c-0.483,0.215-1.007,0.357-1.554,0.429c0.558-0.338,0.991-0.868,1.19-1.502
		c-0.521,0.308-1.104,0.536-1.72,0.657c-0.494-0.526-1.2-0.854-1.979-0.854c-1.496,0-2.711,1.213-2.711,2.71
		c0,0.212,0.023,0.419,0.069,0.616c-2.252-0.111-4.25-1.19-5.586-2.831c-0.231,0.398-0.365,0.866-0.365,1.361
		c0,0.94,0.479,1.772,1.204,2.257c-0.441-0.015-0.861-0.138-1.227-0.339v0.031c0,1.314,0.937,2.41,2.174,2.656
		c-0.227,0.062-0.47,0.098-0.718,0.098c-0.171,0-0.343-0.018-0.511-0.049c0.35,1.074,1.347,1.859,2.531,1.883
		c-0.928,0.726-2.095,1.16-3.366,1.16c-0.22,0-0.433-0.014-0.644-0.037c1.2,0.768,2.621,1.215,4.155,1.215
		c4.983,0,7.71-4.129,7.71-7.711c0-0.115-0.004-0.232-0.006-0.351C485.592,277.212,486.054,276.734,486.416,276.191z"></path></g><path d="M479.812,263.868c-8.837,0-16,7.163-16,16s7.163,16,16,16s16-7.163,16-16S488.649,263.868,479.812,263.868z
	 M479.812,293.868c-7.732,0-14-6.269-14-14s6.268-14,14-14s14,6.269,14,14S487.545,293.868,479.812,293.868z"></path></symbol><symbol id="icon-weibo-bordered" viewBox="463.812 263.868 32 32"><path d="M479.812,263.868c-8.838,0-16,7.163-16,16s7.162,16,16,16c8.837,0,16-7.163,16-16S488.649,263.868,479.812,263.868z
	 M479.812,293.868c-7.732,0-14-6.269-14-14s6.268-14,14-14c7.731,0,14,6.269,14,14S487.545,293.868,479.812,293.868z"></path><g><path d="M478.552,285.348c-2.616,0.261-4.876-0.926-5.044-2.649c-0.167-1.722,1.814-3.33,4.433-3.588
		c2.609-0.263,4.871,0.926,5.041,2.647C483.147,283.479,481.164,285.089,478.552,285.348 M483.782,279.63
		c-0.226-0.065-0.374-0.109-0.259-0.403c0.25-0.639,0.276-1.188,0.005-1.581c-0.515-0.734-1.915-0.693-3.521-0.021
		c0,0-0.508,0.224-0.378-0.181c0.247-0.798,0.209-1.468-0.178-1.852c-0.87-0.878-3.194,0.032-5.183,2.027
		c-1.489,1.494-2.357,3.082-2.357,4.453c0,2.619,3.354,4.213,6.631,4.213c4.297,0,7.154-2.504,7.154-4.493
		C485.697,280.594,484.689,279.911,483.782,279.63"></path><path d="M486.637,274.833c-1.039-1.154-2.57-1.592-3.982-1.291l0,0c-0.325,0.068-0.532,0.391-0.465,0.72
		c0.068,0.328,0.391,0.537,0.72,0.466c1.005-0.215,2.092,0.104,2.827,0.92c0.736,0.818,0.938,1.939,0.625,2.918l0,0
		c-0.102,0.318,0.068,0.661,0.39,0.762c0.32,0.104,0.658-0.069,0.763-0.391v-0.001C487.953,277.558,487.674,275.985,486.637,274.833
		"></path><path d="M485.041,276.276c-0.504-0.562-1.25-0.774-1.938-0.63c-0.279,0.06-0.461,0.339-0.396,0.621
		c0.062,0.281,0.335,0.461,0.617,0.398l0,0c0.336-0.071,0.702,0.03,0.947,0.307s0.312,0.649,0.207,0.979l0,0
		c-0.089,0.271,0.062,0.565,0.336,0.654c0.274,0.09,0.564-0.062,0.657-0.336C485.686,277.604,485.549,276.837,485.041,276.276"></path><path d="M478.694,282.227c-0.09,0.156-0.293,0.233-0.451,0.166c-0.151-0.062-0.204-0.235-0.115-0.389
		c0.093-0.155,0.284-0.229,0.44-0.168C478.725,281.892,478.782,282.071,478.694,282.227 M477.862,283.301
		c-0.253,0.405-0.795,0.58-1.202,0.396c-0.403-0.186-0.521-0.655-0.27-1.051c0.248-0.39,0.771-0.566,1.176-0.393
		C477.979,282.423,478.109,282.889,477.862,283.301 M478.812,280.437c-1.244-0.326-2.65,0.294-3.19,1.396
		c-0.553,1.119-0.021,2.369,1.236,2.775c1.303,0.42,2.84-0.225,3.374-1.436C480.758,281.989,480.1,280.77,478.812,280.437"></path></g></symbol><symbol id="icon-arrow-left-bullet" viewBox="0 0 8 16"><path d="M3 8l5 5v3L0 8l8-8v3L3 8z"></path></symbol><symbol id="icon-chevron-down" viewBox="0 0 16 16"><path d="m5.58578644 3-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4c-.39052429.39052429-1.02368927.39052429-1.41421356 0s-.39052429-1.02368927 0-1.41421356z" fill-rule="evenodd" transform="matrix(0 1 -1 0 11 3)"></path></symbol><symbol id="icon-download-rounded"><path d="M0 13c0-.556.449-1 1.002-1h9.996a.999.999 0 110 2H1.002A1.006 1.006 0 010 13zM7 1v6.8l2.482-2.482c.392-.392 1.022-.4 1.403-.02a1.001 1.001 0 010 1.417l-4.177 4.177a1.001 1.001 0 01-1.416 0L1.115 6.715a.991.991 0 01-.016-1.4 1 1 0 011.42.003L5 7.8V1c0-.55.444-.996 1-.996.552 0 1 .445 1 .996z"></path></symbol><symbol id="icon-ext-link" viewBox="0 0 16 16"><path d="M12.9 16H3.1C1.4 16 0 14.6 0 12.9V3.2C0 1.4 1.4 0 3.1 0h3.7v1H3.1C2 1 1 2 1 3.2v9.7C1 14 2 15 3.1 15h9.7c1.2 0 2.1-1 2.1-2.1V8.7h1v4.2c.1 1.7-1.3 3.1-3 3.1z"></path><path d="M12.8 2.5l.7.7-9 8.9-.7-.7 9-8.9z"></path><path d="M9.7 0L16 6.2V0z"></path></symbol><symbol id="icon-remove" viewBox="-296 388 18 18"><path d="M-291.7 396.1h9v2h-9z"></path><path d="M-287 405.5c-4.7 0-8.5-3.8-8.5-8.5s3.8-8.5 8.5-8.5 8.5 3.8 8.5 8.5-3.8 8.5-8.5 8.5zm0-16c-4.1 0-7.5 3.4-7.5 7.5s3.4 7.5 7.5 7.5 7.5-3.4 7.5-7.5-3.4-7.5-7.5-7.5z"></path></symbol><symbol id="icon-rss" viewBox="0 0 18 18"><path d="m.97480857 6.01583891.11675372.00378391c5.75903295.51984988 10.34261021 5.10537458 10.85988231 10.86480098.0494035.5500707-.3564674 1.0360406-.906538 1.0854441-.5500707.0494036-1.0360406-.3564673-1.08544412-.906538-.43079083-4.7965248-4.25151132-8.61886853-9.04770289-9.05180573-.55004837-.04965115-.95570047-.53580366-.90604933-1.08585203.04610464-.5107592.46858035-.89701345.96909831-.90983323zm1.52519143 6.95474179c1.38071187 0 2.5 1.1192881 2.5 2.5s-1.11928813 2.5-2.5 2.5-2.5-1.1192881-2.5-2.5 1.11928813-2.5 2.5-2.5zm-1.43253846-12.96884168c9.09581416.53242539 16.37540296 7.8163886 16.90205336 16.91294558.0319214.5513615-.389168 1.0242056-.9405294 1.056127-.5513615.0319214-1.0242057-.389168-1.0561271-.9405294-.4679958-8.08344784-6.93949306-14.55883389-15.02226722-15.03196077-.55134101-.03227286-.97212889-.50538538-.93985602-1.05672639.03227286-.551341.50538538-.97212888 1.05672638-.93985602z" fill-rule="evenodd"></path></symbol><symbol id="icon-springer-arrow-left"><path d="M15 7a1 1 0 000-2H3.385l2.482-2.482a.994.994 0 00.02-1.403 1.001 1.001 0 00-1.417 0L.294 5.292a1.001 1.001 0 000 1.416l4.176 4.177a.991.991 0 001.4.016 1 1 0 00-.003-1.42L3.385 7H15z"></path></symbol><symbol id="icon-springer-arrow-right"><path d="M1 7a1 1 0 010-2h11.615l-2.482-2.482a.994.994 0 01-.02-1.403 1.001 1.001 0 011.417 0l4.176 4.177a1.001 1.001 0 010 1.416l-4.176 4.177a.991.991 0 01-1.4.016 1 1 0 01.003-1.42L12.615 7H1z"></path></symbol><symbol id="icon-springer-collections" viewBox="3 3 32 32"><path fill-rule="evenodd" d="M25.583333,30.1249997 L25.583333,7.1207574 C25.583333,7.10772495 25.579812,7.10416665 25.5859851,7.10416665 L6.10400517,7.10571021 L6.10400517,30.1355179 C6.10400517,31.1064087 6.89406744,31.8958329 7.86448169,31.8958329 L26.057145,31.8958329 C25.7558021,31.374901 25.583333,30.7700915 25.583333,30.1249997 Z M4.33333333,30.1355179 L4.33333333,7.10571021 C4.33333333,6.12070047 5.12497502,5.33333333 6.10151452,5.33333333 L25.5859851,5.33333333 C26.5617372,5.33333333 27.3541664,6.13359035 27.3541664,7.1207574 L27.3541664,12.4166666 L32.6666663,12.4166666 L32.6666663,30.1098941 C32.6666663,32.0694626 31.0857174,33.6666663 29.1355179,33.6666663 L7.86448169,33.6666663 C5.91736809,33.6666663 4.33333333,32.0857174 4.33333333,30.1355179 Z M27.3541664,14.1874999 L27.3541664,30.1249997 C27.3541664,31.1030039 28.1469954,31.8958329 29.1249997,31.8958329 C30.1030039,31.8958329 30.8958329,31.1030039 30.8958329,30.1249997 L30.8958329,14.1874999 L27.3541664,14.1874999 Z M9.64583326,10.6458333 L22.0416665,10.6458333 L22.0416665,17.7291665 L9.64583326,17.7291665 L9.64583326,10.6458333 Z M11.4166666,12.4166666 L11.4166666,15.9583331 L20.2708331,15.9583331 L20.2708331,12.4166666 L11.4166666,12.4166666 Z M9.64583326,19.4999998 L22.0416665,19.4999998 L22.0416665,21.2708331 L9.64583326,21.2708331 L9.64583326,19.4999998 Z M9.64583326,23.0416665 L22.0416665,23.0416665 L22.0416665,24.8124997 L9.64583326,24.8124997 L9.64583326,23.0416665 Z M9.64583326,26.583333 L22.0416665,26.583333 L22.0416665,28.3541664 L9.64583326,28.3541664 L9.64583326,26.583333 Z"></path></symbol><symbol id="icon-springer-download" viewBox="-301 390 9 14"><path d="M-301 395.6l4.5 5.1 4.5-5.1h-3V390h-3v5.6h-3zm0 6.5h9v1.9h-9z"></path></symbol><symbol id="icon-springer-info" viewBox="0 0 24 24"><!--Generator: Sketch 63.1 (92452) - https://sketch.com--><g id="V&amp;I" stroke="none" stroke-width="1" fill-rule="evenodd"><g id="info" fill-rule="nonzero"><path d="M12,0 C18.627417,0 24,5.372583 24,12 C24,18.627417 18.627417,24 12,24 C5.372583,24 0,18.627417 0,12 C0,5.372583 5.372583,0 12,0 Z M12.5540543,9.1 L11.5540543,9.1 C11.0017696,9.1 10.5540543,9.54771525 10.5540543,10.1 L10.5540543,10.1 L10.5540543,18.1 C10.5540543,18.6522847 11.0017696,19.1 11.5540543,19.1 L11.5540543,19.1 L12.5540543,19.1 C13.1063391,19.1 13.5540543,18.6522847 13.5540543,18.1 L13.5540543,18.1 L13.5540543,10.1 C13.5540543,9.54771525 13.1063391,9.1 12.5540543,9.1 L12.5540543,9.1 Z M12,5 C11.5356863,5 11.1529412,5.14640523 10.8517647,5.43921569 C10.5505882,5.73202614 10.4,6.11546841 10.4,6.58954248 C10.4,7.06361656 10.5505882,7.45054466 10.8517647,7.7503268 C11.1529412,8.05010893 11.5356863,8.2 12,8.2 C12.4768627,8.2 12.8627451,8.05010893 13.1576471,7.7503268 C13.452549,7.45054466 13.6,7.06361656 13.6,6.58954248 C13.6,6.11546841 13.452549,5.73202614 13.1576471,5.43921569 C12.8627451,5.14640523 12.4768627,5 12,5 Z" id="Combined-Shape"></path></g></g></symbol><symbol id="icon-springer-tick-circle" viewBox="0 0 24 24"><g id="Page-1" stroke="none" stroke-width="1" fill-rule="evenodd"><g id="springer-tick-circle" fill-rule="nonzero"><path d="M12,24 C5.372583,24 0,18.627417 0,12 C0,5.372583 5.372583,0 12,0 C18.627417,0 24,5.372583 24,12 C24,18.627417 18.627417,24 12,24 Z M7.657,10.79 C7.45285634,10.6137568 7.18569967,10.5283283 6.91717333,10.5534259 C6.648647,10.5785236 6.40194824,10.7119794 6.234,10.923 C5.87705269,11.3666969 5.93445559,12.0131419 6.364,12.387 L10.261,15.754 C10.6765468,16.112859 11.3037113,16.0695601 11.666,15.657 L17.759,8.713 C18.120307,8.27302248 18.0695334,7.62621189 17.644,7.248 C17.4414817,7.06995024 17.1751516,6.9821166 16.9064461,7.00476032 C16.6377406,7.02740404 16.3898655,7.15856958 16.22,7.368 L10.768,13.489 L7.657,10.79 Z" id="path-1"></path></g></g></symbol><symbol id="icon-updates" viewBox="0 0 18 18"><g fill-rule="nonzero"><path d="M16.98 3.484h-.48c-2.52-.058-5.04 1.161-7.44 2.903-2.46-1.8-4.74-2.903-8.04-2.903-.3 0-.54.29-.54.58v9.813c0 .29.24.523.54.581 2.76.348 4.86 1.045 7.62 2.903.24.116.54.116.72 0 2.76-1.858 4.86-2.555 7.62-2.903.3-.058.54-.29.54-.58V4.064c0-.29-.24-.523-.54-.581zm-15.3 1.22c2.34 0 4.86 1.509 6.72 2.786v8.478c-2.34-1.394-4.38-2.09-6.72-2.439V4.703zm14.58 8.767c-2.34.348-4.38 1.045-6.72 2.439V7.374C12 5.632 14.1 4.645 16.26 4.645v8.826z"></path><path d="M9 .058c-1.56 0-2.76 1.22-2.76 2.671C6.24 4.181 7.5 5.4 9 5.4c1.5 0 2.76-1.22 2.76-2.671 0-1.452-1.2-2.67-2.76-2.67zm0 4.413c-.96 0-1.8-.755-1.8-1.742C7.2 1.8 7.98.987 9 .987s1.8.755 1.8 1.742c0 .93-.84 1.742-1.8 1.742z"></path></g></symbol><symbol id="icon-checklist-banner" viewBox="0 0 56.69 56.69"><style type="text/css">.st0{fill:none;}
	.st1{clip-path:url(#SVGID_2_);fill:none;stroke:#01324B;stroke-width:2;stroke-linecap:round;}
	.st2{clip-path:url(#SVGID_2_);fill:none;stroke:#01324B;stroke-width:2;stroke-linecap:round;stroke-linejoin:round;}</style><rect id="transparent_background" class="st0" width="56.69" height="56.69"></rect><g><clippath id="SVGID_2_"><use xlink:href="#SVGID_1_" style="overflow:visible"></use></clippath><path class="st1" d="M21.14,34.46c0-6.77,5.48-12.26,12.24-12.26c6.76,0,12.24,5.49,12.24,12.26c0,6.77-5.48,12.26-12.24,12.26
		C26.62,46.71,21.14,41.23,21.14,34.46z M40.47,45.12l10.23,9.22c0,0,1.21,1.09,2.3-0.12l2.09-2.32c0,0,1.09-1.21-0.12-2.3
		l-10.23-9.22 M25.45,34.46c0-4.38,3.55-7.94,7.93-7.94c4.38,0,7.93,3.55,7.93,7.94c0,4.38-3.55,7.94-7.93,7.94
		C29,42.39,25.45,38.84,25.45,34.46z M43.03,47.45l4.14-4.81"></path><path class="st2" d="M8.26,9.75H28.6 M8.26,15.98H28.6 M8.26,22.18h12.5 M35.18,16.98V4.86c0,0,0-2.93-2.93-2.93H4.13
		c0,0-2.93,0-2.93,2.93v37.57c0,0,0,2.93,2.93,2.93h15.01 M8.26,9.75H28.6 M8.26,15.98H28.6 M8.26,22.18h12.5"></path></g><g></g><g></g><g></g><g></g><g></g><g></g></symbol><symbol id="icon-submit-closed" viewBox="0 0 18 18"><path d="m15 0c1.1045695 0 2 .8954305 2 2v4.5c0 .27614237-.2238576.5-.5.5s-.5-.22385763-.5-.5v-4.5c0-.51283584-.3860402-.93550716-.8833789-.99327227l-.1166211-.00672773h-9v3c0 1.1045695-.8954305 2-2 2h-3v10c0 .5128358.38604019.9355072.88337887.9932723l.11662113.0067277h4.5c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-4.5c-1.1045695 0-2-.8954305-2-2v-10.17157288c0-.53043297.21071368-1.0391408.58578644-1.41421356l3.82842712-3.82842712c.37507276-.37507276.88378059-.58578644 1.41421356-.58578644zm-2.5 7c3.0375661 0 5.5 2.46243388 5.5 5.5 0 3.0375661-2.4624339 5.5-5.5 5.5-3.03756612 0-5.5-2.4624339-5.5-5.5 0-3.03756612 2.46243388-5.5 5.5-5.5zm0 1c-2.4852814 0-4.5 2.0147186-4.5 4.5s2.0147186 4.5 4.5 4.5 4.5-2.0147186 4.5-4.5-2.0147186-4.5-4.5-4.5zm2.3087379 2.1912621c.2550161.2550162.2550161.668479 0 .9234952l-1.3859024 1.3845831 1.3859024 1.3859023c.2550161.2550162.2550161.668479 0 .9234952-.2550162.2550161-.668479.2550161-.9234952 0l-1.3859023-1.3859024-1.3845831 1.3859024c-.2550162.2550161-.668479.2550161-.9234952 0-.25501614-.2550162-.25501614-.668479 0-.9234952l1.3845831-1.3859023-1.3845831-1.3845831c-.25501614-.2550162-.25501614-.668479 0-.9234952.2550162-.25501614.668479-.25501614.9234952 0l1.3845831 1.3845831 1.3859023-1.3845831c.2550162-.25501614.668479-.25501614.9234952 0zm-9.8087379-8.7782621-3.587 3.587h2.587c.55228475 0 1-.44771525 1-1z"></path></symbol><symbol id="icon-submit-open" viewBox="0 0 18 18"><path d="m15 0c1.1045695 0 2 .8954305 2 2v5.5c0 .27614237-.2238576.5-.5.5s-.5-.22385763-.5-.5v-5.5c0-.51283584-.3860402-.93550716-.8833789-.99327227l-.1166211-.00672773h-9v3c0 1.1045695-.8954305 2-2 2h-3v10c0 .5128358.38604019.9355072.88337887.9932723l.11662113.0067277h7.5c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-7.5c-1.1045695 0-2-.8954305-2-2v-10.17157288c0-.53043297.21071368-1.0391408.58578644-1.41421356l3.82842712-3.82842712c.37507276-.37507276.88378059-.58578644 1.41421356-.58578644zm-.5442863 8.18867991 3.3545404 3.35454039c.2508994.2508994.2538696.6596433.0035959.909917-.2429543.2429542-.6561449.2462671-.9065387-.0089489l-2.2609825-2.3045251.0010427 7.2231989c0 .3569916-.2898381.6371378-.6473715.6371378-.3470771 0-.6473715-.2852563-.6473715-.6371378l-.0010428-7.2231995-2.2611222 2.3046654c-.2531661.2580415-.6562868.2592444-.9065605.0089707-.24295423-.2429542-.24865597-.6576651.0036132-.9099343l3.3546673-3.35466731c.2509089-.25090888.6612706-.25227691.9135302-.00001728zm-.9557137-3.18867991c.2761424 0 .5.22385763.5.5s-.2238576.5-.5.5h-6c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5zm-8.5-3.587-3.587 3.587h2.587c.55228475 0 1-.44771525 1-1zm8.5 1.587c.2761424 0 .5.22385763.5.5s-.2238576.5-.5.5h-6c-.27614237 0-.5-.22385763-.5-.5s.22385763-.5.5-.5z"></path></symbol><symbol id="icon-submit-upcoming" viewBox="0 0 18 18"><path d="m15 0c1.1045695 0 2 .8954305 2 2v4.5c0 .27614237-.2238576.5-.5.5s-.5-.22385763-.5-.5v-4.5c0-.51283584-.3860402-.93550716-.8833789-.99327227l-.1166211-.00672773h-9v3c0 1.1045695-.8954305 2-2 2h-3v10c0 .5128358.38604019.9355072.88337887.9932723l.11662113.0067277h4.5c.27614237 0 .5.2238576.5.5s-.22385763.5-.5.5h-4.5c-1.1045695 0-2-.8954305-2-2v-10.17157288c0-.53043297.21071368-1.0391408.58578644-1.41421356l3.82842712-3.82842712c.37507276-.37507276.88378059-.58578644 1.41421356-.58578644zm-2.5 7c3.0375661 0 5.5 2.46243388 5.5 5.5 0 3.0375661-2.4624339 5.5-5.5 5.5-3.03756612 0-5.5-2.4624339-5.5-5.5 0-1.6607442.73606908-3.14957021 1.89976608-4.15803695l-1.51549374.02214397c-.27613212.00263356-.49998143-.22483432-.49998143-.49020681 0-.24299316.17766103-.44509007.40961587-.48700057l.08928713-.00797472h2.66407569c.2449213 0 .4486219.17766776.490865.40963137l.008038.08929051v2.6642143c0 .275547-.2296028.4989219-.4949753.4989219-.24299317 0-.44342617-.1744719-.4830969-.4093269l-.00710993-.0906783.01983146-1.46576707c-.96740882.82538117-1.58082193 2.05345007-1.58082193 3.42478927 0 2.4852814 2.0147186 4.5 4.5 4.5s4.5-2.0147186 4.5-4.5-2.0147186-4.5-4.5-4.5c-.7684937 0-.7684937-1 0-1zm0 2.85c.3263501 0 .5965265.2405082.6429523.5539478l.0070477.0960522v1.731l.8096194.8093806c.2284567.2284567.2513024.5846637.068537.8386705l-.068537.0805683c-.2284567.2284567-.5846637.2513024-.8386705.068537l-.0805683-.068537-.9707107-.9707107c-.1125218-.1125218-.1855975-.257116-.2103268-.412296l-.0093431-.1180341v-1.9585786c0-.3589851.2910149-.65.65-.65zm-7.5-8.437-3.587 3.587h2.587c.55228475 0 1-.44771525 1-1z"></path></symbol></svg>
</div><iframe id="canary-tool-tip" frameborder="0" class="fat-sapphire-swordtail-U1xAEvvLch4FZ9B40MDPpa-tool-tip" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/mo0yqyx4jxm3k82fauc54s.html" style="display: none;"></iframe><div class="sleepy-harlequin-emu-dhxjcuNLidDBYMxjgfr8 fat-sapphire-swordtail-U1xAEvvLch4FZ9B40MDPpa-iframe-container" id="tame-white-basilisk-thUy9dLE2S4z1QHGeni"><iframe class="sleepy-harlequin-emu-dhxjcuNLidDBYMxjgfr8 fat-sapphire-swordtail-U1xAEvvLch4FZ9B40MDPpa-iframe-styles" id="homeless-red-skunk-9CrW3KFrxSP4HSeLL1" frameborder="0" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/inawfxwq9csbd0lqx1wjk6.html"></iframe></div>


        

        

    <div class="u-vh-full">
        <a class="c-skip-link" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#main-content">Skip to main content</a>
        
            
    
        <div class="u-hide u-show-following-ad"></div>
    
    <aside class="adsbox c-ad c-ad--728x90" data-component-mpu="">
        <div class="c-ad__inner">
            <p class="c-ad__label">Advertisement</p>
            <div id="div-gpt-ad-LB1" data-ad-type="LB1" data-test="LB1-ad" data-pa11y-ignore="" data-gpt="" data-gpt-unitpath="/270604982/bmc/genomebiology/articles" data-gpt-sizes="728x90,970x90" data-gpt-targeting="pos=LB1;doi=10.1186/gb-2009-10-3-r25;type=article;kwrd=Additional Data File,Central Processing Unit,Human Reference Genome,Memory Footprint,NCBI Short Read Archive;pmc=L32030,B12008,L32020,L32010,L15001,L21001;sponsored=TwentyGB;">
                <noscript>
                    <a href="//pubads.g.doubleclick.net/gampad/jump?iu=/270604982/bmc/genomebiology/articles&amp;sz=728x90,970x90&amp;pos=LB1&amp;doi=10.1186/gb-2009-10-3-r25&amp;type=article&amp;kwrd=Additional Data File,Central Processing Unit,Human Reference Genome,Memory Footprint,NCBI Short Read Archive&amp;pmc=L32030,B12008,L32020,L32010,L15001,L21001&amp;sponsored=TwentyGB&amp;">
                        <img data-test="gpt-advert-fallback-img"
                            src="//pubads.g.doubleclick.net/gampad/ad?iu=/270604982/bmc/genomebiology/articles&amp;sz=728x90,970x90&amp;pos=LB1&amp;doi=10.1186/gb-2009-10-3-r25&amp;type=article&amp;kwrd=Additional Data File,Central Processing Unit,Human Reference Genome,Memory Footprint,NCBI Short Read Archive&amp;pmc=L32030,B12008,L32020,L32010,L15001,L21001&amp;sponsored=TwentyGB&amp;"
                            alt="Advertisement"
                            width="728"
                            height="90">
                    </a>
                </noscript>
            </div>
        </div>
    </aside>

        
        

         
    
        <div id="membership-message-loader-desktop" class="placeholder" data-placeholder="/placeholder/v1/membership/message"></div>
    
    
        <div id="top" class="u-position-relative">
    <header class="c-header" data-test="publisher-header">
        <div class="c-header__container">
            <div class="c-header__brand u-mr-48" itemscope="" itemtype="http://schema.org/Organization" data-test="navbar-logo-header">
                <div class="c-logo">
    <a href="https://www.biomedcentral.com/" itemprop="url">
        <img alt="BMC" itemprop="logo" width="76" height="18" role="img" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/logo-bmc-white-flagship-6c54473310.svg">
        <div class="c-logo__strapline">
            <img alt="Part of Springer Nature" width="173" height="16" role="img" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/logo-bmc-white-strapline-sn-f224388d67.svg">
        </div>
    </a>
</div>

            </div>
            <div class="c-header__navigation">
                <button type="button" class="c-header__link u-button-reset u-mr-24" data-expander="" data-expander-target="#publisher-header-search" data-expander-autofocus="firstTabbable" data-test="header-search-button" aria-controls="publisher-header-search" aria-expanded="false" aria-haspopup="true">
                    <span class="u-display-flex u-align-items-center">
                        <span class="u-text-sm">Search</span>
                        <svg class="u-icon u-flex-static u-ml-8" aria-hidden="true" focusable="false">
                            <use xlink:href="#global-icon-search"></use>
                        </svg>
                    </span>
                </button>
                <nav>
                    <ul class="c-header__menu u-hide-at-lt-lg" data-enhanced-menu="" data-test="publisher-navigation">
                        
                            
                                <li class="c-header__item u-hide-at-lt-lg">
                                    <a class="c-header__link" href="https://www.biomedcentral.com/journals">
                                        Explore journals
                                    </a>
                                </li>
                            
                                <li class="c-header__item u-hide-at-lt-lg">
                                    <a class="c-header__link" href="https://www.biomedcentral.com/getpublished">
                                        Get published
                                    </a>
                                </li>
                            
                                <li class="c-header__item u-hide-at-lt-lg">
                                    <a class="c-header__link" href="https://www.biomedcentral.com/about">
                                        About BMC
                                    </a>
                                </li>
                            
                        
                        <li class="c-header__item">
                            <a href="https://genomebiology.biomedcentral.com/login" class="c-header__link">Login</a>
                        </li>
                    </ul><div class="c-dropdown u-hide-at-lg"><button class="c-dropdown__button c-header__item" aria-expanded="false" aria-haspopup="true">Menu</button><ul class="c-dropdown__menu u-hide c-dropdown__menu--right"><li class="c-dropdown__item"><a class="c-dropdown__link" href="https://www.biomedcentral.com/journals">
                                        Explore journals
                                    </a></li><li class="c-dropdown__item"><a class="c-dropdown__link" href="https://www.biomedcentral.com/getpublished">
                                        Get published
                                    </a></li><li class="c-dropdown__item"><a class="c-dropdown__link" href="https://www.biomedcentral.com/about">
                                        About BMC
                                    </a></li><li class="c-dropdown__item"><a class="c-dropdown__link" href="https://genomebiology.biomedcentral.com/login">Login</a></li></ul></div>
                </nav>
            </div>
        </div>
    </header>
    <div class="c-popup-search u-js-hide" id="publisher-header-search" hidden="">
        <div class="u-container">
            <div class="c-popup-search__container">
                <div class="ctx-search">
    <form role="search" class="c-form-field" method="GET" action="https://www.biomedcentral.com/search" data-track="submit" data-track-category="Search and Results" data-track-action="Submit search" data-dynamic-track-label="" data-track-label="" data-test="global-search" novalidate="">
        <label for="publisherSearch" class="c-form-field__label">Search all BMC articles</label>
        <div class="u-display-flex">
            <input id="publisherSearch" class="c-form-field__input" data-search-input="" autocomplete="off" role="textbox" data-test="search-input" name="query" type="text" value="">
            <div>
                <button class="u-button" type="submit" data-test="search-submit-button">
    <span class="u-visually-hidden">Search</span>
    <svg class="u-icon u-flex-static" width="16" height="16" aria-hidden="true" focusable="false">
        <use xlink:href="#global-icon-search"></use>
    </svg>
</button>


            </div>
        </div>
        <input type="hidden" name="searchType" value="publisherSearch">
    </form>
</div>

            </div>
        </div>
    </div>
</div>

    

        
            <header class="c-journal-header c-journal-header--genome-biology ctx-journal-header">
                <div class="u-container">
                    <div class="c-journal-header__inner c-journal-header__inner--featured-journal">
                        
                        <div class="c-journal-title" id="journalTitle">
                            <a href="https://genomebiology.biomedcentral.com/">
    
    <span class="c-journal-title__text  u-visually-hidden">Genome Biology</span>
    
        <div class="c-journal-title__featured-journal-lines c-journal-title__featured-journal-lines--genome-biology">
            <img src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/genome-biology-logo-cf72c0ec54.svg" class="c-journal-title__featured-journal-logo" alt="Genome Biology logo" itemprop="logo" role="img">
        </div>
    
</a>
                        </div>
                        
                    </div>
                </div>
                <div class="c-navbar">
                    <div class="c-navbar__container">
                        
                            <div class="c-navbar__content">
                                <nav class="c-navbar__nav">
                                    <ul class="c-navbar__nav c-navbar__nav--journal" role="menu" data-test="site-navigation">
                                        
                                            <li class="c-navbar__item" role="menuitem">
                                                <a class="c-navbar__link" data-track="click" data-track-category="Home" data-track-action="Clicked journal navigation link" href="https://genomebiology.biomedcentral.com/">Home</a>
                                            </li>
                                        
                                            <li class="c-navbar__item" role="menuitem">
                                                <a class="c-navbar__link" data-track="click" data-track-category="About" data-track-action="Clicked journal navigation link" href="https://genomebiology.biomedcentral.com/about">About</a>
                                            </li>
                                        
                                            <li class="c-navbar__item" role="menuitem">
                                                <a class="c-navbar__link c-navbar__link--is-shown" data-track="click" data-track-category="Articles" data-track-action="Clicked journal navigation link" href="https://genomebiology.biomedcentral.com/articles">Articles</a>
                                            </li>
                                        
                                            <li class="c-navbar__item" role="menuitem">
                                                <a class="c-navbar__link" data-track="click" data-track-category="Submission Guidelines" data-track-action="Clicked journal navigation link" href="https://genomebiology.biomedcentral.com/submission-guidelines">Submission Guidelines</a>
                                            </li>
                                        
                                    </ul>
                                </nav>
                            </div>
                        
                    </div>
                </div>
                <div class="c-journal-header__identity c-journal-header__identity--featured-journal">
                    
                </div>
            </header>
            
        

        <div class="u-container u-mt-32 u-mb-32 u-clearfix" id="main-content" data-component="article-container">
            <main class="c-article-main-column u-float-left js-main-column" data-track-component="article body">
                <div class="c-context-bar u-hide" data-test="context-bar" data-context-bar="" aria-hidden="true">
                    <div class="c-context-bar__container u-container">
                        <div class="c-context-bar__title">
                            Ultrafast and memory-efficient alignment of short DNA sequences to the human genome
                        </div>
                        <div class="c-pdf-container">
                            
    <div class="c-pdf-download u-clear-both">
        <a href="https://genomebiology.biomedcentral.com/counter/pdf/10.1186/gb-2009-10-3-r25.pdf" class="u-button u-button--full-width u-button--primary u-justify-content-space-between c-pdf-download__link" data-article-pdf="true" data-readcube-pdf-url="true" data-test="pdf-link" data-draft-ignore="true" data-track="click" data-track-action="download pdf" data-track-label="button sticky" data-track-external="" download="" tabindex="-1">
            <span class="c-pdf-download__text">Download PDF</span>
            <svg aria-hidden="true" focusable="false" width="16" height="16" class="u-icon"><use xlink:href="#global-icon-download"></use></svg>
        </a>
    </div>

                            
                        </div>
                    </div>
                </div>

                <div class="c-pdf-button__container u-hide-at-lg js-context-bar-sticky-point-mobile">
                    <div class="c-pdf-container">
                        
    <div class="c-pdf-download u-clear-both">
        <a href="https://genomebiology.biomedcentral.com/counter/pdf/10.1186/gb-2009-10-3-r25.pdf" class="u-button u-button--full-width u-button--primary u-justify-content-space-between c-pdf-download__link" data-article-pdf="true" data-readcube-pdf-url="true" data-test="pdf-link" data-draft-ignore="true" data-track="click" data-track-action="download pdf" data-track-label="link" data-track-external="" download="">
            <span class="c-pdf-download__text">Download PDF</span>
            <svg aria-hidden="true" focusable="false" width="16" height="16" class="u-icon"><use xlink:href="#global-icon-download"></use></svg>
        </a>
    </div>

                        
                    </div>
                </div>
                <article lang="en">
                    <div class="c-article-header">
                        
    


                        <ul class="c-article-identifiers" data-test="article-identifier">
                            
    
        <li class="c-article-identifiers__item" data-test="article-category">Software</li>
    
    
        <li class="c-article-identifiers__item">
            <span class="c-article-identifiers__open" data-test="open-access">Open Access</span>
        </li>
    
    

                            <li class="c-article-identifiers__item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#article-info" data-track="click" data-track-action="publication date" data-track-label="link">Published: <time datetime="2009-03-04">04 March 2009</time></a></li>
                        </ul>

                        <h1 class="c-article-title" data-test="article-title" data-article-title="">Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</h1>
                        <ul class="c-article-author-list c-article-author-list--short js-no-scroll" data-test="authors-list" data-component-authors-activator="authors-list"><li class="c-article-author-list__item"><a data-test="author-name" data-track="click" data-track-action="open author" data-track-label="link" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#auth-Ben-Langmead" data-author-popup="auth-Ben-Langmead" data-corresp-id="c1">Ben Langmead<svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-email"></use></svg></a><sup class="u-js-hide"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Aff1" tabindex="-1">1</a></sup>, </li><li class="c-article-author-list__item"><a data-test="author-name" data-track="click" data-track-action="open author" data-track-label="link" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#auth-Cole-Trapnell" data-author-popup="auth-Cole-Trapnell">Cole Trapnell</a><sup class="u-js-hide"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Aff1" tabindex="-1">1</a></sup>, </li><li class="c-article-author-list__item c-article-author-list__item--hide-small-screen"><a data-test="author-name" data-track="click" data-track-action="open author" data-track-label="link" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#auth-Mihai-Pop" data-author-popup="auth-Mihai-Pop">Mihai Pop</a><sup class="u-js-hide"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Aff1" tabindex="-1">1</a></sup> &amp; </li><li class="c-article-author-list__show-more" aria-label="Show all 4 authors for this article" title="Show all 4 authors for this article">…</li><li class="c-article-author-list__item"><a data-test="author-name" data-track="click" data-track-action="open author" data-track-label="link" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#auth-Steven_L-Salzberg" data-author-popup="auth-Steven_L-Salzberg">Steven L Salzberg</a><sup class="u-js-hide"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Aff1" tabindex="-1">1</a></sup>&nbsp;</li></ul><button aria-expanded="false" class="c-article-author-list__button"><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#icon-plus"></use></svg><span>Show authors</span></button>
                        <p class="c-article-info-details" data-container-section="info">
                            
    <a data-test="journal-link" href="https://genomebiology.biomedcentral.com/"><i data-test="journal-title">Genome Biology</i></a>

                            <b data-test="journal-volume"><span class="u-visually-hidden">volume</span>&nbsp;10</b>, Article&nbsp;number:&nbsp;<span data-test="article-number">R25</span> (<span data-test="article-publication-year">2009</span>)
            <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#citeas" class="c-article-info-details__cite-as u-hide-print" data-track="click" data-track-action="cite this article" data-track-label="link">Cite this article</a>
                        </p>
                        
    <div class="c-article-metrics-bar__wrapper u-clear-both">
        <ul class="c-article-metrics-bar u-list-reset">
            
                <li class=" c-article-metrics-bar__item">
                    <p class="c-article-metrics-bar__count">281k <span class="c-article-metrics-bar__label">Accesses</span></p>
                </li>
            
            
                <li class="c-article-metrics-bar__item">
                    <p class="c-article-metrics-bar__count">14055 <span class="c-article-metrics-bar__label">Citations</span></p>
                </li>
            
            
                
                    <li class="c-article-metrics-bar__item">
                        <p class="c-article-metrics-bar__count">82 <span class="c-article-metrics-bar__label">Altmetric</span></p>
                    </li>
                
            
            <li class="c-article-metrics-bar__item">
                <p class="c-article-metrics-bar__details"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/metrics" data-track="click" data-track-action="view metrics" data-track-label="link" rel="nofollow">Metrics <span class="u-visually-hidden">details</span></a></p>
            </li>
        </ul>
    </div>

                        
                        
    

    

                        
                    </div>

                    <section aria-labelledby="Abs1" data-title="Abstract" lang="en"><div class="c-article-section" id="Abs1-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Abs1">Abstract</h2><div class="c-article-section__content" id="Abs1-content"><p>Bowtie is an ultrafast, memory-efficient alignment program for aligning short DNA sequence reads to large genomes. For the human genome, Burrows-Wheeler indexing allows Bowtie to align more than 25 million reads per CPU hour with a memory footprint of approximately 1.3 gigabytes. Bowtie extends previous Burrows-Wheeler techniques with a novel quality-aware backtracking algorithm that permits mismatches. Multiple processor cores can be used simultaneously to achieve even greater alignment speeds. Bowtie is open source <a href="http://bowtie.cbcb.umd.edu/">http://bowtie.cbcb.umd.edu</a>.</p></div></div></section>
                    
    


                    <section data-title="Rationale"><div class="c-article-section" id="Sec1-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Sec1">Rationale</h2><div class="c-article-section__content" id="Sec1-content"><p>Improvements in the efficiency of DNA sequencing have both broadened the applications for sequencing and dramatically increased the size of sequencing datasets. Technologies from Illumina (San Diego, CA, USA) and Applied Biosystems (Foster City, CA, USA) have been used to profile methylation patterns (MeDIP-Seq) [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 1" title="Down TA, Rakyan VK, Turner DJ, Flicek P, Li H, Kulesha E, Graf S, Johnson N, Herrero J, Tomazou EM, Thorne NP, Backdahl L, Herberth M, Howe KL, Jackson DK, Miretti MM, Marioni JC, Birney E, Hubbard TJ, Durbin R, Tavare S, Beck S: A Bayesian deconvolution strategy for immunoprecipitation-based DNA methylome analysis. Nat Biotechnol. 2008, 26: 779-785. 10.1038/nbt1414." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR1" id="ref-link-section-d3015500e454">1</a>], to map DNA-protein interactions (ChIP-Seq) [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 2" title="Johnson DS, Mortazavi A, Myers RM, Wold B: Genome-wide mapping of in vivo protein-DNA interactions. Science. 2007, 316: 1497-1502. 10.1126/science.1141319." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR2" id="ref-link-section-d3015500e457">2</a>], and to identify differentially expressed genes (RNA-Seq) [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 3" title="Marioni JC, Mason CE, Mane SM, Stephens M, Gilad Y: RNA-seq: an assessment of technical reproducibility and comparison with gene expression arrays. Genome Res. 2008, 18: 1509-1517. 10.1101/gr.079558.108." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR3" id="ref-link-section-d3015500e460">3</a>] in the human genome and other species. The Illumina instrument was recently used to re-sequence three human genomes, one from a cancer patient and two from previously unsequenced ethnic groups [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 4" title="Bentley DR, Balasubramanian S, Swerdlow HP, Smith GP, Milton J, Brown CG, Hall KP, Evers DJ, Barnes CL, Bignell HR, Boutell JM, Bryant J, Carter RJ, Keira Cheetham R, Cox AJ, Ellis DJ, Flatbush MR, Gormley NA, Humphray SJ, Irving LJ, Karbelashvili MS, Kirk SM, Li H, Liu X, Maisinger KS, Murray LJ, Obradovic B, Ost T, Parkinson ML, Pratt MR, et al: Accurate whole human genome sequencing using reversible terminator chemistry. Nature. 2008, 456: 53-59. 10.1038/nature07517." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR4" id="ref-link-section-d3015500e463">4</a>–<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 6" title="Wang J, Wang W, Li R, Li Y, Tian G, Goodman L, Fan W, Zhang J, Li J, Zhang J, Guo Y, Feng B, Li H, Lu Y, Fang X, Liang H, Du Z, Li D, Zhao Y, Hu Y, Yang Z, Zheng H, Hellmann I, Inouye M, Pool J, Yi X, Zhao J, Duan J, Zhou Y, Qin J, et al: The diploid genome sequence of an Asian individual. Nature. 2008, 456: 60-65. 10.1038/nature07484." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR6" id="ref-link-section-d3015500e466">6</a>]. Each of these studies required the alignment of large numbers of short DNA sequences ('short reads') onto the human genome. For example, two of the studies [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 4" title="Bentley DR, Balasubramanian S, Swerdlow HP, Smith GP, Milton J, Brown CG, Hall KP, Evers DJ, Barnes CL, Bignell HR, Boutell JM, Bryant J, Carter RJ, Keira Cheetham R, Cox AJ, Ellis DJ, Flatbush MR, Gormley NA, Humphray SJ, Irving LJ, Karbelashvili MS, Kirk SM, Li H, Liu X, Maisinger KS, Murray LJ, Obradovic B, Ost T, Parkinson ML, Pratt MR, et al: Accurate whole human genome sequencing using reversible terminator chemistry. Nature. 2008, 456: 53-59. 10.1038/nature07517." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR4" id="ref-link-section-d3015500e470">4</a>, <a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 5" title="Ley TJ, Mardis ER, Ding L, Fulton B, McLellan MD, Chen K, Dooling D, Dunford-Shore BH, McGrath S, Hickenbotham M, Cook L, Abbott R, Larson DE, Koboldt DC, Pohl C, Smith S, Hawkins A, Abbott S, Locke D, Hillier LW, Miner T, Fulton L, Magrini V, Wylie T, Glasscock J, Conyers J, Sander N, Shi X, Osborne JR, Minx P, et al: DNA sequencing of a cytogenetically normal acute myeloid leukaemia genome. Nature. 2008, 456: 66-72. 10.1038/nature07485." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR5" id="ref-link-section-d3015500e473">5</a>] used the short read alignment tool Maq [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 7" title="Li H, Ruan J, Durbin R: Mapping short DNA sequencing reads and calling variants using mapping quality scores. Genome Res. 2008, 18: 1851-1858. 10.1101/gr.078212.108." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR7" id="ref-link-section-d3015500e476">7</a>] to align more than 130 billion bases (about 45× coverage) of short Illumina reads to a human reference genome in order to detect genetic variations. The third human re-sequencing study [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 6" title="Wang J, Wang W, Li R, Li Y, Tian G, Goodman L, Fan W, Zhang J, Li J, Zhang J, Guo Y, Feng B, Li H, Lu Y, Fang X, Liang H, Du Z, Li D, Zhao Y, Hu Y, Yang Z, Zheng H, Hellmann I, Inouye M, Pool J, Yi X, Zhao J, Duan J, Zhou Y, Qin J, et al: The diploid genome sequence of an Asian individual. Nature. 2008, 456: 60-65. 10.1038/nature07484." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR6" id="ref-link-section-d3015500e479">6</a>] used the SOAP program [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 8" title="Li R, Li Y, Kristiansen K, Wang J: SOAP: short oligonucleotide alignment program. Bioinformatics. 2008, 24: 713-714. 10.1093/bioinformatics/btn025." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR8" id="ref-link-section-d3015500e482">8</a>] to align more than 100 billion bases to the reference genome. In addition to these projects, the 1,000 Genomes project is in the process of using high-throughput sequencing instruments to sequence a total of about six trillion base pairs of human DNA [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 9" title="Kaiser J: DNA sequencing. A plan to capture human diversity in 1000 genomes. Science. 2008, 319: 395-10.1126/science.319.5862.395." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR9" id="ref-link-section-d3015500e485">9</a>].</p><p>With existing methods, the computational cost of aligning many short reads to a mammalian genome is very large. For example, extrapolating from the results presented here in Tables <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab1">1</a> and <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab2">2</a>, one can see that Maq would require more than 5 central processing unit (CPU)-months and SOAP more than 3 CPU-years to align the 140 billion bases from the study by Ley and coworkers [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 5" title="Ley TJ, Mardis ER, Ding L, Fulton B, McLellan MD, Chen K, Dooling D, Dunford-Shore BH, McGrath S, Hickenbotham M, Cook L, Abbott R, Larson DE, Koboldt DC, Pohl C, Smith S, Hawkins A, Abbott S, Locke D, Hillier LW, Miner T, Fulton L, Magrini V, Wylie T, Glasscock J, Conyers J, Sander N, Shi X, Osborne JR, Minx P, et al: DNA sequencing of a cytogenetically normal acute myeloid leukaemia genome. Nature. 2008, 456: 66-72. 10.1038/nature07485." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR5" id="ref-link-section-d3015500e497">5</a>]. Although using Maq or SOAP for this purpose has been shown to be feasible by using multiple CPUs, there is a clear need for new tools that consume less time and computational resources.</p><div class="c-article-table" data-test="inline-table" data-container-section="table" id="table-1"><figure><figcaption class="c-article-table__figcaption"><b id="Tab1" data-test="table-caption">Table 1 Bowtie alignment performance versus SOAP and Maq</b></figcaption><div class="u-text-right u-hide-print"><a class="c-article__pill-button" data-test="table-link" data-track="click" data-track-action="view table" data-track-label="button" rel="nofollow" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/tables/1" aria-label="Full size table 1"><span>Full size table</span><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-chevron-right"></use></svg></a></div></figure></div><div class="c-article-table" data-test="inline-table" data-container-section="table" id="table-2"><figure><figcaption class="c-article-table__figcaption"><b id="Tab2" data-test="table-caption">Table 2 Bowtie alignment performance versus Maq with filtered read set</b></figcaption><div class="u-text-right u-hide-print"><a class="c-article__pill-button" data-test="table-link" data-track="click" data-track-action="view table" data-track-label="button" rel="nofollow" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/tables/2" aria-label="Full size table 2"><span>Full size table</span><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-chevron-right"></use></svg></a></div></figure></div><p>Maq and SOAP take the same basic algorithmic approach as other recent read mapping tools such as RMAP [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 10" title="Smith AD, Xuan Z, Zhang MQ: Using quality scores and longer reads improves accuracy of Solexa read mapping. BMC Bioinformatics. 2008, 9: 128-10.1186/1471-2105-9-128." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR10" id="ref-link-section-d3015500e1097">10</a>], ZOOM [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 11" title="Lin H, Zhang Z, Zhang MQ, Ma B, Li M: ZOOM! Zillions Of Oligos Mapped. Bioinformatics. 2008, 24: 2431-2437. 10.1093/bioinformatics/btn416." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR11" id="ref-link-section-d3015500e1100">11</a>], and SHRiMP [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 12" title="SHRiMP - SHort Read Mapping Package. [
                    http://compbio.cs.toronto.edu/shrimp/
                    
                  ]" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR12" id="ref-link-section-d3015500e1103">12</a>]. Each tool builds a hash table of short oligomers present in either the reads (SHRiMP, Maq, RMAP, and ZOOM) or the reference (SOAP). Some employ recent theoretical advances to align reads quickly without sacrificing sensitivity. For example, ZOOM uses 'spaced seeds' to significantly outperform RMAP, which is based on a simpler algorithm developed by Baeza-Yaetes and Perleberg [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 13" title="Baeza-Yates RA, Perleberg CH: Fast and practical approximate string matching. Inf Process Lett. 1996, 59: 21-27. 10.1016/0020-0190(96)00083-X." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR13" id="ref-link-section-d3015500e1106">13</a>]. Spaced seeds have been shown to yield higher sensitivity than contiguous seeds of the same length [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 14" title="Burkhardt S, Kärkkäinen J: Better Filtering with Gapped q-Grams. Fundam Inf. 2003, 56: 51-70." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR14" id="ref-link-section-d3015500e1109">14</a>, <a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 15" title="Ma B, Tromp J, Li M: PatternHunter: faster and more sensitive homology search. Bioinformatics. 2002, 18: 440-445. 10.1093/bioinformatics/18.3.440." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR15" id="ref-link-section-d3015500e1113">15</a>]. SHRiMP employs a combination of spaced seeds and the Smith-Waterman [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 16" title="Smith TF, Waterman MS: Identification of common molecular subsequences. J Mol Biol. 1981, 147: 195-197. 10.1016/0022-2836(81)90087-5." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR16" id="ref-link-section-d3015500e1116">16</a>] algorithm to align reads with high sensitivity at the expense of speed. Eland is a commercial alignment program available from Illumina that uses a hash-based algorithm to align reads.</p><p>Bowtie uses a different and novel indexing strategy to create an ultrafast, memory-efficient short read aligner geared toward mammalian re-sequencing. In our experiments using reads from the 1,000 Genomes project, Bowtie aligns 35-base pair (bp) reads at a rate of more than 25 million reads per CPU-hour, which is more than 35 times faster than Maq and 300 times faster than SOAP under the same conditions (see Tables <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab1">1</a> and <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab2">2</a>). Bowtie employs a Burrows-Wheeler index based on the full-text minute-space (FM) index, which has a memory footprint of only about 1.3 gigabytes (GB) for the human genome. The small footprint allows Bowtie to run on a typical desktop computer with 2 GB of RAM. The index is small enough to be distributed over the internet and to be stored on disk and re-used. Multiple processor cores can be used simultaneously to achieve even greater alignment speed. We have used Bowtie to align 14.3× coverage worth of human Illumina reads from the 1,000 Genomes project in about 14 hours on a single desktop computer with four processor cores.</p><p>Bowtie makes a number of compromises to achieve this speed, but these trade-offs are reasonable within the context of mammalian re-sequencing projects. If one or more exact matches exist for a read, then Bowtie is guaranteed to report one, but if the best match is an inexact one then Bowtie is not guaranteed in all cases to find the highest quality alignment. With its highest performance settings, Bowtie may fail to align a small number of reads with valid alignments, if those reads have multiple mismatches. If the stronger guarantees are desired, Bowtie supports options that increase accuracy at the cost of some performance. For instance, the '--best' option will guarantee that all alignments reported are best in terms of minimizing mismatches in the seed portion of the read, although this option incurs additional computational cost.</p><p>With its default options, Bowtie's sensitivity measured in terms of reads aligned is equal to SOAP's and somewhat less than Maq's. Command line options allow the user to increase sensitivity at the cost of greater running time, and to enable Bowtie to report multiple hits for a read. Bowtie can align reads as short as four bases and as long as 1,024 bases. The input to a single run of Bowtie may comprise a mixture of reads with different lengths.</p></div></div></section><section data-title="Bowtie description and results"><div class="c-article-section" id="Sec2-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Sec2">Bowtie description and results</h2><div class="c-article-section__content" id="Sec2-content"><p>Bowtie indexes the reference genome using a scheme based on the Burrows-Wheeler transform (BWT) [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 17" title="Burrows M, Wheeler DJ: A Block Sorting Lossless Data Compression Algorithm. Technical Report 124. 1994, Palo Alto, CA: Digital Equipment Corporation" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR17" id="ref-link-section-d3015500e1141">17</a>] and the FM index [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 18" title="Ferragina P, Manzini G: Opportunistic data structures with applications. [
                    http://web.unipmn.it/~manzini/papers/focs00draft.pdf
                    
                  ]" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR18" id="ref-link-section-d3015500e1144">18</a>, <a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 19" title="Ferragina P, Manzini G: An experimental study of an opportunistic index. Proceedings of the Twelfth Annual ACM-SIAM Symposium on Discrete algorithms. 2001, Washington, DC: Society for Industrial and Applied Mathematics, 269-278." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR19" id="ref-link-section-d3015500e1147">19</a>]. A Bowtie index for the human genome fits in 2.2 GB on disk and has a memory footprint of as little as 1.3 GB at alignment time, allowing it to be queried on a workstation with under 2 GB of RAM.</p><p>The common method for searching in an FM index is the exact-matching algorithm of Ferragina and Manzini [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 18" title="Ferragina P, Manzini G: Opportunistic data structures with applications. [
                    http://web.unipmn.it/~manzini/papers/focs00draft.pdf
                    
                  ]" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR18" id="ref-link-section-d3015500e1153">18</a>]. Bowtie does not simply adopt this algorithm because exact matching does not allow for sequencing errors or genetic variations. We introduce two novel extensions that make the technique applicable to short read alignment: a quality-aware backtracking algorithm that allows mismatches and favors high-quality alignments; and 'double indexing', a strategy to avoid excessive backtracking. The Bowtie aligner follows a policy similar to Maq's, in that it allows a small number of mismatches within the high-quality end of each read, and it places an upper limit on the sum of the quality values at mismatched alignment positions.</p><h3 class="c-article__sub-heading" id="Sec3">Burrows-Wheeler indexing</h3><p>The BWT is a reversible permutation of the characters in a text. Although originally developed within the context of data compression, BWT-based indexing allows large texts to be searched efficiently in a small memory footprint. It has been applied to bioinformatics applications, including oligomer counting [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 20" title="Healy J, Thomas EE, Schwartz JT, Wigler M: Annotating large genomes with exact word matches. Genome Res. 2003, 13: 2306-2315. 10.1101/gr.1350803." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR20" id="ref-link-section-d3015500e1163">20</a>], whole-genome alignment [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 21" title="Lippert RA: Space-efficient whole genome comparisons with Burrows-Wheeler transforms. J Comput Biol. 2005, 12: 407-415. 10.1089/cmb.2005.12.407." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR21" id="ref-link-section-d3015500e1166">21</a>], tiling microarray probe design [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 22" title="Graf S, Nielsen FG, Kurtz S, Huynen MA, Birney E, Stunnenberg H, Flicek P: Optimized design and assessment of whole genome tiling arrays. Bioinformatics. 2007, 23: i195-i204. 10.1093/bioinformatics/btm200." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR22" id="ref-link-section-d3015500e1169">22</a>], and Smith-Waterman alignment to a human-sized reference [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 23" title="Lam TW, Sung WK, Tam SL, Wong CK, Yiu SM: Compressed indexing and local alignment of DNA. Bioinformatics. 2008, 24: 791-797. 10.1093/bioinformatics/btn032." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR23" id="ref-link-section-d3015500e1172">23</a>].</p><p>The Burrows-Wheeler transformation of a text T, BWT(T), is constructed as follows. The character $ is appended to T, where $ is not in T and is lexicographically less than all characters in T. The Burrows-Wheeler matrix of T is constructed as the matrix whose rows comprise all cyclic rotations of T$. The rows are then sorted lexicographically. BWT(T) is the sequence of characters in the rightmost column of the Burrows-Wheeler matrix (Figure <a data-track="click" data-track-label="link" data-track-action="figure anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Fig1">1a</a>). BWT(T) has the same length as the original text T.</p><div class="c-article-section__figure js-c-reading-companion-figures-item" data-test="figure" data-container-section="figure" id="figure-1" data-title="Figure 1"><figure><figcaption><b id="Fig1" class="c-article-section__figure-caption" data-test="figure-caption-text">Figure 1</b></figcaption><div class="c-article-section__figure-content"><div class="c-article-section__figure-item"><a class="c-article-section__figure-link" data-test="img-link" data-track="click" data-track-label="image" data-track-action="view figure" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/1" rel="nofollow"><picture><source type="image/webp" srcset="//media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig1_HTML.jpg?as=webp"><img aria-describedby="Fig1" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/13059_2008_Article_2175_Fig1_HTML.jpg" alt="figure 1" loading="lazy"></picture></a></div><div class="c-article-section__figure-description" data-test="bottom-caption" id="figure-1-desc"><p>Burrows-Wheeler transform. <b>(a)</b> The Burrows-Wheeler matrix and transformation for 'acaacg'. <b>(b)</b> Steps taken by EXACTMATCH to identify the range of rows, and thus the set of reference suffixes, prefixed by 'aac'. <b>(c)</b> UNPERMUTE repeatedly applies the last first (LF) mapping to recover the original text (in red on the top line) from the Burrows-Wheeler transform (in black in the rightmost column).</p></div></div><div class="u-text-right u-hide-print"><a class="c-article__pill-button" data-test="article-link" data-track="click" data-track-label="button" data-track-action="view figure" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/1" data-track-dest="link:Figure1 Full size image" aria-label="Full size image figure 1" rel="nofollow"><span>Full size image</span><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-chevron-right"></use></svg></a></div></figure></div><p>This matrix has a property called 'last first (LF) mapping'. The i<sup>th</sup> occurrence of character X in the last column corresponds to the same text character as the i<sup>th</sup> occurrence of X in the first column. This property underlies algorithms that use the BWT index to navigate or search the text. Figure <a data-track="click" data-track-label="link" data-track-action="figure anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Fig1">1b</a> illustrates UNPERMUTE, an algorithm that applies the LF mapping repeatedly to re-create T from BWT(T).</p><p>The LF mapping is also used in exact matching. Because the matrix is sorted lexicographically, rows beginning with a given sequence appear consecutively. In a series of steps, the EXACTMATCH algorithm (Figure <a data-track="click" data-track-label="link" data-track-action="figure anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Fig1">1c</a>) calculates the range of matrix rows beginning with successively longer suffixes of the query. At each step, the size of the range either shrinks or remains the same. When the algorithm completes, rows beginning with S<sub>0</sub> (the entire query) correspond to exact occurrences of the query in the text. If the range is empty, the text does not contain the query. UNPERMUTE is attributable to Burrows and Wheeler [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 17" title="Burrows M, Wheeler DJ: A Block Sorting Lossless Data Compression Algorithm. Technical Report 124. 1994, Palo Alto, CA: Digital Equipment Corporation" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR17" id="ref-link-section-d3015500e1226">17</a>] and EXACTMATCH to Ferragina and Manzini [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 18" title="Ferragina P, Manzini G: Opportunistic data structures with applications. [
                    http://web.unipmn.it/~manzini/papers/focs00draft.pdf
                    
                  ]" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR18" id="ref-link-section-d3015500e1229">18</a>]. See Additional data file 1 (Supplementary Discussion 1) for details.</p><h3 class="c-article__sub-heading" id="Sec4">Searching for inexact alignments</h3><p>EXACTMATCH is insufficient for short read alignment because alignments may contain mismatches, which may be due to sequencing errors, genuine differences between reference and query organisms, or both. We introduce an alignment algorithm that conducts a backtracking search to quickly find alignments that satisfy a specified alignment policy. Each character in a read has a numeric quality value, with lower values indicating a higher likelihood of a sequencing error. Our alignment policy allows a limited number of mismatches and prefers alignments where the sum of the quality values at all mismatched positions is low.</p><p>The search proceeds similarly to EXACTMATCH, calculating matrix ranges for successively longer query suffixes. If the range becomes empty (a suffix does not occur in the text), then the algorithm may select an already-matched query position and substitute a different base there, introducing a mismatch into the alignment. The EXACTMATCH search resumes from just after the substituted position. The algorithm selects only those substitutions that are consistent with the alignment policy and which yield a modified suffix that occurs at least once in the text. If there are multiple candidate substitution positions, then the algorithm greedily selects a position with a minimal quality value.</p><p>Backtracking scenarios play out within the context of a stack structure that grows when a new substitution is introduced and shrinks when the aligner rejects all candidate alignments for the substitutions currently on the stack. See Figure <a data-track="click" data-track-label="link" data-track-action="figure anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Fig2">2</a> for an illustration of how the search might proceed.</p><div class="c-article-section__figure js-c-reading-companion-figures-item" data-test="figure" data-container-section="figure" id="figure-2" data-title="Figure 2"><figure><figcaption><b id="Fig2" class="c-article-section__figure-caption" data-test="figure-caption-text">Figure 2</b></figcaption><div class="c-article-section__figure-content"><div class="c-article-section__figure-item"><a class="c-article-section__figure-link" data-test="img-link" data-track="click" data-track-label="image" data-track-action="view figure" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/2" rel="nofollow"><picture><source type="image/webp" srcset="//media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig2_HTML.jpg?as=webp"><img aria-describedby="Fig2" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/13059_2008_Article_2175_Fig2_HTML.jpg" alt="figure 2" loading="lazy"></picture></a></div><div class="c-article-section__figure-description" data-test="bottom-caption" id="figure-2-desc"><p>Exact matching versus inexact alignment. Illustration of how EXACTMATCH (top) and Bowtie's aligner (bottom) proceed when there is no exact match for query 'ggta' but there is a one-mismatch alignment when 'a' is replaced by 'g'. Boxed pairs of numbers denote ranges of matrix rows beginning with the suffix observed up to that point. A red X marks where the algorithm encounters an empty range and either aborts (as in EXACTMATCH) or backtracks (as in the inexact algorithm). A green check marks where the algorithm finds a nonempty range delimiting one or more occurrences of a reportable alignment for the query.</p></div></div><div class="u-text-right u-hide-print"><a class="c-article__pill-button" data-test="article-link" data-track="click" data-track-label="button" data-track-action="view figure" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/2" data-track-dest="link:Figure2 Full size image" aria-label="Full size image figure 2" rel="nofollow"><span>Full size image</span><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-chevron-right"></use></svg></a></div></figure></div><p>In short, Bowtie conducts a quality-aware, greedy, randomized, depth-first search through the space of possible alignments. If a valid alignment exists, then Bowtie will find it (subject to the backtrack ceiling discussed in the following section). Because the search is greedy, the first valid alignment encountered by Bowtie will not necessarily be the 'best' in terms of number of mismatches or in terms of quality. The user may instruct Bowtie to continue searching until it can prove that any alignment it reports is 'best' in terms of number of mismatches (using the option --best). In our experience, this mode is two to three times slower than the default mode. We expect that the faster default mode will be preferred for large re-sequencing projects.</p><p>The user may also opt for Bowtie to report all alignments up to a specified number (option -k) or all alignments with no limit on the number (option -a) for a given read. If in the course of its search Bowtie finds N possible alignments for a given set of substitutions, but the user has requested only K alignments where K &lt; N, Bowtie will report K of the N alignments selected at random. Note that these modes can be much slower than the default. In our experience, for example, -k 1 is more than twice as fast as -k 2.</p><h3 class="c-article__sub-heading" id="Sec5">Excessive backtracking</h3><p>The aligner as described so far can, in some cases, encounter sequences that cause excessive backtracking. This occurs when the aligner spends most of its effort fruitlessly backtracking to positions close to the 3' end of the query. Bowtie mitigates excessive backtracking with the novel technique of 'double indexing'. Two indices of the genome are created: one containing the BWT of the genome, called the 'forward' index, and a second containing the BWT of the genome with its character sequence reversed (not reverse complemented) called the 'mirror' index. To see how this helps, consider a matching policy that allows one mismatch in the alignment. A valid alignment with one mismatch falls into one of two cases according to which half of the read contains the mismatch. Bowtie proceeds in two phases corresponding to those two cases. Phase 1 loads the forward index into memory and invokes the aligner with the constraint that it may not substitute at positions in the query's right half. Phase 2 uses the mirror index and invokes the aligner on the reversed query, with the constraint that the aligner may not substitute at positions in the reversed query's right half (the original query's left half). The constraints on backtracking into the right half prevent excessive backtracking, whereas the use of two phases and two indices maintains full sensitivity.</p><p>Unfortunately, it is not possible to avoid excessive backtracking fully when alignments are permitted to have two or more mismatches. In our experiments, we have observed that excessive backtracking is significant only when a read has many low-quality positions and does not align or aligns poorly to the reference. These cases can trigger in excess of 200 backtracks per read because there are many legal combinations of low-quality positions to be explored before all possibilities are exhausted. We mitigate this cost by enforcing a limit on the number of backtracks allowed before a search is terminated (default: 125). The limit prevents some legitimate, low-quality alignments from being reported, but we expect that this is a desirable trade-off for most applications.</p><h3 class="c-article__sub-heading" id="Sec6">Phased Maq-like search</h3><p>Bowtie allows the user to select the number of mismatches permitted (default: two) in the high-quality end of a read (default: the first 28 bases) as well as the maximum acceptable quality distance of the overall alignment (default: 70). Quality values are assumed to follow the definition in PHRED [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 24" title="Ewing B, Green P: Base-calling of automated sequencer traces using phred. II. Error probabilities. Genome Res. 1998, 8: 186-194." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR24" id="ref-link-section-d3015500e1287">24</a>], where p is the probability of error and Q = -10log p.</p><p>Both the read and its reverse complement are candidates for alignment to the reference. For clarity, this discussion considers only the forward orientation. See Additional data file 1 (Supplementary Discussion 2) for an explanation of how the reverse complement is incorporated.</p><p>The first 28 bases on the high-quality end of the read are termed the 'seed'. The seed consists of two halves: the 14 bp on the high-quality end (usually the 5' end) and the 14 bp on the low-quality end, termed the 'hi-half' and the 'lo-half', respectively. Assuming the default policy (two mismatches permitted in the seed), a reportable alignment will fall into one of four cases: no mismatches in seed (case 1); no mismatches in hi-half, one or two mismatches in lo-half (case 2); no mismatches in lo-half, one or two mismatches in hi-half (case 3); and one mismatch in hi-half, one mismatch in lo-half (case 4).</p><p>All cases allow any number of mismatches in the nonseed part of the read and all cases are also subject to the quality distance constraint.</p><p>The Bowtie algorithm consists of three phases that alternate between using the forward and mirror indices, as illustrated in Figure <a data-track="click" data-track-label="link" data-track-action="figure anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Fig3">3</a>. Phase 1 uses the mirror index and invokes the aligner to find alignments for cases 1 and 2. Phases 2 and 3 cooperate to find alignments for case 3: Phase 2 finds partial alignments with mismatches only in the hi-half and phase 3 attempts to extend those partial alignments into full alignments. Finally, phase 3 invokes the aligner to find alignments for case 4.</p><div class="c-article-section__figure js-c-reading-companion-figures-item" data-test="figure" data-container-section="figure" id="figure-3" data-title="Figure 3"><figure><figcaption><b id="Fig3" class="c-article-section__figure-caption" data-test="figure-caption-text">Figure 3</b></figcaption><div class="c-article-section__figure-content"><div class="c-article-section__figure-item"><a class="c-article-section__figure-link" data-test="img-link" data-track="click" data-track-label="image" data-track-action="view figure" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/3" rel="nofollow"><picture><source type="image/webp" srcset="//media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig3_HTML.jpg?as=webp"><img aria-describedby="Fig3" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/13059_2008_Article_2175_Fig3_HTML.jpg" alt="figure 3" loading="lazy"></picture></a></div><div class="c-article-section__figure-description" data-test="bottom-caption" id="figure-3-desc"><p>The three phases of the Bowtie algorithm for the Maq-like policy. A three-phase approach finds alignments for two-mismatch cases 1 to 4 while minimizing backtracking. Phase 1 uses the mirror index and invokes the aligner to find alignments for cases 1 and 2. Phases 2 and 3 cooperate to find alignments for case 3: Phase 2 finds partial alignments with mismatches only in the hi-half, and phase 3 attempts to extend those partial alignments into full alignments. Finally, phase 3 invokes the aligner to find alignments for case 4.</p></div></div><div class="u-text-right u-hide-print"><a class="c-article__pill-button" data-test="article-link" data-track="click" data-track-label="button" data-track-action="view figure" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/3" data-track-dest="link:Figure3 Full size image" aria-label="Full size image figure 3" rel="nofollow"><span>Full size image</span><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-chevron-right"></use></svg></a></div></figure></div><h3 class="c-article__sub-heading" id="Sec7">Performance results</h3><p>We evaluated the performance of Bowtie using reads from the 1,000 Genomes project pilot (National Center for Biotechnology Information [NCBI] Short Read Archive:SRR001115). A total of 8.84 million reads, about one lane of data from an Illumina instrument, were trimmed to 35 bp and aligned to the human reference genome [NCBI build 36.3]. Unless specified otherwise, read data are not filtered or modified (besides trimming) from how they appear in the archive. This leads to about 70% to 75% of reads aligning somewhere to the genome. In our experience, this is typical for raw data from the archive. More aggressive filtering leads to higher alignment rates and faster alignment.</p><p>All runs were performed on a single CPU. Bowtie speedups were calculated as a ratio of wall-clock alignment times. Both wall-clock and CPU times are given to demonstrate that input/output load and CPU contention are not significant factors.</p><p>The time required to build the Bowtie index was not included in the Bowtie running times. Unlike competing tools, Bowtie can reuse a pre-computed index for the reference genome across many alignment runs. We anticipate most users will simply download such indices from a public repository. The Bowtie site [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 25" title="Bowtie: An ultrafast memory-efficient short read aligner. [
                    http://bowtie.cbcb.umd.edu/
                    
                  ]" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR25" id="ref-link-section-d3015500e1332">25</a>] provides indices for current builds of the human, chimp, mouse, dog, rat, and <i>Arabidopsis thaliana</i> genomes, as well as many others.</p><p>Results were obtained on two hardware platforms: a desktop workstation with 2.4 GHz Intel Core 2 processor and 2 GB of RAM; and a large-memory server with a four-core 2.4 GHz AMD Opteron processor and 32 GB of RAM. These are denoted 'PC' and 'server', respectively. Both PC and server run Red Hat Enterprise Linux AS release 4.</p><h3 class="c-article__sub-heading" id="Sec8">Comparison to SOAP and Maq</h3><p>Maq is a popular aligner [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 1" title="Down TA, Rakyan VK, Turner DJ, Flicek P, Li H, Kulesha E, Graf S, Johnson N, Herrero J, Tomazou EM, Thorne NP, Backdahl L, Herberth M, Howe KL, Jackson DK, Miretti MM, Marioni JC, Birney E, Hubbard TJ, Durbin R, Tavare S, Beck S: A Bayesian deconvolution strategy for immunoprecipitation-based DNA methylome analysis. Nat Biotechnol. 2008, 26: 779-785. 10.1038/nbt1414." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR1" id="ref-link-section-d3015500e1348">1</a>, <a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 4" title="Bentley DR, Balasubramanian S, Swerdlow HP, Smith GP, Milton J, Brown CG, Hall KP, Evers DJ, Barnes CL, Bignell HR, Boutell JM, Bryant J, Carter RJ, Keira Cheetham R, Cox AJ, Ellis DJ, Flatbush MR, Gormley NA, Humphray SJ, Irving LJ, Karbelashvili MS, Kirk SM, Li H, Liu X, Maisinger KS, Murray LJ, Obradovic B, Ost T, Parkinson ML, Pratt MR, et al: Accurate whole human genome sequencing using reversible terminator chemistry. Nature. 2008, 456: 53-59. 10.1038/nature07517." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR4" id="ref-link-section-d3015500e1351">4</a>, <a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 5" title="Ley TJ, Mardis ER, Ding L, Fulton B, McLellan MD, Chen K, Dooling D, Dunford-Shore BH, McGrath S, Hickenbotham M, Cook L, Abbott R, Larson DE, Koboldt DC, Pohl C, Smith S, Hawkins A, Abbott S, Locke D, Hillier LW, Miner T, Fulton L, Magrini V, Wylie T, Glasscock J, Conyers J, Sander N, Shi X, Osborne JR, Minx P, et al: DNA sequencing of a cytogenetically normal acute myeloid leukaemia genome. Nature. 2008, 456: 66-72. 10.1038/nature07485." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR5" id="ref-link-section-d3015500e1354">5</a>, <a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 26" title="Campbell PJ, Stephens PJ, Pleasance ED, O&#39;Meara S, Li H, Santarius T, Stebbings LA, Leroy C, Edkins S, Hardy C, Teague JW, Menzies A, Goodhead I, Turner DJ, Clee CM, Quail MA, Cox A, Brown C, Durbin R, Hurles ME, Edwards PA, Bignell GR, Stratton MR, Futreal PA: Identification of somatically acquired rearrangements in cancer using genome-wide massively parallel paired-end sequencing. Nat Genet. 2008, 40: 722-729. 10.1038/ng.128." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR26" id="ref-link-section-d3015500e1357">26</a>, <a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 27" title="Holt KE, Parkhill J, Mazzoni CJ, Roumagnac P, Weill FX, Goodhead I, Rance R, Baker S, Maskell DJ, Wain J, Dolecek C, Achtman M, Dougan G: High-throughput sequencing provides insights into genome variation and evolution in Salmonella typhi. Nat Genet. 2008, 40: 987-993. 10.1038/ng.195." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR27" id="ref-link-section-d3015500e1360">27</a>] that is among the fastest competing open source tools for aligning millions of Illumina reads to the human genome. SOAP is another open source tool that has been reported and used in short-read projects [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 6" title="Wang J, Wang W, Li R, Li Y, Tian G, Goodman L, Fan W, Zhang J, Li J, Zhang J, Guo Y, Feng B, Li H, Lu Y, Fang X, Liang H, Du Z, Li D, Zhao Y, Hu Y, Yang Z, Zheng H, Hellmann I, Inouye M, Pool J, Yi X, Zhao J, Duan J, Zhou Y, Qin J, et al: The diploid genome sequence of an Asian individual. Nature. 2008, 456: 60-65. 10.1038/nature07484." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR6" id="ref-link-section-d3015500e1364">6</a>, <a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 28" title="Nagalakshmi U, Wang Z, Waern K, Shou C, Raha D, Gerstein M, Snyder M: The transcriptional landscape of the yeast genome defined by RNA sequencing. Science. 2008, 320: 1344-1349. 10.1126/science.1158441." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR28" id="ref-link-section-d3015500e1367">28</a>].</p><p>Table <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab1">1</a> presents the performance and sensitivity of Bowtie v0.9.6, SOAP v1.10, and Maq v0.6.6. SOAP could not be run on the PC because SOAP's memory footprint exceeds the PC's physical memory. The 'soap.contig' version of the SOAP binary was used. For comparison with SOAP, Bowtie was invoked with '-v 2' to mimic SOAP's default matching policy (which allows up to two mismatches in the alignment and disregards quality values), and with '--maxns 5' to simulate SOAP's default policy of filtering out reads with five or more no-confidence bases. For the Maq comparison Bowtie is run with its default policy, which mimics Maq's default policy of allowing up to two mismatches in the first 28 bases and enforcing an overall limit of 70 on the sum of the quality values at all mismatched read positions. To make Bowtie's memory footprint more comparable to Maq's, Bowtie is invoked with the '-z' option in all experiments to ensure that only the forward or mirror index is resident in memory at one time.</p><p>The number of reads aligned indicates that SOAP (67.3%) and Bowtie -v 2 (67.4%) have comparable sensitivity. Of the reads aligned by either SOAP or Bowtie, 99.7% were aligned by both, 0.2% were aligned by Bowtie but not SOAP, and 0.1% were aligned by SOAP but not Bowtie. Maq (74.7%) and Bowtie (71.9%) also have roughly comparable sensitivity, although Bowtie lags by 2.8%. Of the reads aligned by either Maq or Bowtie, 96.0% were aligned by both, 0.1% were aligned by Bowtie but not Maq, and 3.9% were aligned by Maq but not Bowtie. Of the reads mapped by Maq but not Bowtie, almost all are due to a flexibility in Maq's alignment algorithm that allows some alignments to have three mismatches in the seed. The remainder of the reads mapped by Maq but not Bowtie are due to Bowtie's backtracking ceiling.</p><p>Maq's documentation mentions that reads containing 'poly-A artifacts' can impair Maq's performance. Table <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab2">2</a> presents performance and sensitivity of Bowtie and Maq when the read set is filtered using Maq's 'catfilter' command to eliminate poly-A artifacts. The filter eliminates 438,145 out of 8,839,010 reads. Other experimental parameters are identical to those of the experiments in Table <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab1">1</a>, and the same observations about the relative sensitivity of Bowtie and Maq apply here.</p><h3 class="c-article__sub-heading" id="Sec9">Read length and performance</h3><p>As sequencing technology improves, read lengths are growing beyond the 30-bp to 50-bp commonly seen in public databases today. Bowtie, Maq, and SOAP support reads of lengths up to 1,024, 63, and 60 bp, respectively, and Maq versions 0.7.0 and later support read lengths up to 127 bp. Table <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab3">3</a> shows performance results when the three tools are each used to align three sets of 2 M untrimmed reads, a 36-bp set, a 50-bp set and a 76-bp set, to the human genome on the server platform. Each set of 2 M is randomly sampled from a larger set (NCBI Short Read Archive: SRR003084 for 36-bp, SRR003092 for 50-bp, SRR003196 for 76-bp). Reads were sampled such that the three sets of 2 M have uniform per-base error rate, as calculated from per-base Phred qualities. All reads pass through Maq's 'catfilter'.</p><div class="c-article-table" data-test="inline-table" data-container-section="table" id="table-3"><figure><figcaption class="c-article-table__figcaption"><b id="Tab3" data-test="table-caption">Table 3 Varying read length using Bowtie, Maq and SOAP</b></figcaption><div class="u-text-right u-hide-print"><a class="c-article__pill-button" data-test="table-link" data-track="click" data-track-action="view table" data-track-label="button" rel="nofollow" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/tables/3" aria-label="Full size table 3"><span>Full size table</span><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-chevron-right"></use></svg></a></div></figure></div><p>Bowtie is run both in its Maq-like default mode and in its SOAP-like '-v 2' mode. Bowtie is also given the '-z' option to ensure that only the forward or mirror index is resident in memory at one time. Maq v0.7.1 was used instead of Maq v0.6.6 for the 76-bp set because v0.6.6 cannot align reads longer than 63 bp. SOAP was not run on the 76-bp set because it does not support reads longer than 60 bp.</p><p>The results show that Maq's algorithm scales better overall to longer read lengths than Bowtie or SOAP. However, Bowtie in SOAP-like '-v 2' mode also scales very well. Bowtie in its default Maq-like mode scales well from 36-bp to 50-bp reads but is substantially slower for 76-bp reads, although it is still more than an order of magnitude faster than Maq.</p><h3 class="c-article__sub-heading" id="Sec10">Parallel performance</h3><p>Alignment can be parallelized by distributing reads across concurrent search threads. Bowtie allows the user to specify a desired number of threads (option -p); Bowtie then launches the specified number of threads using the pthreads library. Bowtie threads synchronize with each other when fetching reads, outputting results, switching between indices, and performing various forms of global bookkeeping, such as marking a read as 'done'. Otherwise, threads are free to operate in parallel, substantially speeding up alignment on computers with multiple processor cores. The memory image of the index is shared by all threads, and so the footprint does not increase substantially when multiple threads are used. Table <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab4">4</a> shows performance results for running Bowtie v0.9.6 on the four-core server with one, two, and four threads.</p><div class="c-article-table" data-test="inline-table" data-container-section="table" id="table-4"><figure><figcaption class="c-article-table__figcaption"><b id="Tab4" data-test="table-caption">Table 4 Bowtie parallel alignment performance</b></figcaption><div class="u-text-right u-hide-print"><a class="c-article__pill-button" data-test="table-link" data-track="click" data-track-action="view table" data-track-label="button" rel="nofollow" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/tables/4" aria-label="Full size table 4"><span>Full size table</span><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-chevron-right"></use></svg></a></div></figure></div><h3 class="c-article__sub-heading" id="Sec11">Index building</h3><p>Bowtie uses a flexible indexing algorithm [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 29" title="Kärkkäinen J: Fast BWT in small space by blockwise suffix sorting. Theor Comput Sci. 2007, 387: 249-257." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR29" id="ref-link-section-d3015500e2080">29</a>] that can be configured to trade off between memory usage and running time. Table <a data-track="click" data-track-label="link" data-track-action="table anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#Tab5">5</a> illustrates this trade-off when indexing the entire human reference genome (NCBI build 36.3, contigs). Runs were performed on the server platform. The indexer was run four times with different upper limits on memory usage.</p><div class="c-article-table" data-test="inline-table" data-container-section="table" id="table-5"><figure><figcaption class="c-article-table__figcaption"><b id="Tab5" data-test="table-caption">Table 5 Bowtie index building performance</b></figcaption><div class="u-text-right u-hide-print"><a class="c-article__pill-button" data-test="table-link" data-track="click" data-track-action="view table" data-track-label="button" rel="nofollow" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/tables/5" aria-label="Full size table 5"><span>Full size table</span><svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-chevron-right"></use></svg></a></div></figure></div><p>The reported times compare favorably with alignment times of competing tools that perform indexing during alignment. Less than 5 hours is required for Bowtie to both build and query a whole-human index with 8.84 million reads from the 1,000 Genome project (NCBI Short Read Archive:SRR001115) on a server, more than sixfold faster than the equivalent Maq run. The bottom-most row illustrates that the Bowtie indexer, with appropriate arguments, is memory-efficient enough to run on a typical workstation with 2 GB of RAM. Additional data file 1 (Supplementary discussions 3 and 4) explains the algorithm and the contents of the resulting index.</p><h3 class="c-article__sub-heading" id="Sec12">Software</h3><p>Bowtie is written in C++ and uses the SeqAn library [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 30" title="Doring A, Weese D, Rausch T, Reinert K: SeqAn an efficient, generic C++ library for sequence analysis. BMC Bioinformatics. 2008, 9: 11-10.1186/1471-2105-9-11." href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR30" id="ref-link-section-d3015500e2218">30</a>]. The converter to the Maq mapping format uses code from Maq.</p></div></div></section><section data-title="Discussion"><div class="c-article-section" id="Sec13-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Sec13">Discussion</h2><div class="c-article-section__content" id="Sec13-content"><p>Bowtie exhibits a large performance advantage over both Maq and SOAP when mapping reads to the human genome. Bowtie's sensitivity in terms of reads aligned is comparable to that of SOAP and slightly less than Maq's, although the user may use command-line options to trade slower running time for greater sensitivity. Unlike SOAP, Bowtie's 1.3 GB memory footprint allows it to run on a typical PC with 2 GB of RAM. Bowtie aligns Illumina reads to the human genome at a rate of over 25 million reads per hour. Multiple processor cores can run parallel Bowtie threads to achieve even greater alignment speed; experiments show a speed up of 3.12 for four threads on a typical Opteron server.</p><p>Unlike many other short-read aligners, Bowtie creates a permanent index of the reference that may be re-used across alignment runs. Building the index is fast - Bowtie outperforms competing tools when aligning lanes of Illumina reads even with index construction time included. At 2.2 GB for the human genome, the on-disk size of a Bowtie index is small enough to distribute over the internet. The Bowtie website hosts pre-built indices for the human genome and several other model organisms including chimp, dog, rat, mouse, and chicken.</p><p>Bowtie's speed and small memory footprint are due chiefly to its use of the Burrows-Wheeler index in combination with the novel, quality-aware, backtracking algorithm introduced here. Double indexing is used to avoid the performance penalty of excessive backtracking.</p><p>Bowtie supports standard FASTQ and FASTA input formats, and comes with a conversion program that allows Bowtie output to be used with Maq's consensus generator and single nucleotide polymorphism caller.</p><p>Bowtie does not yet support paired-end alignment or alignments with insertions or deletions, although both improvements are planned for the future. Paired-end alignment is not difficult to implement in Bowtie's framework, and we expect that Bowtie's performance advantage will be comparable to, though perhaps somewhat less than, that of unpaired alignment mode. Support for insertions and deletions is also a conceptually straightforward addition.</p><p>Bowtie is free, open source software available from the Bowtie website [<a data-track="click" data-track-action="reference anchor" data-track-label="link" data-test="citation-ref" aria-label="Reference 25" title="Bowtie: An ultrafast memory-efficient short read aligner. [
                    http://bowtie.cbcb.umd.edu/
                    
                  ]" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#ref-CR25" id="ref-link-section-d3015500e2241">25</a>].</p></div></div></section><section data-title="Additional data files"><div class="c-article-section" id="Sec14-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Sec14">Additional data files</h2><div class="c-article-section__content" id="Sec14-content"><p>The following additional data are included with the online version of this article: a document containing supplementary discussions, tables, and figures pertaining to algorithms for navigating the Burrows-Wheeler transform, the full four-phase version of the alignment algorithm that incorporates the reverse-complement, index construction, and components of the index (Additional data file <a data-track="click" data-track-label="link" data-track-action="supplementary material anchor" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25#MOESM1">1</a>).</p></div></div></section>
                    <section data-title="Abbreviations"><div class="c-article-section" id="abbreviations-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="abbreviations">Abbreviations</h2><div class="c-article-section__content" id="abbreviations-content"><dl class="c-abbreviation_list"><dt class="c-abbreviation_list__term u-text-bold u-float-left u-pr-16"><dfn>bp:</dfn></dt><dd class="c-abbreviation_list__description u-mb-24">
                    <p>base pair</p>
                  </dd><dt class="c-abbreviation_list__term u-text-bold u-float-left u-pr-16"><dfn>BWT:</dfn></dt><dd class="c-abbreviation_list__description u-mb-24">
                    <p>Burrows-Wheeler transform</p>
                  </dd><dt class="c-abbreviation_list__term u-text-bold u-float-left u-pr-16"><dfn>CPU:</dfn></dt><dd class="c-abbreviation_list__description u-mb-24">
                    <p>central processing unit</p>
                  </dd><dt class="c-abbreviation_list__term u-text-bold u-float-left u-pr-16"><dfn>FM:</dfn></dt><dd class="c-abbreviation_list__description u-mb-24">
                    <p>full-text minute-space</p>
                  </dd><dt class="c-abbreviation_list__term u-text-bold u-float-left u-pr-16"><dfn>GB:</dfn></dt><dd class="c-abbreviation_list__description u-mb-24">
                    <p>gigabytes</p>
                  </dd><dt class="c-abbreviation_list__term u-text-bold u-float-left u-pr-16"><dfn>LF:</dfn></dt><dd class="c-abbreviation_list__description u-mb-24">
                    <p>last first</p>
                  </dd><dt class="c-abbreviation_list__term u-text-bold u-float-left u-pr-16"><dfn>NCBI:</dfn></dt><dd class="c-abbreviation_list__description u-mb-24">
                    <p>National Center for Biotechnology Information.</p>
                  </dd></dl></div></div></section><div id="MagazineFulltextArticleBodySuffix"><section aria-labelledby="Bib1" data-title="References"><div class="c-article-section" id="Bib1-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Bib1">References</h2><div class="c-article-section__content" id="Bib1-content"><div data-container-section="references"><ol class="c-article-references" data-track-component="outbound reference"><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="1."><p class="c-article-references__text" id="ref-CR1">Down TA, Rakyan VK, Turner DJ, Flicek P, Li H, Kulesha E, Graf S, Johnson N, Herrero J, Tomazou EM, Thorne NP, Backdahl L, Herberth M, Howe KL, Jackson DK, Miretti MM, Marioni JC, Birney E, Hubbard TJ, Durbin R, Tavare S, Beck S: A Bayesian deconvolution strategy for immunoprecipitation-based DNA methylome analysis. Nat Biotechnol. 2008, 26: 779-785. 10.1038/nbt1414.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18612301" aria-label="PubMed reference 1">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXot1entL4%3D" aria-label="CAS reference 1">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2644410" aria-label="PubMed Central reference 1">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1038/nbt1414" data-track-action="article reference" href="https://doi.org/10.1038%2Fnbt1414" aria-label="Article reference 1">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 1" href="http://scholar.google.com/scholar_lookup?&amp;title=A%20Bayesian%20deconvolution%20strategy%20for%20immunoprecipitation-based%20DNA%20methylome%20analysis.&amp;journal=Nat%20Biotechnol&amp;doi=10.1038%2Fnbt1414&amp;volume=26&amp;pages=779-785&amp;publication_year=2008&amp;author=Down%2CTA&amp;author=Rakyan%2CVK&amp;author=Turner%2CDJ&amp;author=Flicek%2CP&amp;author=Li%2CH&amp;author=Kulesha%2CE&amp;author=Graf%2CS&amp;author=Johnson%2CN&amp;author=Herrero%2CJ&amp;author=Tomazou%2CEM&amp;author=Thorne%2CNP&amp;author=Backdahl%2CL&amp;author=Herberth%2CM&amp;author=Howe%2CKL&amp;author=Jackson%2CDK&amp;author=Miretti%2CMM&amp;author=Marioni%2CJC&amp;author=Birney%2CE&amp;author=Hubbard%2CTJ&amp;author=Durbin%2CR&amp;author=Tavare%2CS&amp;author=Beck%2CS">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="2."><p class="c-article-references__text" id="ref-CR2">Johnson DS, Mortazavi A, Myers RM, Wold B: Genome-wide mapping of in vivo protein-DNA interactions. Science. 2007, 316: 1497-1502. 10.1126/science.1141319.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17540862" aria-label="PubMed reference 2">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD2sXmtFSjtrg%3D" aria-label="CAS reference 2">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1126/science.1141319" data-track-action="article reference" href="https://doi.org/10.1126%2Fscience.1141319" aria-label="Article reference 2">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 2" href="http://scholar.google.com/scholar_lookup?&amp;title=Genome-wide%20mapping%20of%20in%20vivo%20protein-DNA%20interactions.&amp;journal=Science&amp;doi=10.1126%2Fscience.1141319&amp;volume=316&amp;pages=1497-1502&amp;publication_year=2007&amp;author=Johnson%2CDS&amp;author=Mortazavi%2CA&amp;author=Myers%2CRM&amp;author=Wold%2CB">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="3."><p class="c-article-references__text" id="ref-CR3">Marioni JC, Mason CE, Mane SM, Stephens M, Gilad Y: RNA-seq: an assessment of technical reproducibility and comparison with gene expression arrays. Genome Res. 2008, 18: 1509-1517. 10.1101/gr.079558.108.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18550803" aria-label="PubMed reference 3">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtV2qsr3E" aria-label="CAS reference 3">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2527709" aria-label="PubMed Central reference 3">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1101/gr.079558.108" data-track-action="article reference" href="https://doi.org/10.1101%2Fgr.079558.108" aria-label="Article reference 3">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 3" href="http://scholar.google.com/scholar_lookup?&amp;title=RNA-seq%3A%20an%20assessment%20of%20technical%20reproducibility%20and%20comparison%20with%20gene%20expression%20arrays.&amp;journal=Genome%20Res&amp;doi=10.1101%2Fgr.079558.108&amp;volume=18&amp;pages=1509-1517&amp;publication_year=2008&amp;author=Marioni%2CJC&amp;author=Mason%2CCE&amp;author=Mane%2CSM&amp;author=Stephens%2CM&amp;author=Gilad%2CY">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="4."><p class="c-article-references__text" id="ref-CR4">Bentley DR, Balasubramanian S, Swerdlow HP, Smith GP, Milton J, Brown CG, Hall KP, Evers DJ, Barnes CL, Bignell HR, Boutell JM, Bryant J, Carter RJ, Keira Cheetham R, Cox AJ, Ellis DJ, Flatbush MR, Gormley NA, Humphray SJ, Irving LJ, Karbelashvili MS, Kirk SM, Li H, Liu X, Maisinger KS, Murray LJ, Obradovic B, Ost T, Parkinson ML, Pratt MR, et al: Accurate whole human genome sequencing using reversible terminator chemistry. Nature. 2008, 456: 53-59. 10.1038/nature07517.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18987734" aria-label="PubMed reference 4">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtlCjtr3E" aria-label="CAS reference 4">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2581791" aria-label="PubMed Central reference 4">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1038/nature07517" data-track-action="article reference" href="https://doi.org/10.1038%2Fnature07517" aria-label="Article reference 4">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 4" href="http://scholar.google.com/scholar_lookup?&amp;title=Accurate%20whole%20human%20genome%20sequencing%20using%20reversible%20terminator%20chemistry.&amp;journal=Nature&amp;doi=10.1038%2Fnature07517&amp;volume=456&amp;pages=53-59&amp;publication_year=2008&amp;author=Bentley%2CDR&amp;author=Balasubramanian%2CS&amp;author=Swerdlow%2CHP&amp;author=Smith%2CGP&amp;author=Milton%2CJ&amp;author=Brown%2CCG&amp;author=Hall%2CKP&amp;author=Evers%2CDJ&amp;author=Barnes%2CCL&amp;author=Bignell%2CHR&amp;author=Boutell%2CJM&amp;author=Bryant%2CJ&amp;author=Carter%2CRJ&amp;author=Keira%20Cheetham%2CR&amp;author=Cox%2CAJ&amp;author=Ellis%2CDJ&amp;author=Flatbush%2CMR&amp;author=Gormley%2CNA&amp;author=Humphray%2CSJ&amp;author=Irving%2CLJ&amp;author=Karbelashvili%2CMS&amp;author=Kirk%2CSM&amp;author=Li%2CH&amp;author=Liu%2CX&amp;author=Maisinger%2CKS&amp;author=Murray%2CLJ&amp;author=Obradovic%2CB&amp;author=Ost%2CT&amp;author=Parkinson%2CML&amp;author=Pratt%2CMR">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="5."><p class="c-article-references__text" id="ref-CR5">Ley TJ, Mardis ER, Ding L, Fulton B, McLellan MD, Chen K, Dooling D, Dunford-Shore BH, McGrath S, Hickenbotham M, Cook L, Abbott R, Larson DE, Koboldt DC, Pohl C, Smith S, Hawkins A, Abbott S, Locke D, Hillier LW, Miner T, Fulton L, Magrini V, Wylie T, Glasscock J, Conyers J, Sander N, Shi X, Osborne JR, Minx P, et al: DNA sequencing of a cytogenetically normal acute myeloid leukaemia genome. Nature. 2008, 456: 66-72. 10.1038/nature07485.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18987736" aria-label="PubMed reference 5">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtlCjtrfE" aria-label="CAS reference 5">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2603574" aria-label="PubMed Central reference 5">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1038/nature07485" data-track-action="article reference" href="https://doi.org/10.1038%2Fnature07485" aria-label="Article reference 5">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 5" href="http://scholar.google.com/scholar_lookup?&amp;title=DNA%20sequencing%20of%20a%20cytogenetically%20normal%20acute%20myeloid%20leukaemia%20genome.&amp;journal=Nature&amp;doi=10.1038%2Fnature07485&amp;volume=456&amp;pages=66-72&amp;publication_year=2008&amp;author=Ley%2CTJ&amp;author=Mardis%2CER&amp;author=Ding%2CL&amp;author=Fulton%2CB&amp;author=McLellan%2CMD&amp;author=Chen%2CK&amp;author=Dooling%2CD&amp;author=Dunford-Shore%2CBH&amp;author=McGrath%2CS&amp;author=Hickenbotham%2CM&amp;author=Cook%2CL&amp;author=Abbott%2CR&amp;author=Larson%2CDE&amp;author=Koboldt%2CDC&amp;author=Pohl%2CC&amp;author=Smith%2CS&amp;author=Hawkins%2CA&amp;author=Abbott%2CS&amp;author=Locke%2CD&amp;author=Hillier%2CLW&amp;author=Miner%2CT&amp;author=Fulton%2CL&amp;author=Magrini%2CV&amp;author=Wylie%2CT&amp;author=Glasscock%2CJ&amp;author=Conyers%2CJ&amp;author=Sander%2CN&amp;author=Shi%2CX&amp;author=Osborne%2CJR&amp;author=Minx%2CP">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="6."><p class="c-article-references__text" id="ref-CR6">Wang J, Wang W, Li R, Li Y, Tian G, Goodman L, Fan W, Zhang J, Li J, Zhang J, Guo Y, Feng B, Li H, Lu Y, Fang X, Liang H, Du Z, Li D, Zhao Y, Hu Y, Yang Z, Zheng H, Hellmann I, Inouye M, Pool J, Yi X, Zhao J, Duan J, Zhou Y, Qin J, et al: The diploid genome sequence of an Asian individual. Nature. 2008, 456: 60-65. 10.1038/nature07484.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18987735" aria-label="PubMed reference 6">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtlCjtrnK" aria-label="CAS reference 6">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2716080" aria-label="PubMed Central reference 6">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1038/nature07484" data-track-action="article reference" href="https://doi.org/10.1038%2Fnature07484" aria-label="Article reference 6">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 6" href="http://scholar.google.com/scholar_lookup?&amp;title=The%20diploid%20genome%20sequence%20of%20an%20Asian%20individual.&amp;journal=Nature&amp;doi=10.1038%2Fnature07484&amp;volume=456&amp;pages=60-65&amp;publication_year=2008&amp;author=Wang%2CJ&amp;author=Wang%2CW&amp;author=Li%2CR&amp;author=Li%2CY&amp;author=Tian%2CG&amp;author=Goodman%2CL&amp;author=Fan%2CW&amp;author=Zhang%2CJ&amp;author=Li%2CJ&amp;author=Zhang%2CJ&amp;author=Guo%2CY&amp;author=Feng%2CB&amp;author=Li%2CH&amp;author=Lu%2CY&amp;author=Fang%2CX&amp;author=Liang%2CH&amp;author=Du%2CZ&amp;author=Li%2CD&amp;author=Zhao%2CY&amp;author=Hu%2CY&amp;author=Yang%2CZ&amp;author=Zheng%2CH&amp;author=Hellmann%2CI&amp;author=Inouye%2CM&amp;author=Pool%2CJ&amp;author=Yi%2CX&amp;author=Zhao%2CJ&amp;author=Duan%2CJ&amp;author=Zhou%2CY&amp;author=Qin%2CJ">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="7."><p class="c-article-references__text" id="ref-CR7">Li H, Ruan J, Durbin R: Mapping short DNA sequencing reads and calling variants using mapping quality scores. Genome Res. 2008, 18: 1851-1858. 10.1101/gr.078212.108.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18714091" aria-label="PubMed reference 7">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtlKhsrrN" aria-label="CAS reference 7">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2577856" aria-label="PubMed Central reference 7">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1101/gr.078212.108" data-track-action="article reference" href="https://doi.org/10.1101%2Fgr.078212.108" aria-label="Article reference 7">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 7" href="http://scholar.google.com/scholar_lookup?&amp;title=Mapping%20short%20DNA%20sequencing%20reads%20and%20calling%20variants%20using%20mapping%20quality%20scores.&amp;journal=Genome%20Res&amp;doi=10.1101%2Fgr.078212.108&amp;volume=18&amp;pages=1851-1858&amp;publication_year=2008&amp;author=Li%2CH&amp;author=Ruan%2CJ&amp;author=Durbin%2CR">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="8."><p class="c-article-references__text" id="ref-CR8">Li R, Li Y, Kristiansen K, Wang J: SOAP: short oligonucleotide alignment program. Bioinformatics. 2008, 24: 713-714. 10.1093/bioinformatics/btn025.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18227114" aria-label="PubMed reference 8">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXislKlt7Y%3D" aria-label="CAS reference 8">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1093/bioinformatics/btn025" data-track-action="article reference" href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn025" aria-label="Article reference 8">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 8" href="http://scholar.google.com/scholar_lookup?&amp;title=SOAP%3A%20short%20oligonucleotide%20alignment%20program.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2Fbtn025&amp;volume=24&amp;pages=713-714&amp;publication_year=2008&amp;author=Li%2CR&amp;author=Li%2CY&amp;author=Kristiansen%2CK&amp;author=Wang%2CJ">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="9."><p class="c-article-references__text" id="ref-CR9">Kaiser J: DNA sequencing. A plan to capture human diversity in 1000 genomes. Science. 2008, 319: 395-10.1126/science.319.5862.395.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18218868" aria-label="PubMed reference 9">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtl2qsbs%3D" aria-label="CAS reference 9">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1126/science.319.5862.395" data-track-action="article reference" href="https://doi.org/10.1126%2Fscience.319.5862.395" aria-label="Article reference 9">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 9" href="http://scholar.google.com/scholar_lookup?&amp;title=DNA%20sequencing.%20A%20plan%20to%20capture%20human%20diversity%20in%201000%20genomes.&amp;journal=Science&amp;doi=10.1126%2Fscience.319.5862.395&amp;volume=319&amp;publication_year=2008&amp;author=Kaiser%2CJ">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="10."><p class="c-article-references__text" id="ref-CR10">Smith AD, Xuan Z, Zhang MQ: Using quality scores and longer reads improves accuracy of Solexa read mapping. BMC Bioinformatics. 2008, 9: 128-10.1186/1471-2105-9-128.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18307793" aria-label="PubMed reference 10">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2335322" aria-label="PubMed Central reference 10">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1186/1471-2105-9-128" data-track-action="article reference" href="https://doi.org/10.1186%2F1471-2105-9-128" aria-label="Article reference 10">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 10" href="http://scholar.google.com/scholar_lookup?&amp;title=Using%20quality%20scores%20and%20longer%20reads%20improves%20accuracy%20of%20Solexa%20read%20mapping.&amp;journal=BMC%20Bioinformatics&amp;doi=10.1186%2F1471-2105-9-128&amp;volume=9&amp;publication_year=2008&amp;author=Smith%2CAD&amp;author=Xuan%2CZ&amp;author=Zhang%2CMQ">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="11."><p class="c-article-references__text" id="ref-CR11">Lin H, Zhang Z, Zhang MQ, Ma B, Li M: ZOOM! Zillions Of Oligos Mapped. Bioinformatics. 2008, 24: 2431-2437. 10.1093/bioinformatics/btn416.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18684737" aria-label="PubMed reference 11">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXht12gtbrP" aria-label="CAS reference 11">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2732274" aria-label="PubMed Central reference 11">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1093/bioinformatics/btn416" data-track-action="article reference" href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn416" aria-label="Article reference 11">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 11" href="http://scholar.google.com/scholar_lookup?&amp;title=ZOOM%21%20Zillions%20Of%20Oligos%20Mapped.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2Fbtn416&amp;volume=24&amp;pages=2431-2437&amp;publication_year=2008&amp;author=Lin%2CH&amp;author=Zhang%2CZ&amp;author=Zhang%2CMQ&amp;author=Ma%2CB&amp;author=Li%2CM">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="12."><p class="c-article-references__text" id="ref-CR12">SHRiMP - SHort Read Mapping Package. [<a href="http://compbio.cs.toronto.edu/shrimp/">http://compbio.cs.toronto.edu/shrimp/</a>]</p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="13."><p class="c-article-references__text" id="ref-CR13">Baeza-Yates RA, Perleberg CH: Fast and practical approximate string matching. Inf Process Lett. 1996, 59: 21-27. 10.1016/0020-0190(96)00083-X.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="10.1016/0020-0190(96)00083-X" data-track-action="article reference" href="https://doi.org/10.1016%2F0020-0190%2896%2900083-X" aria-label="Article reference 13">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 13" href="http://scholar.google.com/scholar_lookup?&amp;title=Fast%20and%20practical%20approximate%20string%20matching.&amp;journal=Inf%20Process%20Lett&amp;doi=10.1016%2F0020-0190%2896%2900083-X&amp;volume=59&amp;pages=21-27&amp;publication_year=1996&amp;author=Baeza-Yates%2CRA&amp;author=Perleberg%2CCH">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="14."><p class="c-article-references__text" id="ref-CR14">Burkhardt S, Kärkkäinen J: Better Filtering with Gapped q-Grams. Fundam Inf. 2003, 56: 51-70.</p><p class="c-article-references__links u-hide-print"><a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 14" href="http://scholar.google.com/scholar_lookup?&amp;title=Better%20Filtering%20with%20Gapped%20q-Grams.&amp;journal=Fundam%20Inf&amp;volume=56&amp;pages=51-70&amp;publication_year=2003&amp;author=Burkhardt%2CS&amp;author=K%C3%A4rkk%C3%A4inen%2CJ">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="15."><p class="c-article-references__text" id="ref-CR15">Ma B, Tromp J, Li M: PatternHunter: faster and more sensitive homology search. Bioinformatics. 2002, 18: 440-445. 10.1093/bioinformatics/18.3.440.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=11934743" aria-label="PubMed reference 15">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD38XivFKrsrY%3D" aria-label="CAS reference 15">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1093/bioinformatics/18.3.440" data-track-action="article reference" href="https://doi.org/10.1093%2Fbioinformatics%2F18.3.440" aria-label="Article reference 15">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 15" href="http://scholar.google.com/scholar_lookup?&amp;title=PatternHunter%3A%20faster%20and%20more%20sensitive%20homology%20search.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2F18.3.440&amp;volume=18&amp;pages=440-445&amp;publication_year=2002&amp;author=Ma%2CB&amp;author=Tromp%2CJ&amp;author=Li%2CM">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="16."><p class="c-article-references__text" id="ref-CR16">Smith TF, Waterman MS: Identification of common molecular subsequences. J Mol Biol. 1981, 147: 195-197. 10.1016/0022-2836(81)90087-5.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=7265238" aria-label="PubMed reference 16">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BC3cXitlCnsLw%3D" aria-label="CAS reference 16">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1016/0022-2836(81)90087-5" data-track-action="article reference" href="https://doi.org/10.1016%2F0022-2836%2881%2990087-5" aria-label="Article reference 16">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 16" href="http://scholar.google.com/scholar_lookup?&amp;title=Identification%20of%20common%20molecular%20subsequences.&amp;journal=J%20Mol%20Biol&amp;doi=10.1016%2F0022-2836%2881%2990087-5&amp;volume=147&amp;pages=195-197&amp;publication_year=1981&amp;author=Smith%2CTF&amp;author=Waterman%2CMS">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="17."><p class="c-article-references__text" id="ref-CR17">Burrows M, Wheeler DJ: A Block Sorting Lossless Data Compression Algorithm. Technical Report 124. 1994, Palo Alto, CA: Digital Equipment Corporation</p><p class="c-article-references__links u-hide-print"><a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 17" href="http://scholar.google.com/scholar_lookup?&amp;title=A%20Block%20Sorting%20Lossless%20Data%20Compression%20Algorithm.%20Technical%20Report%20124&amp;publication_year=1994&amp;author=Burrows%2CM&amp;author=Wheeler%2CDJ">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="18."><p class="c-article-references__text" id="ref-CR18">Ferragina P, Manzini G: Opportunistic data structures with applications. [<a href="http://web.unipmn.it/~manzini/papers/focs00draft.pdf">http://web.unipmn.it/~manzini/papers/focs00draft.pdf</a>]</p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="19."><p class="c-article-references__text" id="ref-CR19">Ferragina P, Manzini G: An experimental study of an opportunistic index. Proceedings of the Twelfth Annual ACM-SIAM Symposium on Discrete algorithms. 2001, Washington, DC: Society for Industrial and Applied Mathematics, 269-278.</p><p class="c-article-references__links u-hide-print"><a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 19" href="http://scholar.google.com/scholar_lookup?&amp;title=An%20experimental%20study%20of%20an%20opportunistic%20index.&amp;pages=269-278&amp;publication_year=2001&amp;author=Ferragina%2CP&amp;author=Manzini%2CG">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="20."><p class="c-article-references__text" id="ref-CR20">Healy J, Thomas EE, Schwartz JT, Wigler M: Annotating large genomes with exact word matches. Genome Res. 2003, 13: 2306-2315. 10.1101/gr.1350803.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12975312" aria-label="PubMed reference 20">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD3sXotF2hsbw%3D" aria-label="CAS reference 20">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC403711" aria-label="PubMed Central reference 20">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1101/gr.1350803" data-track-action="article reference" href="https://doi.org/10.1101%2Fgr.1350803" aria-label="Article reference 20">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 20" href="http://scholar.google.com/scholar_lookup?&amp;title=Annotating%20large%20genomes%20with%20exact%20word%20matches.&amp;journal=Genome%20Res&amp;doi=10.1101%2Fgr.1350803&amp;volume=13&amp;pages=2306-2315&amp;publication_year=2003&amp;author=Healy%2CJ&amp;author=Thomas%2CEE&amp;author=Schwartz%2CJT&amp;author=Wigler%2CM">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="21."><p class="c-article-references__text" id="ref-CR21">Lippert RA: Space-efficient whole genome comparisons with Burrows-Wheeler transforms. J Comput Biol. 2005, 12: 407-415. 10.1089/cmb.2005.12.407.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=15882139" aria-label="PubMed reference 21">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD2MXjvFeqsL0%3D" aria-label="CAS reference 21">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1089/cmb.2005.12.407" data-track-action="article reference" href="https://doi.org/10.1089%2Fcmb.2005.12.407" aria-label="Article reference 21">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 21" href="http://scholar.google.com/scholar_lookup?&amp;title=Space-efficient%20whole%20genome%20comparisons%20with%20Burrows-Wheeler%20transforms.&amp;journal=J%20Comput%20Biol&amp;doi=10.1089%2Fcmb.2005.12.407&amp;volume=12&amp;pages=407-415&amp;publication_year=2005&amp;author=Lippert%2CRA">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="22."><p class="c-article-references__text" id="ref-CR22">Graf S, Nielsen FG, Kurtz S, Huynen MA, Birney E, Stunnenberg H, Flicek P: Optimized design and assessment of whole genome tiling arrays. Bioinformatics. 2007, 23: i195-i204. 10.1093/bioinformatics/btm200.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17646297" aria-label="PubMed reference 22">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1093/bioinformatics/btm200" data-track-action="article reference" href="https://doi.org/10.1093%2Fbioinformatics%2Fbtm200" aria-label="Article reference 22">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 22" href="http://scholar.google.com/scholar_lookup?&amp;title=Optimized%20design%20and%20assessment%20of%20whole%20genome%20tiling%20arrays.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2Fbtm200&amp;volume=23&amp;pages=i195-i204&amp;publication_year=2007&amp;author=Graf%2CS&amp;author=Nielsen%2CFG&amp;author=Kurtz%2CS&amp;author=Huynen%2CMA&amp;author=Birney%2CE&amp;author=Stunnenberg%2CH&amp;author=Flicek%2CP">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="23."><p class="c-article-references__text" id="ref-CR23">Lam TW, Sung WK, Tam SL, Wong CK, Yiu SM: Compressed indexing and local alignment of DNA. Bioinformatics. 2008, 24: 791-797. 10.1093/bioinformatics/btn032.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18227115" aria-label="PubMed reference 23">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXjsVaktL8%3D" aria-label="CAS reference 23">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1093/bioinformatics/btn032" data-track-action="article reference" href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn032" aria-label="Article reference 23">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 23" href="http://scholar.google.com/scholar_lookup?&amp;title=Compressed%20indexing%20and%20local%20alignment%20of%20DNA.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2Fbtn032&amp;volume=24&amp;pages=791-797&amp;publication_year=2008&amp;author=Lam%2CTW&amp;author=Sung%2CWK&amp;author=Tam%2CSL&amp;author=Wong%2CCK&amp;author=Yiu%2CSM">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="24."><p class="c-article-references__text" id="ref-CR24">Ewing B, Green P: Base-calling of automated sequencer traces using phred. II. Error probabilities. Genome Res. 1998, 8: 186-194.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=9521922" aria-label="PubMed reference 24">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DyaK1cXitlWlu7g%3D" aria-label="CAS reference 24">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1101/gr.8.3.186" data-track-action="article reference" href="https://doi.org/10.1101%2Fgr.8.3.186" aria-label="Article reference 24">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 24" href="http://scholar.google.com/scholar_lookup?&amp;title=Base-calling%20of%20automated%20sequencer%20traces%20using%20phred.%20II.%20Error%20probabilities.&amp;journal=Genome%20Res&amp;doi=10.1101%2Fgr.8.3.186&amp;volume=8&amp;pages=186-194&amp;publication_year=1998&amp;author=Ewing%2CB&amp;author=Green%2CP">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="25."><p class="c-article-references__text" id="ref-CR25">Bowtie: An ultrafast memory-efficient short read aligner. [<a href="http://bowtie.cbcb.umd.edu/">http://bowtie.cbcb.umd.edu/</a>]</p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="26."><p class="c-article-references__text" id="ref-CR26">Campbell PJ, Stephens PJ, Pleasance ED, O'Meara S, Li H, Santarius T, Stebbings LA, Leroy C, Edkins S, Hardy C, Teague JW, Menzies A, Goodhead I, Turner DJ, Clee CM, Quail MA, Cox A, Brown C, Durbin R, Hurles ME, Edwards PA, Bignell GR, Stratton MR, Futreal PA: Identification of somatically acquired rearrangements in cancer using genome-wide massively parallel paired-end sequencing. Nat Genet. 2008, 40: 722-729. 10.1038/ng.128.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18438408" aria-label="PubMed reference 26">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXmsVejsL4%3D" aria-label="CAS reference 26">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2705838" aria-label="PubMed Central reference 26">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1038/ng.128" data-track-action="article reference" href="https://doi.org/10.1038%2Fng.128" aria-label="Article reference 26">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 26" href="http://scholar.google.com/scholar_lookup?&amp;title=Identification%20of%20somatically%20acquired%20rearrangements%20in%20cancer%20using%20genome-wide%20massively%20parallel%20paired-end%20sequencing.&amp;journal=Nat%20Genet&amp;doi=10.1038%2Fng.128&amp;volume=40&amp;pages=722-729&amp;publication_year=2008&amp;author=Campbell%2CPJ&amp;author=Stephens%2CPJ&amp;author=Pleasance%2CED&amp;author=O%27Meara%2CS&amp;author=Li%2CH&amp;author=Santarius%2CT&amp;author=Stebbings%2CLA&amp;author=Leroy%2CC&amp;author=Edkins%2CS&amp;author=Hardy%2CC&amp;author=Teague%2CJW&amp;author=Menzies%2CA&amp;author=Goodhead%2CI&amp;author=Turner%2CDJ&amp;author=Clee%2CCM&amp;author=Quail%2CMA&amp;author=Cox%2CA&amp;author=Brown%2CC&amp;author=Durbin%2CR&amp;author=Hurles%2CME&amp;author=Edwards%2CPA&amp;author=Bignell%2CGR&amp;author=Stratton%2CMR&amp;author=Futreal%2CPA">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="27."><p class="c-article-references__text" id="ref-CR27">Holt KE, Parkhill J, Mazzoni CJ, Roumagnac P, Weill FX, Goodhead I, Rance R, Baker S, Maskell DJ, Wain J, Dolecek C, Achtman M, Dougan G: High-throughput sequencing provides insights into genome variation and evolution in <i>Salmonella typhi</i>. Nat Genet. 2008, 40: 987-993. 10.1038/ng.195.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18660809" aria-label="PubMed reference 27">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXptVKkurk%3D" aria-label="CAS reference 27">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2652037" aria-label="PubMed Central reference 27">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1038/ng.195" data-track-action="article reference" href="https://doi.org/10.1038%2Fng.195" aria-label="Article reference 27">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 27" href="http://scholar.google.com/scholar_lookup?&amp;title=High-throughput%20sequencing%20provides%20insights%20into%20genome%20variation%20and%20evolution%20in%20Salmonella%20typhi.&amp;journal=Nat%20Genet&amp;doi=10.1038%2Fng.195&amp;volume=40&amp;pages=987-993&amp;publication_year=2008&amp;author=Holt%2CKE&amp;author=Parkhill%2CJ&amp;author=Mazzoni%2CCJ&amp;author=Roumagnac%2CP&amp;author=Weill%2CFX&amp;author=Goodhead%2CI&amp;author=Rance%2CR&amp;author=Baker%2CS&amp;author=Maskell%2CDJ&amp;author=Wain%2CJ&amp;author=Dolecek%2CC&amp;author=Achtman%2CM&amp;author=Dougan%2CG">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="28."><p class="c-article-references__text" id="ref-CR28">Nagalakshmi U, Wang Z, Waern K, Shou C, Raha D, Gerstein M, Snyder M: The transcriptional landscape of the yeast genome defined by RNA sequencing. Science. 2008, 320: 1344-1349. 10.1126/science.1158441.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18451266" aria-label="PubMed reference 28">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="cas reference" href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXmslWgtbY%3D" aria-label="CAS reference 28">CAS</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2951732" aria-label="PubMed Central reference 28">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1126/science.1158441" data-track-action="article reference" href="https://doi.org/10.1126%2Fscience.1158441" aria-label="Article reference 28">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 28" href="http://scholar.google.com/scholar_lookup?&amp;title=The%20transcriptional%20landscape%20of%20the%20yeast%20genome%20defined%20by%20RNA%20sequencing.&amp;journal=Science&amp;doi=10.1126%2Fscience.1158441&amp;volume=320&amp;pages=1344-1349&amp;publication_year=2008&amp;author=Nagalakshmi%2CU&amp;author=Wang%2CZ&amp;author=Waern%2CK&amp;author=Shou%2CC&amp;author=Raha%2CD&amp;author=Gerstein%2CM&amp;author=Snyder%2CM">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="29."><p class="c-article-references__text" id="ref-CR29">Kärkkäinen J: Fast BWT in small space by blockwise suffix sorting. Theor Comput Sci. 2007, 387: 249-257.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="10.1016/j.tcs.2007.07.018" data-track-action="article reference" href="https://doi.org/10.1016%2Fj.tcs.2007.07.018" aria-label="Article reference 29">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 29" href="http://scholar.google.com/scholar_lookup?&amp;title=Fast%20BWT%20in%20small%20space%20by%20blockwise%20suffix%20sorting.&amp;journal=Theor%20Comput%20Sci&amp;doi=10.1016%2Fj.tcs.2007.07.018&amp;volume=387&amp;pages=249-257&amp;publication_year=2007&amp;author=K%C3%A4rkk%C3%A4inen%2CJ">
                    Google Scholar</a>&nbsp;
                </p></li><li class="c-article-references__item js-c-reading-companion-references-item" data-counter="30."><p class="c-article-references__text" id="ref-CR30">Doring A, Weese D, Rausch T, Reinert K: SeqAn an efficient, generic C++ library for sequence analysis. BMC Bioinformatics. 2008, 9: 11-10.1186/1471-2105-9-11.</p><p class="c-article-references__links u-hide-print"><a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed reference" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18184432" aria-label="PubMed reference 30">PubMed</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="link" data-track-action="pubmed central reference" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2246154" aria-label="PubMed Central reference 30">PubMed Central</a>&nbsp;
    <a data-track="click" rel="nofollow noopener" data-track-label="10.1186/1471-2105-9-11" data-track-action="article reference" href="https://doi.org/10.1186%2F1471-2105-9-11" aria-label="Article reference 30">Article</a>&nbsp;
    <a data-track="click" data-track-action="google scholar reference" data-track-label="link" rel="nofollow noopener" aria-label="Google Scholar reference 30" href="http://scholar.google.com/scholar_lookup?&amp;title=SeqAn%20an%20efficient%2C%20generic%20C%2B%2B%20library%20for%20sequence%20analysis.&amp;journal=BMC%20Bioinformatics&amp;doi=10.1186%2F1471-2105-9-11&amp;volume=9&amp;publication_year=2008&amp;author=Doring%2CA&amp;author=Weese%2CD&amp;author=Rausch%2CT&amp;author=Reinert%2CK">
                    Google Scholar</a>&nbsp;
                </p></li></ol><p class="c-article-references__download u-hide-print"><a data-track="click" data-track-action="download citation references" data-track-label="link" href="https://citation-needed.springer.com/v2/references/10.1186/gb-2009-10-3-r25?format=refman&amp;flavour=references">Download references<svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-download"></use></svg></a></p></div></div></div></section></div><section data-title="Acknowledgements"><div class="c-article-section" id="Ack1-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Ack1">Acknowledgements</h2><div class="c-article-section__content" id="Ack1-content"><p>The authors would like to thank Arthur Delcher for his thorough review of the manuscript, and Michael Schatz for helpful discussion of algorithms. This research was supported in part by NIH grants R01-LM006845 and R01-GM083873 to SLS.</p></div></div></section><section aria-labelledby="author-information" data-title="Author information"><div class="c-article-section" id="author-information-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="author-information">Author information</h2><div class="c-article-section__content" id="author-information-content"><h3 class="c-article__sub-heading" id="affiliations">Authors and Affiliations</h3><ol class="c-article-author-affiliation__list"><li id="Aff1"><p class="c-article-author-affiliation__address">Center for Bioinformatics and Computational Biology, Institute for Advanced Computer Studies, University of Maryland, College Park, MD, 20742, USA</p><p class="c-article-author-affiliation__authors-list">Ben Langmead,&nbsp;Cole Trapnell,&nbsp;Mihai Pop&nbsp;&amp;&nbsp;Steven L Salzberg</p></li></ol><div class="u-js-hide u-hide-print" data-test="author-info"><span class="c-article__sub-heading">Authors</span><ol class="c-article-authors-search u-list-reset"><li id="auth-Ben-Langmead"><span class="c-article-authors-search__title u-h3 js-search-name">Ben Langmead</span><div class="c-article-authors-search__list"><div class="c-article-authors-search__item c-article-authors-search__list-item--left"><a href="https://www.biomedcentral.com/search?query=author%23Ben%20Langmead" class="c-article-button" data-track="click" data-track-action="author link - publication" data-track-label="link" rel="nofollow">View author publications</a></div><div class="c-article-authors-search__item c-article-authors-search__list-item--right"><p class="search-in-title-js c-article-authors-search__text">You can also search for this author in
                        <span class="c-article-identifiers"><a class="c-article-identifiers__item" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=Ben%20Langmead" data-track="click" data-track-action="author link - pubmed" data-track-label="link" rel="nofollow">PubMed</a><span class="u-hide">&nbsp;</span><a class="c-article-identifiers__item" href="http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22Ben%20Langmead%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en" data-track="click" data-track-action="author link - scholar" data-track-label="link" rel="nofollow">Google Scholar</a></span></p></div></div></li><li id="auth-Cole-Trapnell"><span class="c-article-authors-search__title u-h3 js-search-name">Cole Trapnell</span><div class="c-article-authors-search__list"><div class="c-article-authors-search__item c-article-authors-search__list-item--left"><a href="https://www.biomedcentral.com/search?query=author%23Cole%20Trapnell" class="c-article-button" data-track="click" data-track-action="author link - publication" data-track-label="link" rel="nofollow">View author publications</a></div><div class="c-article-authors-search__item c-article-authors-search__list-item--right"><p class="search-in-title-js c-article-authors-search__text">You can also search for this author in
                        <span class="c-article-identifiers"><a class="c-article-identifiers__item" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=Cole%20Trapnell" data-track="click" data-track-action="author link - pubmed" data-track-label="link" rel="nofollow">PubMed</a><span class="u-hide">&nbsp;</span><a class="c-article-identifiers__item" href="http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22Cole%20Trapnell%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en" data-track="click" data-track-action="author link - scholar" data-track-label="link" rel="nofollow">Google Scholar</a></span></p></div></div></li><li id="auth-Mihai-Pop"><span class="c-article-authors-search__title u-h3 js-search-name">Mihai Pop</span><div class="c-article-authors-search__list"><div class="c-article-authors-search__item c-article-authors-search__list-item--left"><a href="https://www.biomedcentral.com/search?query=author%23Mihai%20Pop" class="c-article-button" data-track="click" data-track-action="author link - publication" data-track-label="link" rel="nofollow">View author publications</a></div><div class="c-article-authors-search__item c-article-authors-search__list-item--right"><p class="search-in-title-js c-article-authors-search__text">You can also search for this author in
                        <span class="c-article-identifiers"><a class="c-article-identifiers__item" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=Mihai%20Pop" data-track="click" data-track-action="author link - pubmed" data-track-label="link" rel="nofollow">PubMed</a><span class="u-hide">&nbsp;</span><a class="c-article-identifiers__item" href="http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22Mihai%20Pop%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en" data-track="click" data-track-action="author link - scholar" data-track-label="link" rel="nofollow">Google Scholar</a></span></p></div></div></li><li id="auth-Steven_L-Salzberg"><span class="c-article-authors-search__title u-h3 js-search-name">Steven L Salzberg</span><div class="c-article-authors-search__list"><div class="c-article-authors-search__item c-article-authors-search__list-item--left"><a href="https://www.biomedcentral.com/search?query=author%23Steven%20L%20Salzberg" class="c-article-button" data-track="click" data-track-action="author link - publication" data-track-label="link" rel="nofollow">View author publications</a></div><div class="c-article-authors-search__item c-article-authors-search__list-item--right"><p class="search-in-title-js c-article-authors-search__text">You can also search for this author in
                        <span class="c-article-identifiers"><a class="c-article-identifiers__item" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=search&amp;term=Steven%20L%20Salzberg" data-track="click" data-track-action="author link - pubmed" data-track-label="link" rel="nofollow">PubMed</a><span class="u-hide">&nbsp;</span><a class="c-article-identifiers__item" href="http://scholar.google.co.uk/scholar?as_q=&amp;num=10&amp;btnG=Search+Scholar&amp;as_epq=&amp;as_oq=&amp;as_eq=&amp;as_occt=any&amp;as_sauthors=%22Steven%20L%20Salzberg%22&amp;as_publication=&amp;as_ylo=&amp;as_yhi=&amp;as_allsubj=all&amp;hl=en" data-track="click" data-track-action="author link - scholar" data-track-label="link" rel="nofollow">Google Scholar</a></span></p></div></div></li></ol></div><h3 class="c-article__sub-heading" id="corresponding-author">Corresponding author</h3><p id="corresponding-author-list">Correspondence to
                <a id="corresp-c1" href="mailto:langmead@cs.umd.edu">Ben Langmead</a>.</p></div></div></section><section data-title="Additional information"><div class="c-article-section" id="additional-information-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="additional-information">Additional information</h2><div class="c-article-section__content" id="additional-information-content"><h3 class="c-article__sub-heading">Authors' contributions</h3><p>BL developed the algorithms, collected results, and wrote most of the software. CT wrote some of the software. CT and MP contributed to discussions on algorithms. BL, CT, MP, and SLS wrote the manuscript.</p></div></div></section><section data-title="Electronic supplementary material"><div class="c-article-section" id="Sec15-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Sec15">Electronic supplementary material</h2><div class="c-article-section__content" id="Sec15-content"><div data-test="supplementary-info"><div id="figshareContainer" class="c-article-figshare-container" data-test="figshare-container"></div><div class="c-article-supplementary__item" data-test="supp-item" id="MOESM1"><h3 class="c-article-supplementary__title u-h3"><a class="print-link" data-track="click" data-track-action="view supplementary info" data-track-label="link" data-test="supp-info-link" href="https://static-content.springer.com/esm/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_2175_MOESM1_ESM.doc" data-supp-info-image="">13059_2008_2175_MOESM1_ESM.doc</a></h3><div class="c-article-supplementary__description" data-component="thumbnail-container"><p>Additional data file 1: Presented are supplementary discussions, tables, and figures pertaining to algorithms for navigating the Burrows-Wheeler transform, the full four-phase version of the alignment algorithm that incorporates the reverse-complement, index construction, and components of the index. (DOC 167 KB)</p></div></div></div></div></div></section><section data-title="Authors’ original submitted files for images"><div class="c-article-section" id="Sec16-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="Sec16">Authors’ original submitted files for images</h2><div class="c-article-section__content" id="Sec16-content"><div data-test="supplementary-info"><p>Below are the links to the authors’ original submitted files for images.</p><div class="c-article-supplementary__item" data-test="supp-item" id="MOESM2"><h3 class="c-article-supplementary__title u-h3"><a class="print-link" data-track="click" data-track-action="view supplementary info" data-track-label="link" data-test="supp-info-link" href="https://static-content.springer.com/esm/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_2175_MOESM2_ESM.eps" data-supp-info-image="">Authors’ original file for figure 1</a></h3></div><div class="c-article-supplementary__item" data-test="supp-item" id="MOESM3"><h3 class="c-article-supplementary__title u-h3"><a class="print-link" data-track="click" data-track-action="view supplementary info" data-track-label="link" data-test="supp-info-link" href="https://static-content.springer.com/esm/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_2175_MOESM3_ESM.eps" data-supp-info-image="">Authors’ original file for figure 2</a></h3></div><div class="c-article-supplementary__item" data-test="supp-item" id="MOESM4"><h3 class="c-article-supplementary__title u-h3"><a class="print-link" data-track="click" data-track-action="view supplementary info" data-track-label="link" data-test="supp-info-link" href="https://static-content.springer.com/esm/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_2175_MOESM4_ESM.eps" data-supp-info-image="">Authors’ original file for figure 3</a></h3></div></div></div></div></section><section data-title="Rights and permissions"><div class="c-article-section" id="rightslink-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="rightslink">Rights and permissions</h2><div class="c-article-section__content" id="rightslink-content"><p class="c-article-rights"><a data-track="click" data-track-action="view rights and permissions" data-track-label="link" href="https://s100.copyright.com/AppDispatchServlet?title=Ultrafast%20and%20memory-efficient%20alignment%20of%20short%20DNA%20sequences%20to%20the%20human%20genome&amp;author=Ben%20Langmead%20et%20al&amp;contentID=10.1186%2Fgb-2009-10-3-r25&amp;copyright=Langmead%20et%20al.%3B%20licensee%20BioMed%20Central%20Ltd.&amp;publication=1474-760X&amp;publicationDate=2009-03-04&amp;publisherName=SpringerNature&amp;orderBeanReset=true">Reprints and Permissions</a></p></div></div></section><section aria-labelledby="article-info" data-title="About this article"><div class="c-article-section" id="article-info-section"><h2 class="c-article-section__title js-section-title js-c-reading-companion-sections-item" id="article-info">About this article</h2><div class="c-article-section__content" id="article-info-content"><div class="c-bibliographic-information"><div class="c-bibliographic-information__column"><h3 class="c-article__sub-heading" id="citeas">Cite this article</h3><p class="c-bibliographic-information__citation">Langmead, B., Trapnell, C., Pop, M. <i>et al.</i> Ultrafast and memory-efficient alignment of short DNA sequences to the human genome.
                    <i>Genome Biol</i> <b>10</b>, R25 (2009). https://doi.org/10.1186/gb-2009-10-3-r25</p><p class="c-bibliographic-information__download-citation u-hide-print"><a data-test="citation-link" data-track="click" data-track-action="download article citation" data-track-label="link" data-track-external="" href="https://citation-needed.springer.com/v2/references/10.1186/gb-2009-10-3-r25?format=refman&amp;flavour=citation">Download citation<svg width="16" height="16" focusable="false" role="img" aria-hidden="true" class="u-icon"><use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#global-icon-download"></use></svg></a></p><ul class="c-bibliographic-information__list" data-test="publication-history"><li class="c-bibliographic-information__list-item"><p>Received<span class="u-hide">: </span><span class="c-bibliographic-information__value"><time datetime="2008-10-21">21 October 2008</time></span></p></li><li class="c-bibliographic-information__list-item"><p>Revised<span class="u-hide">: </span><span class="c-bibliographic-information__value"><time datetime="2008-12-19">19 December 2008</time></span></p></li><li class="c-bibliographic-information__list-item"><p>Accepted<span class="u-hide">: </span><span class="c-bibliographic-information__value"><time datetime="2009-03-04">04 March 2009</time></span></p></li><li class="c-bibliographic-information__list-item"><p>Published<span class="u-hide">: </span><span class="c-bibliographic-information__value"><time datetime="2009-03-04">04 March 2009</time></span></p></li><li class="c-bibliographic-information__list-item c-bibliographic-information__list-item--doi"><p><abbr title="Digital Object Identifier">DOI</abbr><span class="u-hide">: </span><span class="c-bibliographic-information__value">https://doi.org/10.1186/gb-2009-10-3-r25</span></p></li></ul><div data-component="share-box"><div class="c-article-share-box u-display-inline"><h3 class="c-article__sub-heading">Share this article</h3><p class="c-article-share-box__description">Anyone you share the following link with will be able to read this content:</p><button class="js-get-share-url c-article-share-box__button" id="get-share-url" data-track="click" data-track-label="button" data-track-external="" data-track-action="get shareable link">Get shareable link</button><div class="js-no-share-url-container u-display-none" hidden=""><p class="js-c-article-share-box__no-sharelink-info c-article-share-box__no-sharelink-info">Sorry, a shareable link is not currently available for this article.</p></div><div class="js-share-url-container u-display-none" hidden=""><p class="js-share-url c-article-share-box__only-read-input" id="share-url" data-track="click" data-track-label="button" data-track-action="select share url"></p><button class="js-copy-share-url c-article-share-box__button--link-like" id="copy-share-url" data-track="click" data-track-label="button" data-track-action="copy share url" data-track-external="">Copy to clipboard</button></div><p class="js-c-article-share-box__additional-info c-article-share-box__additional-info">
                            Provided by the Springer Nature SharedIt content-sharing initiative
                        </p></div></div><h3 class="c-article__sub-heading">Keywords</h3><ul class="c-article-subject-list"><li class="c-article-subject-list__subject"><span>Additional Data File</span></li><li class="c-article-subject-list__subject"><span>Central Processing Unit</span></li><li class="c-article-subject-list__subject"><span>Human Reference Genome</span></li><li class="c-article-subject-list__subject"><span>Memory Footprint</span></li><li class="c-article-subject-list__subject"><span>NCBI Short Read Archive</span></li></ul><div data-component="article-info-list"></div></div></div></div></div></section>

                    


                </article>
            </main>

            <div class="c-article-extras u-text-sm u-hide-print" data-container-type="reading-companion" data-track-component="reading companion">
                <aside>
                    <div data-test="download-article-link-wrapper" class="js-context-bar-sticky-point-desktop">
                        
    <div class="c-pdf-download u-clear-both">
        <a href="https://genomebiology.biomedcentral.com/counter/pdf/10.1186/gb-2009-10-3-r25.pdf" class="u-button u-button--full-width u-button--primary u-justify-content-space-between c-pdf-download__link" data-article-pdf="true" data-readcube-pdf-url="true" data-test="pdf-link" data-draft-ignore="true" data-track="click" data-track-action="download pdf" data-track-label="link" data-track-external="" download="">
            <span class="c-pdf-download__text">Download PDF</span>
            <svg aria-hidden="true" focusable="false" width="16" height="16" class="u-icon"><use xlink:href="#global-icon-download"></use></svg>
        </a>
    </div>

                        
                    </div>

                    
    
        <div class="c-article-associated-content__container">
            <section>
                <h2 class="c-article-associated-content__title u-mb-24">Associated Content</h2>
                
                    <div class="c-article-associated-content__collection collection u-mb-24">
                        <section>
                            <p class="c-article-associated-content__collection-label u-sans-serif u-text-bold u-mb-8">Collection</p>
                            <h3 class="c-article-associated-content__collection-title u-h3 u-mb-8">
                                <a href="https://www.biomedcentral.com/collections/TwentyGB" class="u-link-inherit" data-track="click" data-track-action="view collection" data-track-label="link" data-test="collection-link">Twenty years with Genome Biology</a>
                            </h3>
                        </section>
                    </div>
                
                
            </section>
        </div>
    


                    <div class="c-reading-companion">
                        <div class="c-reading-companion__sticky" data-component="reading-companion-sticky" data-test="reading-companion-sticky" style="top: 40px; width: 389.375px;">
                            <ul class="c-reading-companion__tabs" role="tablist"><li role="presentation"><button data-tab-target="sections" role="tab" id="tab-sections" aria-controls="tabpanel-sections" aria-selected="true" class="c-reading-companion__tab c-reading-companion__tab--active" data-track="click" data-track-action="sections tab" data-track-label="tab">Sections</button></li><li role="presentation"><button data-tab-target="figures" role="tab" id="tab-figures" aria-controls="tabpanel-figures" aria-selected="false" tabindex="-1" class="c-reading-companion__tab" data-track="click" data-track-action="figures tab" data-track-label="tab">Figures</button></li><li role="presentation"><button data-tab-target="references" role="tab" id="tab-references" aria-controls="tabpanel-references" aria-selected="false" tabindex="-1" class="c-reading-companion__tab" data-track="click" data-track-action="references tab" data-track-label="tab">References</button></li></ul><div class="c-reading-companion__panel c-reading-companion__sections c-reading-companion__panel--active" id="tabpanel-sections" aria-labelledby="tab-sections" role="tabpanel"><div class="c-reading-companion__scroll-pane" style="max-height: none;"><ul class="c-reading-companion__sections-list"><li id="rc-sec-Abs1" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Abs1" data-track="click" data-track-action="section anchor" data-track-label="link:Abstract">Abstract</a></li><li id="rc-sec-Sec1" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Sec1" data-track="click" data-track-action="section anchor" data-track-label="link:Rationale">Rationale</a></li><li id="rc-sec-Sec2" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Sec2" data-track="click" data-track-action="section anchor" data-track-label="link:Bowtie description and results">Bowtie description and results</a></li><li id="rc-sec-Sec13" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Sec13" data-track="click" data-track-action="section anchor" data-track-label="link:Discussion">Discussion</a></li><li id="rc-sec-Sec14" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Sec14" data-track="click" data-track-action="section anchor" data-track-label="link:Additional data files">Additional data files</a></li><li id="rc-sec-abbreviations" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#abbreviations" data-track="click" data-track-action="section anchor" data-track-label="link:Abbreviations">Abbreviations</a></li><li id="rc-sec-Bib1" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Bib1" data-track="click" data-track-action="section anchor" data-track-label="link:References">References</a></li><li id="rc-sec-Ack1" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Ack1" data-track="click" data-track-action="section anchor" data-track-label="link:Acknowledgements">Acknowledgements</a></li><li id="rc-sec-author-information" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#author-information" data-track="click" data-track-action="section anchor" data-track-label="link:Author information">Author information</a></li><li id="rc-sec-additional-information" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#additional-information" data-track="click" data-track-action="section anchor" data-track-label="link:Additional information">Additional information</a></li><li id="rc-sec-Sec15" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Sec15" data-track="click" data-track-action="section anchor" data-track-label="link:Electronic supplementary material">Electronic supplementary material</a></li><li id="rc-sec-Sec16" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Sec16" data-track="click" data-track-action="section anchor" data-track-label="link:Authors original submitted files for images">Authors’ original submitted files for images</a></li><li id="rc-sec-rightslink" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#rightslink" data-track="click" data-track-action="section anchor" data-track-label="link:Rights and permissions">Rights and permissions</a></li><li id="rc-sec-article-info" class="c-reading-companion__section-item"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#article-info" data-track="click" data-track-action="section anchor" data-track-label="link:About this article">About this article</a></li></ul></div>
                                <div class="js-ad u-lazy-ad-wrapper u-mt-16 u-show" data-component-mpu="">
                                    
    
    <aside class="adsbox c-ad c-ad--300x250 u-mt-16" data-component-mpu="">
        <div class="c-ad__inner">
            <p class="c-ad__label">Advertisement</p>
            <div id="div-gpt-ad-MPU1" data-ad-type="MPU1" data-test="MPU1-ad" data-pa11y-ignore="" data-gpt="" data-gpt-unitpath="/270604982/bmc/genomebiology/articles" data-gpt-sizes="300x250" data-gpt-targeting="pos=MPU1;doi=10.1186/gb-2009-10-3-r25;type=article;kwrd=Additional Data File,Central Processing Unit,Human Reference Genome,Memory Footprint,NCBI Short Read Archive;pmc=L32030,B12008,L32020,L32010,L15001,L21001;sponsored=TwentyGB;">
                <noscript>
                    <a href="//pubads.g.doubleclick.net/gampad/jump?iu=/270604982/bmc/genomebiology/articles&amp;sz=300x250&amp;pos=MPU1&amp;doi=10.1186/gb-2009-10-3-r25&amp;type=article&amp;kwrd=Additional Data File,Central Processing Unit,Human Reference Genome,Memory Footprint,NCBI Short Read Archive&amp;pmc=L32030,B12008,L32020,L32010,L15001,L21001&amp;sponsored=TwentyGB&amp;">
                        <img data-test="gpt-advert-fallback-img"
                            src="//pubads.g.doubleclick.net/gampad/ad?iu=/270604982/bmc/genomebiology/articles&amp;sz=300x250&amp;pos=MPU1&amp;doi=10.1186/gb-2009-10-3-r25&amp;type=article&amp;kwrd=Additional Data File,Central Processing Unit,Human Reference Genome,Memory Footprint,NCBI Short Read Archive&amp;pmc=L32030,B12008,L32020,L32010,L15001,L21001&amp;sponsored=TwentyGB&amp;"
                            alt="Advertisement"
                            width="300"
                            height="250">
                    </a>
                </noscript>
            </div>
        </div>
    </aside>

                                </div>
                            </div>
                            <div class="c-reading-companion__panel c-reading-companion__figures c-reading-companion__panel--full-width" id="tabpanel-figures" aria-labelledby="tab-figures" role="tabpanel"><div class="c-reading-companion__scroll-pane"><ul class="c-reading-companion__figures-list"><li class="c-reading-companion__figure-item"><figure><figcaption><b class="c-reading-companion__figure-title u-h4" id="rc-Fig1">Figure 1</b></figcaption><picture><source type="image/webp" data-srcset="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig1_HTML.jpg?as=webp"><img data-src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig1_HTML.jpg" alt="figure 1" aria-describedby="rc-Fig1"></picture><p class="c-reading-companion__figure-links"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Fig1" data-track="click" data-track-action="figure anchor" data-track-label="link">View in article</a><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/1" class="c-reading-companion__figure-full-link" data-track="click" data-track-action="view figure" data-track-label="link" rel="nofollow">Full size image<svg width="16" height="16" class="u-icon"><use href="#global-icon-chevron-right"></use></svg></a></p></figure></li><li class="c-reading-companion__figure-item"><figure><figcaption><b class="c-reading-companion__figure-title u-h4" id="rc-Fig2">Figure 2</b></figcaption><picture><source type="image/webp" data-srcset="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig2_HTML.jpg?as=webp"><img data-src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig2_HTML.jpg" alt="figure 2" aria-describedby="rc-Fig2"></picture><p class="c-reading-companion__figure-links"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Fig2" data-track="click" data-track-action="figure anchor" data-track-label="link">View in article</a><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/2" class="c-reading-companion__figure-full-link" data-track="click" data-track-action="view figure" data-track-label="link" rel="nofollow">Full size image<svg width="16" height="16" class="u-icon"><use href="#global-icon-chevron-right"></use></svg></a></p></figure></li><li class="c-reading-companion__figure-item"><figure><figcaption><b class="c-reading-companion__figure-title u-h4" id="rc-Fig3">Figure 3</b></figcaption><picture><source type="image/webp" data-srcset="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig3_HTML.jpg?as=webp"><img data-src="https://media.springernature.com/lw685/springer-static/image/art%3A10.1186%2Fgb-2009-10-3-r25/MediaObjects/13059_2008_Article_2175_Fig3_HTML.jpg" alt="figure 3" aria-describedby="rc-Fig3"></picture><p class="c-reading-companion__figure-links"><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader#Fig3" data-track="click" data-track-action="figure anchor" data-track-label="link">View in article</a><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25/figures/3" class="c-reading-companion__figure-full-link" data-track="click" data-track-action="view figure" data-track-label="link" rel="nofollow">Full size image<svg width="16" height="16" class="u-icon"><use href="#global-icon-chevron-right"></use></svg></a></p></figure></li></ul></div></div>
                            <div class="c-reading-companion__panel c-reading-companion__references c-reading-companion__panel--full-width" id="tabpanel-references" aria-labelledby="tab-references" role="tabpanel"><div class="c-reading-companion__scroll-pane"><ol class="c-reading-companion__references-list c-reading-companion__references-list--numeric"><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR1">Down TA, Rakyan VK, Turner DJ, Flicek P, Li H, Kulesha E, Graf S, Johnson N, Herrero J, Tomazou EM, Thorne NP, Backdahl L, Herberth M, Howe KL, Jackson DK, Miretti MM, Marioni JC, Birney E, Hubbard TJ, Durbin R, Tavare S, Beck S: A Bayesian deconvolution strategy for immunoprecipitation-based DNA methylome analysis. Nat Biotechnol. 2008, 26: 779-785. 10.1038/nbt1414.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18612301" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXot1entL4%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2644410" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1038%2Fnbt1414" data-track="click" data-track-action="article reference" data-track-label="10.1038/nbt1414">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=A%20Bayesian%20deconvolution%20strategy%20for%20immunoprecipitation-based%20DNA%20methylome%20analysis.&amp;journal=Nat%20Biotechnol&amp;doi=10.1038%2Fnbt1414&amp;volume=26&amp;pages=779-785&amp;publication_year=2008&amp;author=Down%2CTA&amp;author=Rakyan%2CVK&amp;author=Turner%2CDJ&amp;author=Flicek%2CP&amp;author=Li%2CH&amp;author=Kulesha%2CE&amp;author=Graf%2CS&amp;author=Johnson%2CN&amp;author=Herrero%2CJ&amp;author=Tomazou%2CEM&amp;author=Thorne%2CNP&amp;author=Backdahl%2CL&amp;author=Herberth%2CM&amp;author=Howe%2CKL&amp;author=Jackson%2CDK&amp;author=Miretti%2CMM&amp;author=Marioni%2CJC&amp;author=Birney%2CE&amp;author=Hubbard%2CTJ&amp;author=Durbin%2CR&amp;author=Tavare%2CS&amp;author=Beck%2CS" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR2">Johnson DS, Mortazavi A, Myers RM, Wold B: Genome-wide mapping of in vivo protein-DNA interactions. Science. 2007, 316: 1497-1502. 10.1126/science.1141319.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17540862" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD2sXmtFSjtrg%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="https://doi.org/10.1126%2Fscience.1141319" data-track="click" data-track-action="article reference" data-track-label="10.1126/science.1141319">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Genome-wide%20mapping%20of%20in%20vivo%20protein-DNA%20interactions.&amp;journal=Science&amp;doi=10.1126%2Fscience.1141319&amp;volume=316&amp;pages=1497-1502&amp;publication_year=2007&amp;author=Johnson%2CDS&amp;author=Mortazavi%2CA&amp;author=Myers%2CRM&amp;author=Wold%2CB" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR3">Marioni JC, Mason CE, Mane SM, Stephens M, Gilad Y: RNA-seq: an assessment of technical reproducibility and comparison with gene expression arrays. Genome Res. 2008, 18: 1509-1517. 10.1101/gr.079558.108.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18550803" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtV2qsr3E" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2527709" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1101%2Fgr.079558.108" data-track="click" data-track-action="article reference" data-track-label="10.1101/gr.079558.108">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=RNA-seq%3A%20an%20assessment%20of%20technical%20reproducibility%20and%20comparison%20with%20gene%20expression%20arrays.&amp;journal=Genome%20Res&amp;doi=10.1101%2Fgr.079558.108&amp;volume=18&amp;pages=1509-1517&amp;publication_year=2008&amp;author=Marioni%2CJC&amp;author=Mason%2CCE&amp;author=Mane%2CSM&amp;author=Stephens%2CM&amp;author=Gilad%2CY" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR4">Bentley DR, Balasubramanian S, Swerdlow HP, Smith GP, Milton J, Brown CG, Hall KP, Evers DJ, Barnes CL, Bignell HR, Boutell JM, Bryant J, Carter RJ, Keira Cheetham R, Cox AJ, Ellis DJ, Flatbush MR, Gormley NA, Humphray SJ, Irving LJ, Karbelashvili MS, Kirk SM, Li H, Liu X, Maisinger KS, Murray LJ, Obradovic B, Ost T, Parkinson ML, Pratt MR, et al: Accurate whole human genome sequencing using reversible terminator chemistry. Nature. 2008, 456: 53-59. 10.1038/nature07517.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18987734" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtlCjtr3E" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2581791" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1038%2Fnature07517" data-track="click" data-track-action="article reference" data-track-label="10.1038/nature07517">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Accurate%20whole%20human%20genome%20sequencing%20using%20reversible%20terminator%20chemistry.&amp;journal=Nature&amp;doi=10.1038%2Fnature07517&amp;volume=456&amp;pages=53-59&amp;publication_year=2008&amp;author=Bentley%2CDR&amp;author=Balasubramanian%2CS&amp;author=Swerdlow%2CHP&amp;author=Smith%2CGP&amp;author=Milton%2CJ&amp;author=Brown%2CCG&amp;author=Hall%2CKP&amp;author=Evers%2CDJ&amp;author=Barnes%2CCL&amp;author=Bignell%2CHR&amp;author=Boutell%2CJM&amp;author=Bryant%2CJ&amp;author=Carter%2CRJ&amp;author=Keira%20Cheetham%2CR&amp;author=Cox%2CAJ&amp;author=Ellis%2CDJ&amp;author=Flatbush%2CMR&amp;author=Gormley%2CNA&amp;author=Humphray%2CSJ&amp;author=Irving%2CLJ&amp;author=Karbelashvili%2CMS&amp;author=Kirk%2CSM&amp;author=Li%2CH&amp;author=Liu%2CX&amp;author=Maisinger%2CKS&amp;author=Murray%2CLJ&amp;author=Obradovic%2CB&amp;author=Ost%2CT&amp;author=Parkinson%2CML&amp;author=Pratt%2CMR" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR5">Ley TJ, Mardis ER, Ding L, Fulton B, McLellan MD, Chen K, Dooling D, Dunford-Shore BH, McGrath S, Hickenbotham M, Cook L, Abbott R, Larson DE, Koboldt DC, Pohl C, Smith S, Hawkins A, Abbott S, Locke D, Hillier LW, Miner T, Fulton L, Magrini V, Wylie T, Glasscock J, Conyers J, Sander N, Shi X, Osborne JR, Minx P, et al: DNA sequencing of a cytogenetically normal acute myeloid leukaemia genome. Nature. 2008, 456: 66-72. 10.1038/nature07485.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18987736" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtlCjtrfE" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2603574" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1038%2Fnature07485" data-track="click" data-track-action="article reference" data-track-label="10.1038/nature07485">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=DNA%20sequencing%20of%20a%20cytogenetically%20normal%20acute%20myeloid%20leukaemia%20genome.&amp;journal=Nature&amp;doi=10.1038%2Fnature07485&amp;volume=456&amp;pages=66-72&amp;publication_year=2008&amp;author=Ley%2CTJ&amp;author=Mardis%2CER&amp;author=Ding%2CL&amp;author=Fulton%2CB&amp;author=McLellan%2CMD&amp;author=Chen%2CK&amp;author=Dooling%2CD&amp;author=Dunford-Shore%2CBH&amp;author=McGrath%2CS&amp;author=Hickenbotham%2CM&amp;author=Cook%2CL&amp;author=Abbott%2CR&amp;author=Larson%2CDE&amp;author=Koboldt%2CDC&amp;author=Pohl%2CC&amp;author=Smith%2CS&amp;author=Hawkins%2CA&amp;author=Abbott%2CS&amp;author=Locke%2CD&amp;author=Hillier%2CLW&amp;author=Miner%2CT&amp;author=Fulton%2CL&amp;author=Magrini%2CV&amp;author=Wylie%2CT&amp;author=Glasscock%2CJ&amp;author=Conyers%2CJ&amp;author=Sander%2CN&amp;author=Shi%2CX&amp;author=Osborne%2CJR&amp;author=Minx%2CP" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR6">Wang J, Wang W, Li R, Li Y, Tian G, Goodman L, Fan W, Zhang J, Li J, Zhang J, Guo Y, Feng B, Li H, Lu Y, Fang X, Liang H, Du Z, Li D, Zhao Y, Hu Y, Yang Z, Zheng H, Hellmann I, Inouye M, Pool J, Yi X, Zhao J, Duan J, Zhou Y, Qin J, et al: The diploid genome sequence of an Asian individual. Nature. 2008, 456: 60-65. 10.1038/nature07484.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18987735" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtlCjtrnK" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2716080" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1038%2Fnature07484" data-track="click" data-track-action="article reference" data-track-label="10.1038/nature07484">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=The%20diploid%20genome%20sequence%20of%20an%20Asian%20individual.&amp;journal=Nature&amp;doi=10.1038%2Fnature07484&amp;volume=456&amp;pages=60-65&amp;publication_year=2008&amp;author=Wang%2CJ&amp;author=Wang%2CW&amp;author=Li%2CR&amp;author=Li%2CY&amp;author=Tian%2CG&amp;author=Goodman%2CL&amp;author=Fan%2CW&amp;author=Zhang%2CJ&amp;author=Li%2CJ&amp;author=Zhang%2CJ&amp;author=Guo%2CY&amp;author=Feng%2CB&amp;author=Li%2CH&amp;author=Lu%2CY&amp;author=Fang%2CX&amp;author=Liang%2CH&amp;author=Du%2CZ&amp;author=Li%2CD&amp;author=Zhao%2CY&amp;author=Hu%2CY&amp;author=Yang%2CZ&amp;author=Zheng%2CH&amp;author=Hellmann%2CI&amp;author=Inouye%2CM&amp;author=Pool%2CJ&amp;author=Yi%2CX&amp;author=Zhao%2CJ&amp;author=Duan%2CJ&amp;author=Zhou%2CY&amp;author=Qin%2CJ" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR7">Li H, Ruan J, Durbin R: Mapping short DNA sequencing reads and calling variants using mapping quality scores. Genome Res. 2008, 18: 1851-1858. 10.1101/gr.078212.108.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18714091" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtlKhsrrN" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2577856" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1101%2Fgr.078212.108" data-track="click" data-track-action="article reference" data-track-label="10.1101/gr.078212.108">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Mapping%20short%20DNA%20sequencing%20reads%20and%20calling%20variants%20using%20mapping%20quality%20scores.&amp;journal=Genome%20Res&amp;doi=10.1101%2Fgr.078212.108&amp;volume=18&amp;pages=1851-1858&amp;publication_year=2008&amp;author=Li%2CH&amp;author=Ruan%2CJ&amp;author=Durbin%2CR" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR8">Li R, Li Y, Kristiansen K, Wang J: SOAP: short oligonucleotide alignment program. Bioinformatics. 2008, 24: 713-714. 10.1093/bioinformatics/btn025.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18227114" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXislKlt7Y%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn025" data-track="click" data-track-action="article reference" data-track-label="10.1093/bioinformatics/btn025">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=SOAP%3A%20short%20oligonucleotide%20alignment%20program.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2Fbtn025&amp;volume=24&amp;pages=713-714&amp;publication_year=2008&amp;author=Li%2CR&amp;author=Li%2CY&amp;author=Kristiansen%2CK&amp;author=Wang%2CJ" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR9">Kaiser J: DNA sequencing. A plan to capture human diversity in 1000 genomes. Science. 2008, 319: 395-10.1126/science.319.5862.395.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18218868" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXhtl2qsbs%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="https://doi.org/10.1126%2Fscience.319.5862.395" data-track="click" data-track-action="article reference" data-track-label="10.1126/science.319.5862.395">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=DNA%20sequencing.%20A%20plan%20to%20capture%20human%20diversity%20in%201000%20genomes.&amp;journal=Science&amp;doi=10.1126%2Fscience.319.5862.395&amp;volume=319&amp;publication_year=2008&amp;author=Kaiser%2CJ" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR10">Smith AD, Xuan Z, Zhang MQ: Using quality scores and longer reads improves accuracy of Solexa read mapping. BMC Bioinformatics. 2008, 9: 128-10.1186/1471-2105-9-128.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18307793" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2335322" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1186%2F1471-2105-9-128" data-track="click" data-track-action="article reference" data-track-label="10.1186/1471-2105-9-128">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Using%20quality%20scores%20and%20longer%20reads%20improves%20accuracy%20of%20Solexa%20read%20mapping.&amp;journal=BMC%20Bioinformatics&amp;doi=10.1186%2F1471-2105-9-128&amp;volume=9&amp;publication_year=2008&amp;author=Smith%2CAD&amp;author=Xuan%2CZ&amp;author=Zhang%2CMQ" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR11">Lin H, Zhang Z, Zhang MQ, Ma B, Li M: ZOOM! Zillions Of Oligos Mapped. Bioinformatics. 2008, 24: 2431-2437. 10.1093/bioinformatics/btn416.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18684737" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXht12gtbrP" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2732274" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn416" data-track="click" data-track-action="article reference" data-track-label="10.1093/bioinformatics/btn416">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=ZOOM%21%20Zillions%20Of%20Oligos%20Mapped.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2Fbtn416&amp;volume=24&amp;pages=2431-2437&amp;publication_year=2008&amp;author=Lin%2CH&amp;author=Zhang%2CZ&amp;author=Zhang%2CMQ&amp;author=Ma%2CB&amp;author=Li%2CM" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR12">SHRiMP - SHort Read Mapping Package. [<a href="http://compbio.cs.toronto.edu/shrimp/">http://compbio.cs.toronto.edu/shrimp/</a>]</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR13">Baeza-Yates RA, Perleberg CH: Fast and practical approximate string matching. Inf Process Lett. 1996, 59: 21-27. 10.1016/0020-0190(96)00083-X.</p><p class="c-reading-companion__reference-links"><a href="https://doi.org/10.1016%2F0020-0190%2896%2900083-X" data-track="click" data-track-action="article reference" data-track-label="10.1016/0020-0190(96)00083-X">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Fast%20and%20practical%20approximate%20string%20matching.&amp;journal=Inf%20Process%20Lett&amp;doi=10.1016%2F0020-0190%2896%2900083-X&amp;volume=59&amp;pages=21-27&amp;publication_year=1996&amp;author=Baeza-Yates%2CRA&amp;author=Perleberg%2CCH" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR14">Burkhardt S, Kärkkäinen J: Better Filtering with Gapped q-Grams. Fundam Inf. 2003, 56: 51-70.</p><p class="c-reading-companion__reference-links"><a href="http://scholar.google.com/scholar_lookup?&amp;title=Better%20Filtering%20with%20Gapped%20q-Grams.&amp;journal=Fundam%20Inf&amp;volume=56&amp;pages=51-70&amp;publication_year=2003&amp;author=Burkhardt%2CS&amp;author=K%C3%A4rkk%C3%A4inen%2CJ" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR15">Ma B, Tromp J, Li M: PatternHunter: faster and more sensitive homology search. Bioinformatics. 2002, 18: 440-445. 10.1093/bioinformatics/18.3.440.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=11934743" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD38XivFKrsrY%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="https://doi.org/10.1093%2Fbioinformatics%2F18.3.440" data-track="click" data-track-action="article reference" data-track-label="10.1093/bioinformatics/18.3.440">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=PatternHunter%3A%20faster%20and%20more%20sensitive%20homology%20search.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2F18.3.440&amp;volume=18&amp;pages=440-445&amp;publication_year=2002&amp;author=Ma%2CB&amp;author=Tromp%2CJ&amp;author=Li%2CM" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR16">Smith TF, Waterman MS: Identification of common molecular subsequences. J Mol Biol. 1981, 147: 195-197. 10.1016/0022-2836(81)90087-5.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=7265238" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BC3cXitlCnsLw%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="https://doi.org/10.1016%2F0022-2836%2881%2990087-5" data-track="click" data-track-action="article reference" data-track-label="10.1016/0022-2836(81)90087-5">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Identification%20of%20common%20molecular%20subsequences.&amp;journal=J%20Mol%20Biol&amp;doi=10.1016%2F0022-2836%2881%2990087-5&amp;volume=147&amp;pages=195-197&amp;publication_year=1981&amp;author=Smith%2CTF&amp;author=Waterman%2CMS" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR17">Burrows M, Wheeler DJ: A Block Sorting Lossless Data Compression Algorithm. Technical Report 124. 1994, Palo Alto, CA: Digital Equipment Corporation</p><p class="c-reading-companion__reference-links"><a href="http://scholar.google.com/scholar_lookup?&amp;title=A%20Block%20Sorting%20Lossless%20Data%20Compression%20Algorithm.%20Technical%20Report%20124&amp;publication_year=1994&amp;author=Burrows%2CM&amp;author=Wheeler%2CDJ" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR18">Ferragina P, Manzini G: Opportunistic data structures with applications. [<a href="http://web.unipmn.it/~manzini/papers/focs00draft.pdf">http://web.unipmn.it/~manzini/papers/focs00draft.pdf</a>]</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR19">Ferragina P, Manzini G: An experimental study of an opportunistic index. Proceedings of the Twelfth Annual ACM-SIAM Symposium on Discrete algorithms. 2001, Washington, DC: Society for Industrial and Applied Mathematics, 269-278.</p><p class="c-reading-companion__reference-links"><a href="http://scholar.google.com/scholar_lookup?&amp;title=An%20experimental%20study%20of%20an%20opportunistic%20index.&amp;pages=269-278&amp;publication_year=2001&amp;author=Ferragina%2CP&amp;author=Manzini%2CG" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR20">Healy J, Thomas EE, Schwartz JT, Wigler M: Annotating large genomes with exact word matches. Genome Res. 2003, 13: 2306-2315. 10.1101/gr.1350803.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12975312" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD3sXotF2hsbw%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC403711" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1101%2Fgr.1350803" data-track="click" data-track-action="article reference" data-track-label="10.1101/gr.1350803">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Annotating%20large%20genomes%20with%20exact%20word%20matches.&amp;journal=Genome%20Res&amp;doi=10.1101%2Fgr.1350803&amp;volume=13&amp;pages=2306-2315&amp;publication_year=2003&amp;author=Healy%2CJ&amp;author=Thomas%2CEE&amp;author=Schwartz%2CJT&amp;author=Wigler%2CM" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR21">Lippert RA: Space-efficient whole genome comparisons with Burrows-Wheeler transforms. J Comput Biol. 2005, 12: 407-415. 10.1089/cmb.2005.12.407.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=15882139" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD2MXjvFeqsL0%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="https://doi.org/10.1089%2Fcmb.2005.12.407" data-track="click" data-track-action="article reference" data-track-label="10.1089/cmb.2005.12.407">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Space-efficient%20whole%20genome%20comparisons%20with%20Burrows-Wheeler%20transforms.&amp;journal=J%20Comput%20Biol&amp;doi=10.1089%2Fcmb.2005.12.407&amp;volume=12&amp;pages=407-415&amp;publication_year=2005&amp;author=Lippert%2CRA" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR22">Graf S, Nielsen FG, Kurtz S, Huynen MA, Birney E, Stunnenberg H, Flicek P: Optimized design and assessment of whole genome tiling arrays. Bioinformatics. 2007, 23: i195-i204. 10.1093/bioinformatics/btm200.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17646297" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtm200" data-track="click" data-track-action="article reference" data-track-label="10.1093/bioinformatics/btm200">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Optimized%20design%20and%20assessment%20of%20whole%20genome%20tiling%20arrays.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2Fbtm200&amp;volume=23&amp;pages=i195-i204&amp;publication_year=2007&amp;author=Graf%2CS&amp;author=Nielsen%2CFG&amp;author=Kurtz%2CS&amp;author=Huynen%2CMA&amp;author=Birney%2CE&amp;author=Stunnenberg%2CH&amp;author=Flicek%2CP" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR23">Lam TW, Sung WK, Tam SL, Wong CK, Yiu SM: Compressed indexing and local alignment of DNA. Bioinformatics. 2008, 24: 791-797. 10.1093/bioinformatics/btn032.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18227115" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXjsVaktL8%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtn032" data-track="click" data-track-action="article reference" data-track-label="10.1093/bioinformatics/btn032">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Compressed%20indexing%20and%20local%20alignment%20of%20DNA.&amp;journal=Bioinformatics&amp;doi=10.1093%2Fbioinformatics%2Fbtn032&amp;volume=24&amp;pages=791-797&amp;publication_year=2008&amp;author=Lam%2CTW&amp;author=Sung%2CWK&amp;author=Tam%2CSL&amp;author=Wong%2CCK&amp;author=Yiu%2CSM" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR24">Ewing B, Green P: Base-calling of automated sequencer traces using phred. II. Error probabilities. Genome Res. 1998, 8: 186-194.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=9521922" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DyaK1cXitlWlu7g%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="https://doi.org/10.1101%2Fgr.8.3.186" data-track="click" data-track-action="article reference" data-track-label="10.1101/gr.8.3.186">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Base-calling%20of%20automated%20sequencer%20traces%20using%20phred.%20II.%20Error%20probabilities.&amp;journal=Genome%20Res&amp;doi=10.1101%2Fgr.8.3.186&amp;volume=8&amp;pages=186-194&amp;publication_year=1998&amp;author=Ewing%2CB&amp;author=Green%2CP" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR25">Bowtie: An ultrafast memory-efficient short read aligner. [<a href="http://bowtie.cbcb.umd.edu/">http://bowtie.cbcb.umd.edu/</a>]</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR26">Campbell PJ, Stephens PJ, Pleasance ED, O'Meara S, Li H, Santarius T, Stebbings LA, Leroy C, Edkins S, Hardy C, Teague JW, Menzies A, Goodhead I, Turner DJ, Clee CM, Quail MA, Cox A, Brown C, Durbin R, Hurles ME, Edwards PA, Bignell GR, Stratton MR, Futreal PA: Identification of somatically acquired rearrangements in cancer using genome-wide massively parallel paired-end sequencing. Nat Genet. 2008, 40: 722-729. 10.1038/ng.128.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18438408" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXmsVejsL4%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2705838" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1038%2Fng.128" data-track="click" data-track-action="article reference" data-track-label="10.1038/ng.128">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Identification%20of%20somatically%20acquired%20rearrangements%20in%20cancer%20using%20genome-wide%20massively%20parallel%20paired-end%20sequencing.&amp;journal=Nat%20Genet&amp;doi=10.1038%2Fng.128&amp;volume=40&amp;pages=722-729&amp;publication_year=2008&amp;author=Campbell%2CPJ&amp;author=Stephens%2CPJ&amp;author=Pleasance%2CED&amp;author=O%27Meara%2CS&amp;author=Li%2CH&amp;author=Santarius%2CT&amp;author=Stebbings%2CLA&amp;author=Leroy%2CC&amp;author=Edkins%2CS&amp;author=Hardy%2CC&amp;author=Teague%2CJW&amp;author=Menzies%2CA&amp;author=Goodhead%2CI&amp;author=Turner%2CDJ&amp;author=Clee%2CCM&amp;author=Quail%2CMA&amp;author=Cox%2CA&amp;author=Brown%2CC&amp;author=Durbin%2CR&amp;author=Hurles%2CME&amp;author=Edwards%2CPA&amp;author=Bignell%2CGR&amp;author=Stratton%2CMR&amp;author=Futreal%2CPA" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR27">Holt KE, Parkhill J, Mazzoni CJ, Roumagnac P, Weill FX, Goodhead I, Rance R, Baker S, Maskell DJ, Wain J, Dolecek C, Achtman M, Dougan G: High-throughput sequencing provides insights into genome variation and evolution in <i>Salmonella typhi</i>. Nat Genet. 2008, 40: 987-993. 10.1038/ng.195.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18660809" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXptVKkurk%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2652037" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1038%2Fng.195" data-track="click" data-track-action="article reference" data-track-label="10.1038/ng.195">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=High-throughput%20sequencing%20provides%20insights%20into%20genome%20variation%20and%20evolution%20in%20Salmonella%20typhi.&amp;journal=Nat%20Genet&amp;doi=10.1038%2Fng.195&amp;volume=40&amp;pages=987-993&amp;publication_year=2008&amp;author=Holt%2CKE&amp;author=Parkhill%2CJ&amp;author=Mazzoni%2CCJ&amp;author=Roumagnac%2CP&amp;author=Weill%2CFX&amp;author=Goodhead%2CI&amp;author=Rance%2CR&amp;author=Baker%2CS&amp;author=Maskell%2CDJ&amp;author=Wain%2CJ&amp;author=Dolecek%2CC&amp;author=Achtman%2CM&amp;author=Dougan%2CG" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR28">Nagalakshmi U, Wang Z, Waern K, Shou C, Raha D, Gerstein M, Snyder M: The transcriptional landscape of the yeast genome defined by RNA sequencing. Science. 2008, 320: 1344-1349. 10.1126/science.1158441.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18451266" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="https://genomebiology.biomedcentral.com/articles/cas-redirect/1:CAS:528:DC%2BD1cXmslWgtbY%3D" data-track="click" data-track-action="cas reference" data-track-label="link">CAS</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2951732" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1126%2Fscience.1158441" data-track="click" data-track-action="article reference" data-track-label="10.1126/science.1158441">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=The%20transcriptional%20landscape%20of%20the%20yeast%20genome%20defined%20by%20RNA%20sequencing.&amp;journal=Science&amp;doi=10.1126%2Fscience.1158441&amp;volume=320&amp;pages=1344-1349&amp;publication_year=2008&amp;author=Nagalakshmi%2CU&amp;author=Wang%2CZ&amp;author=Waern%2CK&amp;author=Shou%2CC&amp;author=Raha%2CD&amp;author=Gerstein%2CM&amp;author=Snyder%2CM" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR29">Kärkkäinen J: Fast BWT in small space by blockwise suffix sorting. Theor Comput Sci. 2007, 387: 249-257.</p><p class="c-reading-companion__reference-links"><a href="https://doi.org/10.1016%2Fj.tcs.2007.07.018" data-track="click" data-track-action="article reference" data-track-label="10.1016/j.tcs.2007.07.018">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=Fast%20BWT%20in%20small%20space%20by%20blockwise%20suffix%20sorting.&amp;journal=Theor%20Comput%20Sci&amp;doi=10.1016%2Fj.tcs.2007.07.018&amp;volume=387&amp;pages=249-257&amp;publication_year=2007&amp;author=K%C3%A4rkk%C3%A4inen%2CJ" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li><li class="c-reading-companion__reference-item"><p class="c-reading-companion__reference-citation u-font-family-serif" id="rc-ref-CR30">Doring A, Weese D, Rausch T, Reinert K: SeqAn an efficient, generic C++ library for sequence analysis. BMC Bioinformatics. 2008, 9: 11-10.1186/1471-2105-9-11.</p><p class="c-reading-companion__reference-links"><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18184432" data-track="click" data-track-action="pubmed reference" data-track-label="link">PubMed</a>&nbsp;<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2246154" data-track="click" data-track-action="pubmed central reference" data-track-label="link">PubMed Central</a>&nbsp;<a href="https://doi.org/10.1186%2F1471-2105-9-11" data-track="click" data-track-action="article reference" data-track-label="10.1186/1471-2105-9-11">Article</a>&nbsp;<a href="http://scholar.google.com/scholar_lookup?&amp;title=SeqAn%20an%20efficient%2C%20generic%20C%2B%2B%20library%20for%20sequence%20analysis.&amp;journal=BMC%20Bioinformatics&amp;doi=10.1186%2F1471-2105-9-11&amp;volume=9&amp;publication_year=2008&amp;author=Doring%2CA&amp;author=Weese%2CD&amp;author=Rausch%2CT&amp;author=Reinert%2CK" data-track="click" data-track-action="google scholar reference" data-track-label="link">
                    Google Scholar</a>&nbsp;</p></li></ol></div></div>
                        </div>
                    </div>
                </aside>
            </div>
        </div>

        
            <div class="c-journal-footer">
                <div class="c-journal-footer__inner">
                    <div class="c-journal-footer__summary">
                        <h4 class="c-journal-title c-journal-title--footer">
                            
                            <span class="c-journal-title__text">Genome Biology</span>
                        </h4>
                         <p class="c-journal-footer__issn">ISSN: 1474-760X</p>
                    </div>
                    
                        <div class="c-journal-footer__contact">
                            <h4 class="c-journal-footer__contact-title ">Contact us</h4>
                            <ul class="c-journal-footer__contact-list">
                                
                                    <li class="c-journal-footer__contact-item">Submission enquiries: <a href="mailto:editorial@genomebiology.com">editorial@genomebiology.com</a></li>
                                
                                
                                    <li class="c-journal-footer__contact-item">General enquiries: <a href="mailto:info@biomedcentral.com">info@biomedcentral.com</a></li>
                                
                            </ul>
                        </div>
                    
                </div>
            </div>
        
        
    <img rel="nofollow" class="tracker" style="display:none" src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/gb-2009-10-3-r25" alt="">

         
    <footer>
        
            <div class="c-publisher-footer" data-test="publisher-footer">
    <div class="u-container">
        <div class="u-display-flex u-flex-wrap u-justify-content-space-between" data-test="publisher-footer-menu">
            <div class="u-display-flex">
                
                    
                        <ul class="c-list-group c-list-group--sm u-mr-24 u-mb-16">
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="http://blogs.biomedcentral.com/">Read more on our blogs</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/login">Receive BMC newsletters</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/account">Manage article alerts</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://authorservices.springernature.com/go/sn/?utm_source=Website&amp;utm_medium=BMC&amp;utm_campaign=SNAS+Referrals+2022&amp;utm_id=ref2022">Language editing for authors</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://authorservices.springernature.com/go/sn/?utm_source=Website&amp;utm_medium=BMC&amp;utm_campaign=SNAS+Referrals+2022&amp;utm_id=ref2022">Scientific editing for authors</a>
                                </li>
                            
                        </ul>
                    
                        <ul class="c-list-group c-list-group--sm u-mr-24 u-mb-16">
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/about/policies">Policies</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/accessibility">Accessibility</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/about/press-centre">Press center</a>
                                </li>
                            
                        </ul>
                    
                        <ul class="c-list-group c-list-group--sm u-mr-24 u-mb-16">
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://support.biomedcentral.com/support/home">Support and Contact</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://biomedcentral.typeform.com/to/VLXboo">Leave feedback</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/about/jobs">Careers</a>
                                </li>
                            
                        </ul>
                    
                
            </div>
            <div class="u-mb-24">
                <h3 id="social-menu" class="u-text-sm u-reset-margin u-text-normal">Follow BMC</h3>
                <ul class="u-display-flex u-list-reset" data-test="footer-social-links">
                    
                        <li class="u-mt-8 u-mr-8">
                            <a href="https://twitter.com/biomedcentral" data-track="click" data-track-category="Social" data-track-action="Clicked BMC Twitter" class="u-gray-link">
                                <span class="u-visually-hidden">BMC Twitter page</span>
                                <svg class="u-icon u-text-lg" width="24" height="24" aria-hidden="true" focusable="false">
                                    <use xlink:href="#icon-twitter-bordered"></use>
                                </svg>
                            </a>
                        </li>
                    
                        <li class="u-mt-8 u-mr-8">
                            <a href="https://www.facebook.com/BioMedCentral" data-track="click" data-track-category="Social" data-track-action="Clicked BMC Facebook" class="u-gray-link">
                                <span class="u-visually-hidden">BMC Facebook page</span>
                                <svg class="u-icon u-text-lg" width="24" height="24" aria-hidden="true" focusable="false">
                                    <use xlink:href="#icon-facebook-bordered"></use>
                                </svg>
                            </a>
                        </li>
                    
                        <li class="u-mt-8 u-mr-8">
                            <a href="http://www.weibo.com/biomedcentral" data-track="click" data-track-category="Social" data-track-action="Clicked BMC Weibo" class="u-gray-link">
                                <span class="u-visually-hidden">BMC Weibo page</span>
                                <svg class="u-icon u-text-lg" width="24" height="24" aria-hidden="true" focusable="false">
                                    <use xlink:href="#icon-weibo-bordered"></use>
                                </svg>
                            </a>
                        </li>
                    
                </ul>
            </div>
        </div>
        <p class="u-reset-margin">
            By using this website, you agree to our
            <a class="u-gray-link" href="https://www.biomedcentral.com/terms-and-conditions">Terms and Conditions</a>,
            <a class="u-gray-link" href="https://www.springernature.com/ccpa">California Privacy Statement</a>,
            <a class="u-gray-link" href="https://www.biomedcentral.com/privacy-statement">Privacy
                statement</a> and
            <a class="u-gray-link" href="https://www.biomedcentral.com/cookies" data-test="cookie-link">Cookies</a> policy.
            
                <a class="optanon-toggle-display u-gray-link" data-cc-action="preferences" href="javascript:void(0);">Manage cookies/Do not sell my data</a> we use in the preference centre.
            
        </p>
    </div>
</div>

        
        <div class="c-corporate-footer">
    <div class="u-container">
        <img src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/logo-springernature-acb40b85fb.svg" class="c-corporate-footer__logo" alt="Springer Nature" itemprop="logo" role="img">
        <p class="c-corporate-footer__legal" data-test="copyright"> © 2022 BioMed Central Ltd unless otherwise stated. Part of
            <a class="c-corporate-footer__link" href="https://www.springernature.com/" itemscope="" itemtype="http://schema.org/Organization" itemid="#parentOrganization">Springer Nature</a>.
        </p>
    </div>
</div>

    </footer>

    </div>



        
    <svg class="u-hide hide">
        <symbol id="global-icon-chevron-right" viewBox="0 0 10 10" xmlns="http://www.w3.org/2000/svg">
            <path d="m5.96738168 4.70639573 2.39518594-2.41447274c.37913917-.38219212.98637524-.38972225 1.35419292-.01894278.37750606.38054586.37784436.99719163-.00013556 1.37821513l-4.03074001 4.06319683c-.37758093.38062133-.98937525.38100976-1.367372-.00003075l-4.03091981-4.06337806c-.37759778-.38063832-.38381821-.99150444-.01600053-1.3622839.37750607-.38054587.98772445-.38240057 1.37006824.00302197l2.39538588 2.4146743.96295325.98624457z" fill="currentColor" fill-rule="evenodd" transform="matrix(0 -1 1 0 0 10)"></path>
        </symbol>
        <symbol id="global-icon-download" viewBox="0 0 16 16">
            <path d="M2 14c0-.556.449-1 1.002-1h9.996a.999.999 0 110 2H3.002A1.006 1.006 0 012 14zM9 2v6.8l2.482-2.482c.392-.392 1.022-.4 1.403-.02a1.001 1.001 0 010 1.417l-4.177 4.177a1.001 1.001 0 01-1.416 0L3.115 7.715a.991.991 0 01-.016-1.4 1 1 0 011.42.003L7 8.8V2c0-.55.444-.996 1-.996.552 0 1 .445 1 .996z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="global-icon-email" viewBox="0 0 18 18">
            <path d="M1.995 2h14.01A2 2 0 0118 4.006v9.988A2 2 0 0116.005 16H1.995A2 2 0 010 13.994V4.006A2 2 0 011.995 2zM1 13.994A1 1 0 001.995 15h14.01A1 1 0 0017 13.994V4.006A1 1 0 0016.005 3H1.995A1 1 0 001 4.006zM9 11L2 7V5.557l7 4 7-4V7z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="global-icon-institution" viewBox="0 0 18 18">
            <path d="M14 8a1 1 0 011 1v6h1.5a.5.5 0 01.5.5v.5h.5a.5.5 0 01.5.5V18H0v-1.5a.5.5 0 01.5-.5H1v-.5a.5.5 0 01.5-.5H3V9a1 1 0 112 0v6h8V9a1 1 0 011-1zM6 8l2 1v4l-2 1zm6 0v6l-2-1V9zM9.573.401l7.036 4.925A.92.92 0 0116.081 7H1.92a.92.92 0 01-.528-1.674L8.427.401a1 1 0 011.146 0zM9 2.441L5.345 5h7.31z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="global-icon-search" viewBox="0 0 22 22">
            <path fill-rule="evenodd" d="M21.697 20.261a1.028 1.028 0 01.01 1.448 1.034 1.034 0 01-1.448-.01l-4.267-4.267A9.812 9.811 0 010 9.812a9.812 9.811 0 1117.43 6.182zM9.812 18.222A8.41 8.41 0 109.81 1.403a8.41 8.41 0 000 16.82z"></path>
        </symbol>
        <symbol id="icon-info" viewBox="0 0 18 18">
            <path d="m9 0c4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9zm0 7h-1.5l-.11662113.00672773c-.49733868.05776511-.88337887.48043643-.88337887.99327227 0 .47338693.32893365.86994729.77070917.97358929l.1126697.01968298.11662113.00672773h.5v3h-.5l-.11662113.0067277c-.42082504.0488782-.76196299.3590206-.85696816.7639815l-.01968298.1126697-.00672773.1166211.00672773.1166211c.04887817.4208251.35902055.761963.76398144.8569682l.1126697.019683.11662113.0067277h3l.1166211-.0067277c.4973387-.0577651.8833789-.4804365.8833789-.9932723 0-.4733869-.3289337-.8699473-.7707092-.9735893l-.1126697-.019683-.1166211-.0067277h-.5v-4l-.00672773-.11662113c-.04887817-.42082504-.35902055-.76196299-.76398144-.85696816l-.1126697-.01968298zm0-3.25c-.69035594 0-1.25.55964406-1.25 1.25s.55964406 1.25 1.25 1.25 1.25-.55964406 1.25-1.25-.55964406-1.25-1.25-1.25z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="icon-success" viewBox="0 0 18 18">
            <path d="M9 0a9 9 0 110 18A9 9 0 019 0zm3.486 4.982l-4.718 5.506L5.14 8.465a.991.991 0 00-1.423.133 1.06 1.06 0 00.13 1.463l3.407 2.733a1 1 0 001.387-.133l5.385-6.334a1.06 1.06 0 00-.116-1.464.991.991 0 00-1.424.119z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="icon-chevron-down" viewBox="0 0 16 16">
            <path d="m5.58578644 3-3.29289322-3.29289322c-.39052429-.39052429-.39052429-1.02368927 0-1.41421356s1.02368927-.39052429 1.41421356 0l4 4c.39052429.39052429.39052429 1.02368927 0 1.41421356l-4 4c-.39052429.39052429-1.02368927.39052429-1.41421356 0s-.39052429-1.02368927 0-1.41421356z" fill-rule="evenodd" transform="matrix(0 1 -1 0 11 1)"></path>
        </symbol>
        <symbol id="icon-warning" viewBox="0 0 18 18">
            <path d="m9 11.75c.69035594 0 1.25.5596441 1.25 1.25s-.55964406 1.25-1.25 1.25-1.25-.5596441-1.25-1.25.55964406-1.25 1.25-1.25zm.41320045-7.75c.55228475 0 1.00000005.44771525 1.00000005 1l-.0034543.08304548-.3333333 4c-.043191.51829212-.47645714.91695452-.99654578.91695452h-.15973424c-.52008864 0-.95335475-.3986624-.99654576-.91695452l-.33333333-4c-.04586475-.55037702.36312325-1.03372649.91350028-1.07959124l.04148683-.00259031zm-.41320045 14c-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9 4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="icon-plus" viewBox="0 0 16 16">
            <path d="m2.00087166 7h4.99912834v-4.99912834c0-.55276616.44386482-1.00087166 1-1.00087166.55228475 0 1 .44463086 1 1.00087166v4.99912834h4.9991283c.5527662 0 1.0008717.44386482 1.0008717 1 0 .55228475-.4446309 1-1.0008717 1h-4.9991283v4.9991283c0 .5527662-.44386482 1.0008717-1 1.0008717-.55228475 0-1-.4446309-1-1.0008717v-4.9991283h-4.99912834c-.55276616 0-1.00087166-.44386482-1.00087166-1 0-.55228475.44463086-1 1.00087166-1z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="icon-minus" viewBox="0 0 16 16">
            <path d="m2.00087166 7h11.99825664c.5527662 0 1.0008717.44386482 1.0008717 1 0 .55228475-.4446309 1-1.0008717 1h-11.99825664c-.55276616 0-1.00087166-.44386482-1.00087166-1 0-.55228475.44463086-1 1.00087166-1z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="icon-error" viewBox="0 0 18 18">
            <path d="m9 0c4.9705627 0 9 4.02943725 9 9 0 4.9705627-4.0294373 9-9 9-4.97056275 0-9-4.0294373-9-9 0-4.97056275 4.02943725-9 9-9zm2.8630343 4.71100931-2.8630343 2.86303426-2.86303426-2.86303426c-.39658757-.39658757-1.03281091-.39438847-1.4265779-.00062147-.39651227.39651226-.39348876 1.03246767.00062147 1.4265779l2.86303426 2.86303426-2.86303426 2.8630343c-.39658757.3965875-.39438847 1.0328109-.00062147 1.4265779.39651226.3965122 1.03246767.3934887 1.4265779-.0006215l2.86303426-2.8630343 2.8630343 2.8630343c.3965875.3965876 1.0328109.3943885 1.4265779.0006215.3965122-.3965123.3934887-1.0324677-.0006215-1.4265779l-2.8630343-2.8630343 2.8630343-2.86303426c.3965876-.39658757.3943885-1.03281091.0006215-1.4265779-.3965123-.39651227-1.0324677-.39348876-1.4265779.00062147z" fill="currentColor" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="icon-springer-arrow-left">
            <path d="M15 7a1 1 0 000-2H3.385l2.482-2.482a.994.994 0 00.02-1.403 1.001 1.001 0 00-1.417 0L.294 5.292a1.001 1.001 0 000 1.416l4.176 4.177a.991.991 0 001.4.016 1 1 0 00-.003-1.42L3.385 7H15z"></path>
        </symbol>
        <symbol id="icon-springer-arrow-right">
            <path d="M1 7a1 1 0 010-2h11.615l-2.482-2.482a.994.994 0 01-.02-1.403 1.001 1.001 0 011.417 0l4.176 4.177a1.001 1.001 0 010 1.416l-4.176 4.177a.991.991 0 01-1.4.016 1 1 0 01.003-1.42L12.615 7H1z"></path>
        </symbol>
        <symbol id="icon-arrow-up" viewBox="0 0 16 16">
            <path d="m12.716625 4.46975946-4.03074003-4.17620792c-.37758093-.39120768-.98937525-.39160691-1.367372.0000316l-4.03091981 4.1763942c-.37759778.39122514-.38381821 1.01908149-.01600053 1.40017357.37750607.39113012.98772445.3930364 1.37006824-.00310603l2.39538588-2.48183446v11.61478958l.00649339.1166211c.055753.4973387.46370161.8833789.95867408.8833789.49497246 0 .90292107-.3860402.95867408-.8833789l.00649338-.1166211v-11.61478958l2.39518592 2.4816273c.3791392.39282216.9863753.40056173 1.3541929.01946965.3775061-.39113012.3778444-1.02492687-.0001355-1.41654791z" fill-rule="evenodd"></path>
        </symbol>
        <symbol id="icon-tick" viewBox="0 0 24 24">
            <path d="M12,24 C5.372583,24 0,18.627417 0,12 C0,5.372583 5.372583,0 12,0 C18.627417,0 24,5.372583 24,12 C24,18.627417 18.627417,24 12,24 Z M7.657,10.79 C7.45285634,10.6137568 7.18569967,10.5283283 6.91717333,10.5534259 C6.648647,10.5785236 6.40194824,10.7119794 6.234,10.923 C5.87705269,11.3666969 5.93445559,12.0131419 6.364,12.387 L10.261,15.754 C10.6765468,16.112859 11.3037113,16.0695601 11.666,15.657 L17.759,8.713 C18.120307,8.27302248 18.0695334,7.62621189 17.644,7.248 C17.4414817,7.06995024 17.1751516,6.9821166 16.9064461,7.00476032 C16.6377406,7.02740404 16.3898655,7.15856958 16.22,7.368 L10.768,13.489 L7.657,10.79 Z"></path>
        </symbol>
        <symbol id="icon-expand-image" viewBox="0 0 18 18">
            <path d="m7.49754099 11.9178212c.38955542-.3895554.38761957-1.0207846-.00290473-1.4113089-.39324695-.3932469-1.02238878-.3918247-1.41130883-.0029047l-4.10273549 4.1027355.00055454-3.5103985c.00008852-.5603185-.44832171-1.006032-1.00155062-1.0059446-.53903074.0000852-.97857527.4487442-.97866268 1.0021075l-.00093318 5.9072465c-.00008751.553948.44841131 1.001882 1.00174994 1.0017946l5.906983-.0009331c.5539233-.0000875 1.00197907-.4486389 1.00206646-1.0018679.00008515-.5390307-.45026621-.9784332-1.00588841-.9783454l-3.51010549.0005545zm3.00571741-5.83449376c-.3895554.38955541-.3876196 1.02078454.0029047 1.41130883.393247.39324696 1.0223888.39182478 1.4113089.00290473l4.1027355-4.10273549-.0005546 3.5103985c-.0000885.56031852.4483217 1.006032 1.0015506 1.00594461.5390308-.00008516.9785753-.44874418.9786627-1.00210749l.0009332-5.9072465c.0000875-.553948-.4484113-1.00188204-1.0017499-1.00179463l-5.906983.00093313c-.5539233.00008751-1.0019791.44863892-1.0020665 1.00186784-.0000852.53903074.4502662.97843325 1.0058884.97834547l3.5101055-.00055449z" fill="currentColor" fill-rule="evenodd"></path>
        </symbol>
    </svg>

        
    


<script src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/app-es5-bundle-6a48e032fc.js.下载" nomodule="true"></script><script src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/app-es6-bundle-716d0ac638.js.下载" type="module"></script><script src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/global-article-es5-bundle-c4f64c8770.js.下载" nomodule="true"></script><script src="./Ultrafast and memory-efficient alignment of short DNA sequences to the human genome _ Genome Biology _ Full Text_files/global-article-es6-bundle-93d9e3b3d0.js.下载" type="module"></script><a id="canary-test-data" href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25?report=reader" style="display: none;"></a></body></html>